{
 "metadata": {
  "name": ""
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import pd_analysis as pda"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 8
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import pandas as pd\n",
      "#import alib.plots\n",
      "import pandasql\n",
      "import cPickle"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 3
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import psycopg2"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 485
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "conn = psycopg2.connect(database=\"mousegenome\", user=\"adrian\", password=\"adrian\", host=\"localhost\")\n",
      "import pandas.io.sql as sql"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 490
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cur = conn.cursor()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 487
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cur.execute(\"select * from mm9_gene_motifs_distinct where max_motif_score >= 0.7\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 488
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "mm9_gene_motifs = sql.read_frame(\"select * from mm9_gene_motifs_distinct where max_motif_score >= 0.7\", conn)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 491
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "mm9_gene_motifs"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<pre>\n",
        "&ltclass 'pandas.core.frame.DataFrame'&gt\n",
        "Int64Index: 276795 entries, 0 to 276794\n",
        "Data columns (total 6 columns):\n",
        "genename           276795  non-null values\n",
        "chrom              276795  non-null values\n",
        "motif_type         276795  non-null values\n",
        "motif_name         276795  non-null values\n",
        "max_motif_score    276795  non-null values\n",
        "motif_count        276795  non-null values\n",
        "dtypes: float64(1), int64(1), object(4)\n",
        "</pre>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 492,
       "text": [
        "<class 'pandas.core.frame.DataFrame'>\n",
        "Int64Index: 276795 entries, 0 to 276794\n",
        "Data columns (total 6 columns):\n",
        "genename           276795  non-null values\n",
        "chrom              276795  non-null values\n",
        "motif_type         276795  non-null values\n",
        "motif_name         276795  non-null values\n",
        "max_motif_score    276795  non-null values\n",
        "motif_count        276795  non-null values\n",
        "dtypes: float64(1), int64(1), object(4)"
       ]
      }
     ],
     "prompt_number": 492
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# load amalgamated table of statistics by probeset from pickled pandas file.  This file was generated by the pd_calc_stats notebook or script\n",
      "cp_both_wide_info = cPickle.load( open(\"/data/adrian/Dropbox/Projects/Broad/PD_mouse/results/2013_july_9/cp_both_wide_info.pandas.pickle\") )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 4
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "reload(pda)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 16,
       "text": [
        "<module 'pd_analysis' from 'pd_analysis.pyc'>"
       ]
      }
     ],
     "prompt_number": 16
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "factortypes = {}\n",
      "for c in cp_both_wide_info.drop([\"probe_id\", \"symbol\", \"gene_name\"], axis=1).columns:\n",
      "    for (k, v) in dict(pda.tm.d(c)).items():\n",
      "        if isinstance(k, list):\n",
      "            k = tuple(k)\n",
      "        if isinstance(v, list):\n",
      "            v = tuple(v)\n",
      "        if factortypes.has_key(k):\n",
      "            factortypes[k].add(v)\n",
      "        else:\n",
      "            factortypes[k] = set([v])"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 384
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "factortypes"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 385,
       "text": [
        "{u'aim_lm': {u'bh', u'bonf', u'expr_coef', u'pval', u'rsq_adj', u'sign'},\n",
        " u'cg': {u'ctrl', u'exp'},\n",
        " u'cmp': {(u'6-OHDA, acuteHigh', u'6-OHDA, acuteSaline'),\n",
        "  (u'6-OHDA, chronicHigh', u'6-OHDA, acuteHigh'),\n",
        "  (u'6-OHDA, chronicHigh', u'6-OHDA, chronicLow'),\n",
        "  (u'6-OHDA, chronicHigh', u'6-OHDA, chronicSaline'),\n",
        "  (u'6-OHDA, chronicLow', u'6-OHDA, chronicSaline'),\n",
        "  (u'6-OHDA, chronicSaline', u'Ascorbate, chronicSaline')},\n",
        " u'ct': {u'cp101', u'cp73'},\n",
        " u'mc': {u'bh', u'bonf', u'nominal'},\n",
        " u'mg': {u'allchronic', u'chronicHigh', u'chronicLow'},\n",
        " u'st': {u'cv',\n",
        "  u'fc_means',\n",
        "  u'fc_medians',\n",
        "  u'mean',\n",
        "  u'median',\n",
        "  u'pval',\n",
        "  u'stddev',\n",
        "  u't'},\n",
        " u'tt': {u'levene', u'student ttest', u'welch ttest'}}"
       ]
      }
     ],
     "prompt_number": 385
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "test_info = [[\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
      "fc_info = [[\"st\", \"fc_medians\"]]\n",
      "fc_threshold = log2(1.5)\n",
      "\n",
      "summary = {}\n",
      "top_tfs = {}\n",
      "\n",
      "for fc_threshold in [1.5]:\n",
      "    for comparison in factortypes[\"cmp\"]:\n",
      "        for celltype in factortypes[\"ct\"]:\n",
      "            #print comparison\n",
      "            test_query = test_info + [ [\"cmp\", list(comparison)] ] + [ [\"ct\", celltype] ]\n",
      "            fc_query = fc_info + [ [\"cmp\", list(comparison)] ] + [ [\"ct\", celltype] ]\n",
      "            #td = dict(tm.d(c))\n",
      "            if pda.tm.e(test_query) in set(cp_both_wide_info.columns):\n",
      "                print pda.tm.e(test_query)\n",
      "                current_group = cp_both_wide_info.select(\n",
      "                        lambda x: cp_both_wide_info.ix[x, pda.tm.e(test_query) ] < 0.10\n",
      "                              and abs(cp_both_wide_info.ix[x, pda.tm.e(fc_query) ]) >= log2(fc_threshold)\n",
      "                                and isinstance(cp_both_wide_info.ix[x, \"symbol\" ], str) ) \n",
      "                k = pda.tm.e([ [\"cmp\", list(comparison)] ] \n",
      "                                 + [ [\"ct\", celltype] ]\n",
      "                                 + [ [\"dir\", \"any\"]])\n",
      "                try:\n",
      "                    top_tfs[k] = countTFcombos( current_group, 1)\n",
      "                except:\n",
      "                    top_tfs[k] = None\n",
      "                \n",
      "                \n",
      "                current_group = cp_both_wide_info.select(\n",
      "                        lambda x: cp_both_wide_info.ix[x, pda.tm.e(test_query) ] < 0.10\n",
      "                              and cp_both_wide_info.ix[x, pda.tm.e(fc_query) ] >= log2(fc_threshold)\n",
      "                                and isinstance(cp_both_wide_info.ix[x, \"symbol\" ], str) ) \n",
      "                k = pda.tm.e([ [\"cmp\", list(comparison)] ] \n",
      "                                 + [ [\"ct\", celltype] ]\n",
      "                                 + [ [\"dir\", \"up\"]])                \n",
      "                \n",
      "                try:\n",
      "                    top_tfs[ k ] = countTFcombos( current_group, 1)\n",
      "                except:\n",
      "                    top_tfs[ k ] = None\n",
      "\n",
      "                    \n",
      "                current_group = cp_both_wide_info.select(\n",
      "                        lambda x: cp_both_wide_info.ix[x, pda.tm.e(test_query) ] < 0.10\n",
      "                              and cp_both_wide_info.ix[x, pda.tm.e(fc_query) ] <= -log2(fc_threshold)\n",
      "                                and isinstance(cp_both_wide_info.ix[x, \"symbol\" ], str) ) \n",
      "                k = pda.tm.e([ [\"cmp\", list(comparison)] ] \n",
      "                                 + [ [\"ct\", celltype] ]\n",
      "                                 + [ [\"dir\", \"down\"]])                \n",
      "                try:\n",
      "                    top_tfs[ k ] = countTFcombos( current_group, 1)\n",
      "                except:\n",
      "                    top_tfs[ k ] = None\n",
      "                    \n",
      "                \n",
      "#                summary[ (celltype, comparison, fc_threshold, \"changed\") ] = len(uniqueSym( cp_both_wide_info.select(\n",
      "#                        lambda x: cp_both_wide_info.ix[x, tm.e(test_query) ] < 0.10\n",
      "#                              and abs(cp_both_wide_info.ix[x, tm.e(fc_query) ]) >= log2(fc_threshold)\n",
      "#                                and isinstance(cp_both_wide_info.ix[x, \"symbol\" ], str) ).symbol ))\n",
      "#                summary[ (celltype, comparison, fc_threshold, \"up\") ] = len(uniqueSym( cp_both_wide_info.select(\n",
      "#                        lambda x: cp_both_wide_info.ix[x, tm.e(test_query) ] < 0.10\n",
      "#                              and cp_both_wide_info.ix[x, tm.e(fc_query) ] >= log2(fc_threshold)\n",
      "#                                and isinstance(cp_both_wide_info.ix[x, \"symbol\" ], str) ).symbol ))\n",
      "#                summary[ (celltype, comparison, fc_threshold, \"down\") ] = len(uniqueSym( cp_both_wide_info.select(\n",
      "#                        lambda x: cp_both_wide_info.ix[x, tm.e(test_query) ] < 0.10\n",
      "#                              and cp_both_wide_info.ix[x, tm.e(fc_query) ] <= -log2(fc_threshold)\n",
      "#                                and isinstance(cp_both_wide_info.ix[x, \"symbol\" ], str) ).symbol ))\n",
      "            else:\n",
      "                print \"--\""
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
        "5489"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "1628"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "3861"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
        "3928"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "2851"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "1077"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
        "73865"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "33804"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "40276"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "2256"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "1628"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "628"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, acuteHigh\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
        "108649"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "59944"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "49980"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "--"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
        "108341"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "47501"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "61562"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "10738"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "6312"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "4457"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
        "12688"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "8096"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "4592"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
        "0"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "0"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "0"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
        "112094"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "38617"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "74058"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "--"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n"
       ]
      }
     ],
     "prompt_number": 467
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "test_info = [[\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
      "fc_info = [[\"st\", \"fc_medians\"]]\n",
      "fc_threshold = log2(1.5)\n",
      "\n",
      "summary = {}\n",
      "top_motifs = {}\n",
      "\n",
      "for fc_threshold in [1.5]:\n",
      "    for comparison in factortypes[\"cmp\"]:\n",
      "        for celltype in factortypes[\"ct\"]:\n",
      "            #print comparison\n",
      "            test_query = test_info + [ [\"cmp\", list(comparison)] ] + [ [\"ct\", celltype] ]\n",
      "            fc_query = fc_info + [ [\"cmp\", list(comparison)] ] + [ [\"ct\", celltype] ]\n",
      "            #td = dict(tm.d(c))\n",
      "            if pda.tm.e(test_query) in set(cp_both_wide_info.columns):\n",
      "                print pda.tm.e(test_query)\n",
      "                current_group = cp_both_wide_info.select(\n",
      "                        lambda x: cp_both_wide_info.ix[x, pda.tm.e(test_query) ] < 0.10\n",
      "                              and abs(cp_both_wide_info.ix[x, pda.tm.e(fc_query) ]) >= log2(fc_threshold)\n",
      "                                and isinstance(cp_both_wide_info.ix[x, \"symbol\" ], str) ) \n",
      "                k = pda.tm.e([ [\"cmp\", list(comparison)] ] \n",
      "                                 + [ [\"ct\", celltype] ]\n",
      "                                 + [ [\"dir\", \"any\"]])\n",
      "                try:\n",
      "                    top_motifs[k] = countMotifCombos( current_group, 1)\n",
      "                except:\n",
      "                    top_motifs[k] = None\n",
      "                \n",
      "                \n",
      "                current_group = cp_both_wide_info.select(\n",
      "                        lambda x: cp_both_wide_info.ix[x, pda.tm.e(test_query) ] < 0.10\n",
      "                              and cp_both_wide_info.ix[x, pda.tm.e(fc_query) ] >= log2(fc_threshold)\n",
      "                                and isinstance(cp_both_wide_info.ix[x, \"symbol\" ], str) ) \n",
      "                k = pda.tm.e([ [\"cmp\", list(comparison)] ] \n",
      "                                 + [ [\"ct\", celltype] ]\n",
      "                                 + [ [\"dir\", \"up\"]])                \n",
      "                \n",
      "                try:\n",
      "                    top_motifs[ k ] = countMotifCombos( current_group, 1)\n",
      "                except:\n",
      "                    top_motifs[ k ] = None\n",
      "\n",
      "                    \n",
      "                current_group = cp_both_wide_info.select(\n",
      "                        lambda x: cp_both_wide_info.ix[x, pda.tm.e(test_query) ] < 0.10\n",
      "                              and cp_both_wide_info.ix[x, pda.tm.e(fc_query) ] <= -log2(fc_threshold)\n",
      "                                and isinstance(cp_both_wide_info.ix[x, \"symbol\" ], str) ) \n",
      "                k = pda.tm.e([ [\"cmp\", list(comparison)] ] \n",
      "                                 + [ [\"ct\", celltype] ]\n",
      "                                 + [ [\"dir\", \"down\"]])                \n",
      "                try:\n",
      "                    top_motifs[ k ] = countMotifCombos( current_group, 1)\n",
      "                except:\n",
      "                    top_motifs[ k ] = None\n",
      "                    "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
        "3460"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "1161"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "2299"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
        "2107"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "1388"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "719"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
        "44440"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "21454"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "23185"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "1281"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "962"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "319"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, acuteHigh\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
        "66931"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "38759"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "28813"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "66192"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "29880"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "36795"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "6491"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "4233"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "2280"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
        "7263"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "4425"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "2838"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
        "0"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "0"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "0"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]\n",
        "70370"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "23142"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "47617"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n"
       ]
      }
     ],
     "prompt_number": 521
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ks = top_tfs.keys()\n",
      "ks.sort()\n",
      "for k in ks:\n",
      "    print k\n",
      "    if top_tfs[k] is None:\n",
      "        print \"n/a\"\n",
      "    else:\n",
      "        print top_tfs[k][0:10]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "[['cmp', [u'6-OHDA, acuteHigh', u'6-OHDA, acuteSaline']], ['ct', u'cp73'], ['dir', 'any']]\n",
        "        tf  count  total_genes_in_group\n",
        "20     MYC   2238                  4727\n",
        "95    CREM   1972                  4727\n",
        "78    SPI1   1919                  4727\n",
        "169   SOX2   1842                  4727\n",
        "48   RUNX1   1838                  4727\n",
        "26    FLI1   1808                  4727\n",
        "89   HNF4A   1614                  4727\n",
        "193   E2F1   1593                  4727\n",
        "117  NANOG   1496                  4727\n",
        "61    EGR1   1489                  4727\n",
        "[['cmp', [u'6-OHDA, acuteHigh', u'6-OHDA, acuteSaline']], ['ct', u'cp73'], ['dir', 'down']]\n",
        "         tf  count  total_genes_in_group\n",
        "19      MYC   1560                  3338\n",
        "114    SPI1   1305                  3338\n",
        "38    RUNX1   1260                  3338\n",
        "94     CREM   1253                  3338\n",
        "25     FLI1   1241                  3338\n",
        "168    SOX2   1222                  3338\n",
        "88    HNF4A   1117                  3338\n",
        "192    E2F1   1110                  3338\n",
        "184  POU5F1   1003                  3338\n",
        "60     EGR1    984                  3338\n",
        "[['cmp', [u'6-OHDA, acuteHigh', u'6-OHDA, acuteSaline']], ['ct', u'cp73'], ['dir', 'up']]\n",
        "        tf  count  total_genes_in_group\n",
        "95    CREM    732                  1413\n",
        "18     MYC    688                  1413\n",
        "151   SOX2    628                  1413\n",
        "114   SPI1    624                  1413\n",
        "25   CREB1    610                  1413\n",
        "37   RUNX1    587                  1413\n",
        "24    FLI1    574                  1413\n",
        "115  NANOG    521                  1413\n",
        "60    EGR1    513                  1413\n",
        "89   HNF4A    505                  1413\n",
        "[['cmp', [u'6-OHDA, chronicHigh', u'6-OHDA, acuteHigh']], ['ct', u'cp73'], ['dir', 'any']]\n",
        "        tf  count  total_genes_in_group\n",
        "21     MYC   2069                  4491\n",
        "168   SOX2   1862                  4491\n",
        "165   SPI1   1778                  4491\n",
        "41   RUNX1   1772                  4491\n",
        "97    CREM   1737                  4491\n",
        "27    FLI1   1689                  4491\n",
        "91   HNF4A   1547                  4491\n",
        "64    EGR1   1502                  4491\n",
        "82   STAT3   1470                  4491\n",
        "192   E2F1   1453                  4491\n",
        "[['cmp', [u'6-OHDA, chronicHigh', u'6-OHDA, acuteHigh']], ['ct', u'cp73'], ['dir', 'down']]\n",
        "        tf  count  total_genes_in_group\n",
        "147   SOX2    876                  1852\n",
        "20     MYC    869                  1852\n",
        "38   RUNX1    788                  1852\n",
        "78    SPI1    783                  1852\n",
        "176   CREM    738                  1852\n",
        "62    EGR1    736                  1852\n",
        "116  STAT3    711                  1852\n",
        "152   MITF    699                  1852\n",
        "89   HNF4A    699                  1852\n",
        "26    FLI1    697                  1852"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[['cmp', [u'6-OHDA, chronicHigh', u'6-OHDA, acuteHigh']], ['ct', u'cp73'], ['dir', 'up']]\n",
        "         tf  count  total_genes_in_group\n",
        "21      MYC   1226                  2681\n",
        "80     SPI1   1018                  2681\n",
        "97     CREM   1017                  2681\n",
        "168    SOX2   1013                  2681\n",
        "27     FLI1   1013                  2681\n",
        "67    RUNX1   1012                  2681\n",
        "193    E2F1    896                  2681\n",
        "91    HNF4A    867                  2681\n",
        "185  POU5F1    817                  2681\n",
        "82    NANOG    803                  2681\n",
        "[['cmp', [u'6-OHDA, chronicHigh', u'6-OHDA, chronicLow']], ['ct', u'cp101'], ['dir', 'any']]\n",
        "n/a\n",
        "[['cmp', [u'6-OHDA, chronicHigh', u'6-OHDA, chronicLow']], ['ct', u'cp101'], ['dir', 'down']]\n",
        "n/a\n",
        "[['cmp', [u'6-OHDA, chronicHigh', u'6-OHDA, chronicLow']], ['ct', u'cp101'], ['dir', 'up']]\n",
        "n/a\n",
        "[['cmp', [u'6-OHDA, chronicHigh', u'6-OHDA, chronicLow']], ['ct', u'cp73'], ['dir', 'any']]\n",
        "        tf  count  total_genes_in_group\n",
        "145   SOX2    223                   498\n",
        "17     MYC    218                   498\n",
        "110   SPI1    195                   498\n",
        "111  STAT3    189                   498\n",
        "61   RUNX1    186                   498\n",
        "91    CREM    178                   498\n",
        "108   TP63    176                   498\n",
        "75   NANOG    175                   498\n",
        "57    EGR1    175                   498\n",
        "23    FLI1    174                   498\n",
        "[['cmp', [u'6-OHDA, chronicHigh', u'6-OHDA, chronicLow']], ['ct', u'cp73'], ['dir', 'down']]\n",
        "        tf  count  total_genes_in_group\n",
        "16     MYC     87                   198\n",
        "141   SOX2     76                   198\n",
        "109   SPI1     76                   198\n",
        "76   STAT3     75                   198\n",
        "85   HNF4A     71                   198\n",
        "35   RUNX1     67                   198\n",
        "59    EGR1     66                   198\n",
        "22    FLI1     63                   198\n",
        "80    TP53     63                   198\n",
        "110  NANOG     58                   198\n",
        "[['cmp', [u'6-OHDA, chronicHigh', u'6-OHDA, chronicLow']], ['ct', u'cp73'], ['dir', 'up']]\n",
        "        tf  count  total_genes_in_group\n",
        "138   SOX2    147                   300\n",
        "16     MYC    131                   300\n",
        "107   TP63    125                   300\n",
        "90    CREM    122                   300\n",
        "35   RUNX1    119                   300\n",
        "109   SPI1    119                   300\n",
        "74   NANOG    117                   300\n",
        "110  STAT3    114                   300\n",
        "22    FLI1    111                   300\n",
        "161  SUZ12    110                   300\n",
        "[['cmp', [u'6-OHDA, chronicHigh', u'6-OHDA, chronicSaline']], ['ct', u'cp101'], ['dir', 'any']]\n",
        "        tf  count  total_genes_in_group\n",
        "19     MYC    190                   408\n",
        "146   SOX2    178                   408\n",
        "60    EGR1    174                   408\n",
        "65   RUNX1    164                   408\n",
        "94    CREM    163                   408\n",
        "25    FLI1    154                   408\n",
        "115  STAT3    151                   408\n",
        "181     AR    149                   408\n",
        "114   SPI1    146                   408\n",
        "88   HNF4A    143                   408"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[['cmp', [u'6-OHDA, chronicHigh', u'6-OHDA, chronicSaline']], ['ct', u'cp101'], ['dir', 'down']]\n",
        "        tf  count  total_genes_in_group\n",
        "177     AR     83                   166\n",
        "60    EGR1     73                   166\n",
        "18     MYC     71                   166\n",
        "78   STAT3     70                   166\n",
        "143   SOX2     68                   166\n",
        "83    TP53     67                   166\n",
        "147   MITF     62                   166\n",
        "87   HNF4A     61                   166\n",
        "64   RUNX1     61                   166\n",
        "113  NANOG     61                   166\n",
        "[['cmp', [u'6-OHDA, chronicHigh', u'6-OHDA, chronicSaline']], ['ct', u'cp101'], ['dir', 'up']]\n",
        "         tf  count  total_genes_in_group\n",
        "17      MYC    120                   243\n",
        "138    SOX2    111                   243\n",
        "88     CREM    109                   243\n",
        "37    RUNX1    104                   243\n",
        "56     EGR1    101                   243\n",
        "23     FLI1     96                   243\n",
        "71     SPI1     90                   243\n",
        "175  POU5F1     87                   243\n",
        "82    HNF4A     83                   243\n",
        "8     STAT3     81                   243\n",
        "[['cmp', [u'6-OHDA, chronicHigh', u'6-OHDA, chronicSaline']], ['ct', u'cp73'], ['dir', 'any']]\n",
        "        tf  count  total_genes_in_group\n",
        "18     MYC   1988                  4511\n",
        "151   SOX2   1869                  4511\n",
        "163   SPI1   1775                  4511\n",
        "92    CREM   1757                  4511\n",
        "38   RUNX1   1735                  4511\n",
        "26    FLI1   1608                  4511\n",
        "86   HNF4A   1568                  4511\n",
        "59    EGR1   1511                  4511\n",
        "77   STAT3   1499                  4511\n",
        "113  NANOG   1454                  4511\n",
        "[['cmp', [u'6-OHDA, chronicHigh', u'6-OHDA, chronicSaline']], ['ct', u'cp73'], ['dir', 'down']]\n",
        "        tf  count  total_genes_in_group\n",
        "17     MYC   1145                  2668\n",
        "147   SOX2   1103                  2668\n",
        "164   SPI1   1007                  2668\n",
        "39   RUNX1    988                  2668\n",
        "88   HNF4A    954                  2668\n",
        "94    CREM    902                  2668\n",
        "78   STAT3    894                  2668\n",
        "25    FLI1    893                  2668\n",
        "59    EGR1    866                  2668\n",
        "183     AR    855                  2668\n",
        "[['cmp', [u'6-OHDA, chronicHigh', u'6-OHDA, chronicSaline']], ['ct', u'cp73'], ['dir', 'up']]\n",
        "        tf  count  total_genes_in_group\n",
        "93    CREM    868                  1871\n",
        "18     MYC    860                  1871\n",
        "75    SPI1    780                  1871\n",
        "152   SOX2    778                  1871\n",
        "38   RUNX1    761                  1871\n",
        "25    FLI1    725                  1871\n",
        "26   CREB1    701                  1871\n",
        "59    EGR1    653                  1871\n",
        "114  NANOG    641                  1871\n",
        "87   HNF4A    625                  1871"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[['cmp', [u'6-OHDA, chronicLow', u'6-OHDA, chronicSaline']], ['ct', u'cp101'], ['dir', 'any']]\n",
        "         tf  count  total_genes_in_group\n",
        "14      MYC     42                    76\n",
        "132    SOX2     41                    76\n",
        "169  POU5F1     36                    76\n",
        "88     CREM     35                    76\n",
        "107   STAT3     33                    76\n",
        "106    SPI1     31                    76\n",
        "153    KLF4     30                    76\n",
        "155   SUZ12     30                    76\n",
        "53     EGR1     30                    76\n",
        "73    NANOG     28                    76\n",
        "[['cmp', [u'6-OHDA, chronicLow', u'6-OHDA, chronicSaline']], ['ct', u'cp101'], ['dir', 'down']]\n",
        "         tf  count  total_genes_in_group\n",
        "123   SUZ12     13                    26\n",
        "13      MYC     12                    26\n",
        "135      AR     12                    26\n",
        "107    SOX2     11                    26\n",
        "136  POU5F1     10                    26\n",
        "126   SIN3B      9                    26\n",
        "44     EGR1      9                    26\n",
        "70     CREM      9                    26\n",
        "97    EP300      9                    26\n",
        "61     TP53      9                    26\n",
        "[['cmp', [u'6-OHDA, chronicLow', u'6-OHDA, chronicSaline']], ['ct', u'cp101'], ['dir', 'up']]\n",
        "         tf  count  total_genes_in_group\n",
        "13      MYC     30                    50\n",
        "129    SOX2     30                    50\n",
        "82     CREM     26                    50\n",
        "161  POU5F1     26                    50\n",
        "100    SPI1     25                    50\n",
        "101   STAT3     24                    50\n",
        "11     KLF4     23                    50\n",
        "158    REST     22                    50\n",
        "31    RUNX1     21                    50\n",
        "51     EGR1     21                    50\n",
        "[['cmp', [u'6-OHDA, chronicLow', u'6-OHDA, chronicSaline']], ['ct', u'cp73'], ['dir', 'any']]\n",
        "        tf  count  total_genes_in_group\n",
        "17     MYC   1307                  2980\n",
        "147   SOX2   1284                  2980\n",
        "78    SPI1   1191                  2980\n",
        "40   RUNX1   1172                  2980\n",
        "95    CREM   1160                  2980\n",
        "89   HNF4A   1081                  2980\n",
        "25    FLI1   1071                  2980\n",
        "62    EGR1   1053                  2980\n",
        "81   STAT3   1028                  2980\n",
        "116  NANOG    993                  2980\n",
        "[['cmp', [u'6-OHDA, chronicLow', u'6-OHDA, chronicSaline']], ['ct', u'cp73'], ['dir', 'down']]\n",
        "        tf  count  total_genes_in_group\n",
        "152   SOX2    737                  1683\n",
        "19     MYC    700                  1683\n",
        "78    SPI1    646                  1683\n",
        "38   RUNX1    642                  1683\n",
        "90   HNF4A    638                  1683\n",
        "81   STAT3    593                  1683\n",
        "61    EGR1    576                  1683\n",
        "25    FLI1    573                  1683\n",
        "184     AR    569                  1683\n",
        "147   MITF    558                  1683"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[['cmp', [u'6-OHDA, chronicLow', u'6-OHDA, chronicSaline']], ['ct', u'cp73'], ['dir', 'up']]\n",
        "        tf  count  total_genes_in_group\n",
        "95    CREM    629                  1307\n",
        "21     MYC    614                  1307\n",
        "151   SOX2    553                  1307\n",
        "77    SPI1    547                  1307\n",
        "66   RUNX1    536                  1307\n",
        "27    FLI1    498                  1307\n",
        "28   CREB1    487                  1307\n",
        "61    EGR1    478                  1307\n",
        "115  NANOG    449                  1307\n",
        "89   HNF4A    445                  1307\n",
        "[['cmp', [u'6-OHDA, chronicSaline', u'Ascorbate, chronicSaline']], ['ct', u'cp101'], ['dir', 'any']]\n",
        "        tf  count  total_genes_in_group\n",
        "161  SUZ12     73                   154\n",
        "76   STAT3     71                   154\n",
        "139   SOX2     63                   154\n",
        "60    EGR1     59                   154\n",
        "16     MYC     58                   154\n",
        "56    MITF     57                   154\n",
        "175     AR     57                   154\n",
        "98   SMAD4     54                   154\n",
        "81    TP53     53                   154\n",
        "36   RUNX1     53                   154\n",
        "[['cmp', [u'6-OHDA, chronicSaline', u'Ascorbate, chronicSaline']], ['ct', u'cp101'], ['dir', 'down']]\n",
        "         tf  count  total_genes_in_group\n",
        "155      AR     22                    51\n",
        "143   SUZ12     21                    51\n",
        "52     EGR1     20                    51\n",
        "122    SOX2     18                    51\n",
        "69    STAT3     18                    51\n",
        "72     TP53     16                    51\n",
        "14      MYC     16                    51\n",
        "97     SPI1     15                    51\n",
        "126    MITF     15                    51\n",
        "16   SETDB1     14                    51\n",
        "[['cmp', [u'6-OHDA, chronicSaline', u'Ascorbate, chronicSaline']], ['ct', u'cp101'], ['dir', 'up']]\n",
        "        tf  count  total_genes_in_group\n",
        "111  STAT3     53                   103\n",
        "161  SUZ12     52                   103\n",
        "143   SOX2     45                   103\n",
        "98   SMAD4     45                   103\n",
        "60   RUNX1     43                   103\n",
        "52    MITF     42                   103\n",
        "16     MYC     42                   103\n",
        "174   REST     40                   103\n",
        "134   MTF2     40                   103\n",
        "92    CREM     39                   103\n",
        "[['cmp', [u'6-OHDA, chronicSaline', u'Ascorbate, chronicSaline']], ['ct', u'cp73'], ['dir', 'any']]\n",
        "         tf  count  total_genes_in_group\n",
        "138    SOX2     97                   220\n",
        "160   SUZ12     96                   220\n",
        "57     EGR1     94                   220\n",
        "62    RUNX1     88                   220\n",
        "175      AR     83                   220\n",
        "108    TP63     82                   220\n",
        "87    HNF4A     81                   220\n",
        "82     TP53     76                   220\n",
        "17      MYC     76                   220\n",
        "176  POU5F1     76                   220"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[['cmp', [u'6-OHDA, chronicSaline', u'Ascorbate, chronicSaline']], ['ct', u'cp73'], ['dir', 'down']]\n",
        "        tf  count  total_genes_in_group\n",
        "57    EGR1     69                   150\n",
        "140   SOX2     65                   150\n",
        "157  SUZ12     63                   150\n",
        "62   RUNX1     61                   150\n",
        "171     AR     58                   150\n",
        "136   MITF     57                   150\n",
        "87   HNF4A     56                   150\n",
        "111  STAT3     55                   150\n",
        "82    TP53     54                   150\n",
        "77   NANOG     53                   150\n",
        "[['cmp', [u'6-OHDA, chronicSaline', u'Ascorbate, chronicSaline']], ['ct', u'cp73'], ['dir', 'up']]\n",
        "         tf  count  total_genes_in_group\n",
        "152   SUZ12     33                    70\n",
        "132    SOX2     32                    70\n",
        "104    TP63     30                    70\n",
        "60    RUNX1     27                    70\n",
        "16      MYC     26                    70\n",
        "83    HNF4A     25                    70\n",
        "56     EGR1     25                    70\n",
        "164      AR     25                    70\n",
        "165  POU5F1     24                    70\n",
        "148    SPI1     23                    70\n"
       ]
      }
     ],
     "prompt_number": 432
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import scipy.stats"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 435
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "scipy.stats.hypergeom.sf( 33, 22000, 5042, 70) "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 436,
       "text": [
        "2.2422561041688134e-06"
       ]
      }
     ],
     "prompt_number": 436
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "total_genes = 22000"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 439
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "\n",
      "\n",
      "p = overlaps.apply( lambda x: (\n",
      "      scipy.stats.hypergeom.sf(\n",
      "          x.ix[\"match_count\"]-1, # number of differentially expressed genes in set\n",
      "          total_genes,           # total number of genes\n",
      "          x.ix[\"size of set\"],   # number of genes in current set\n",
      "          len( test_set ))),     # total number of genes in test set\n",
      "                               axis=1)\n",
      "    p = pd.DataFrame(p, columns=[\"hypergeom p-val\"])"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for k in top_motifs.keys():\n",
      "    if top_motifs[k] is not None:\n",
      "        print k\n",
      "        print top_motifs[k].merge(mm9_motif_gene_counts, \n",
      "                                                left_on=\"motif_name\", \n",
      "                                                right_index=True).sort_index(by=\"pval\", ascending=True)[0:20]g"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"up\"]]\n",
        "            motif_name       motif_name_x  count  total_genes_in_group  \\\n",
        "22              SP1.p2             SP1.p2     61                    70   \n",
        "101           PATZ1.p2           PATZ1.p2     53                    70   \n",
        "40              MAZ.p2             MAZ.p2     54                    70   \n",
        "13            TFDP1.p2           TFDP1.p2     43                    70   \n",
        "121          TFAP2B.p2          TFAP2B.p2     34                    70   \n",
        "8         MYFfamily.p2       MYFfamily.p2     19                    70   \n",
        "93            MYOD1.p2           MYOD1.p2     12                    70   \n",
        "125         E2F1..5.p2         E2F1..5.p2      7                    70   \n",
        "1              KLF4.p3            KLF4.p3     48                    70   \n",
        "54         SNAI1..3.p2        SNAI1..3.p2     23                    70   \n",
        "105            PAX5.p2            PAX5.p2     30                    70   \n",
        "41             HIC1.p2            HIC1.p2     25                    70   \n",
        "35   AHR_ARNT_ARNT2.p2  AHR_ARNT_ARNT2.p2     14                    70   \n",
        "123            MTF1.p2            MTF1.p2     19                    70   \n",
        "23            GTF2I.p2           GTF2I.p2     33                    70   \n",
        "99            ZBTB6.p2           ZBTB6.p2      9                    70   \n",
        "75             MAFB.p2            MAFB.p2     19                    70   \n",
        "113         NHLH1,2.p2         NHLH1,2.p2     17                    70   \n",
        "78         GTF2A1,2.p2        GTF2A1,2.p2      5                    70   \n",
        "9              NFE2.p2            NFE2.p2      5                    70   \n",
        "\n",
        "             pval  motif_gene_count       motif_name_y  \n",
        "22   6.524818e-09             12044             SP1.p2  \n",
        "101  8.755004e-07             10334           PATZ1.p2  \n",
        "40   1.832899e-06             10878             MAZ.p2  \n",
        "13   2.185568e-05              8067           TFDP1.p2  \n",
        "121  2.460102e-04              6217          TFAP2B.p2  \n",
        "8    5.129105e-04              2672       MYFfamily.p2  \n",
        "93   8.193439e-04              1312           MYOD1.p2  \n",
        "125  8.254138e-04               480         E2F1..5.p2  \n",
        "1    9.716530e-04             10899            KLF4.p3  \n",
        "54   1.737088e-03              3921        SNAI1..3.p2  \n",
        "105  2.076146e-03              5810            PAX5.p2  \n",
        "41   2.331597e-03              4518            HIC1.p2  \n",
        "35   3.477976e-03              1983  AHR_ARNT_ARNT2.p2  \n",
        "123  3.609777e-03              3146            MTF1.p2  \n",
        "23   3.814036e-03              6874           GTF2I.p2  \n",
        "99   6.386588e-03              1063           ZBTB6.p2  \n",
        "75   7.421592e-03              3359            MAFB.p2  \n",
        "113  7.908756e-03              2886         NHLH1,2.p2  \n",
        "78   1.128368e-02               424        GTF2A1,2.p2  \n",
        "9    1.261010e-02               436            NFE2.p2  \n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"any\"]]\n",
        "            motif_name       motif_name_x  count  total_genes_in_group  \\\n",
        "142          TFAP2B.p2          TFAP2B.p2    103                   220   \n",
        "14            TFDP1.p2           TFDP1.p2    119                   220   \n",
        "22              SP1.p2             SP1.p2    156                   220   \n",
        "46              MAZ.p2             MAZ.p2    143                   220   \n",
        "105      TFAP2{A,C}.p2      TFAP2{A,C}.p2    114                   220   \n",
        "88             MAFB.p2            MAFB.p2     60                   220   \n",
        "48             HIC1.p2            HIC1.p2     73                   220   \n",
        "119           PATZ1.p2           PATZ1.p2    135                   220   \n",
        "23            GTF2I.p2           GTF2I.p2     99                   220   \n",
        "63         SNAI1..3.p2        SNAI1..3.p2     64                   220   \n",
        "131         NHLH1,2.p2         NHLH1,2.p2     51                   220   \n",
        "39   AHR_ARNT_ARNT2.p2  AHR_ARNT_ARNT2.p2     39                   220   \n",
        "66       TCF4_dimer.p2      TCF4_dimer.p2     71                   220   \n",
        "120            LMO2.p2            LMO2.p2     33                   220   \n",
        "109           MYOD1.p2           MYOD1.p2     27                   220   \n",
        "144            MTF1.p2            MTF1.p2     50                   220   \n",
        "55             MZF1.p2            MZF1.p2    109                   220   \n",
        "65           ZFP161.p2          ZFP161.p2     51                   220   \n",
        "8         MYFfamily.p2       MYFfamily.p2     43                   220   \n",
        "146         E2F1..5.p2         E2F1..5.p2     13                   220   \n",
        "\n",
        "             pval  motif_gene_count       motif_name_y  \n",
        "142  3.354500e-09              6217          TFAP2B.p2  \n",
        "14   9.385966e-08              8067           TFDP1.p2  \n",
        "22   5.780760e-07             12044             SP1.p2  \n",
        "46   2.114417e-06             10878             MAZ.p2  \n",
        "105  3.127564e-06              8076      TFAP2{A,C}.p2  \n",
        "88   3.139023e-06              3359            MAFB.p2  \n",
        "48   7.353396e-06              4518            HIC1.p2  \n",
        "119  1.145116e-05             10334           PATZ1.p2  \n",
        "23   1.177612e-05              6874           GTF2I.p2  \n",
        "63   2.587288e-05              3921        SNAI1..3.p2  \n",
        "131  2.947899e-05              2886         NHLH1,2.p2  \n",
        "39   3.240285e-05              1983  AHR_ARNT_ARNT2.p2  \n",
        "66   1.153681e-04              4716      TCF4_dimer.p2  \n",
        "120  1.841044e-04              1701            LMO2.p2  \n",
        "109  3.004484e-04              1312           MYOD1.p2  \n",
        "144  4.993781e-04              3146            MTF1.p2  \n",
        "55   6.592152e-04              8515            MZF1.p2  \n",
        "65   9.570400e-04              3321          ZFP161.p2  \n",
        "8    1.049781e-03              2672       MYFfamily.p2  \n",
        "146  1.147377e-03               480         E2F1..5.p2  "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"any\"]]\n",
        "                  motif_name             motif_name_x  count  \\\n",
        "14                  TFDP1.p2                 TFDP1.p2     45   \n",
        "81                   MAFB.p2                  MAFB.p2     25   \n",
        "108                 PATZ1.p2                 PATZ1.p2     50   \n",
        "96             TFAP2{A,C}.p2            TFAP2{A,C}.p2     42   \n",
        "34         AHR_ARNT_ARNT2.p2        AHR_ARNT_ARNT2.p2     16   \n",
        "22                    SP1.p2                   SP1.p2     55   \n",
        "127                TFAP2B.p2                TFAP2B.p2     34   \n",
        "114                 CREB1.p2                 CREB1.p2     11   \n",
        "40                   SPZ1.p2                  SPZ1.p2     14   \n",
        "99                   ATF6.p2                  ATF6.p2      9   \n",
        "42                    MAZ.p2                   MAZ.p2     50   \n",
        "124           SOX{8,9,10}.p2           SOX{8,9,10}.p2     10   \n",
        "48                    JUN.p2                   JUN.p2     12   \n",
        "64   HBP1_HMGB_SSRP1_UBTF.p2  HBP1_HMGB_SSRP1_UBTF.p2     13   \n",
        "1                    KLF4.p3                  KLF4.p3     49   \n",
        "97             NFY{A,B,C}.p2            NFY{A,B,C}.p2     18   \n",
        "116                  ZEB1.p2                  ZEB1.p2     15   \n",
        "60                 ZNF384.p2                ZNF384.p2      6   \n",
        "109                  LMO2.p2                  LMO2.p2     12   \n",
        "123            ATF5_CREB3.p2            ATF5_CREB3.p2     13   \n",
        "\n",
        "     total_genes_in_group      pval  motif_gene_count             motif_name_y  \n",
        "14                     76  0.000052              8067                 TFDP1.p2  \n",
        "81                     76  0.000097              3359                  MAFB.p2  \n",
        "108                    76  0.000716             10334                 PATZ1.p2  \n",
        "96                     76  0.000738              8076            TFAP2{A,C}.p2  \n",
        "34                     76  0.001050              1983        AHR_ARNT_ARNT2.p2  \n",
        "22                     76  0.001204             12044                   SP1.p2  \n",
        "127                    76  0.001566              6217                TFAP2B.p2  \n",
        "114                    76  0.001610              1125                 CREB1.p2  \n",
        "40                     76  0.002449              1750                  SPZ1.p2  \n",
        "99                     76  0.002838               862                  ATF6.p2  \n",
        "42                     76  0.002916             10878                   MAZ.p2  \n",
        "124                    76  0.003843              1077           SOX{8,9,10}.p2  \n",
        "48                     76  0.004444              1473                   JUN.p2  \n",
        "64                     76  0.005099              1696  HBP1_HMGB_SSRP1_UBTF.p2  \n",
        "1                      76  0.006105             10899                  KLF4.p3  \n",
        "97                     76  0.008070              2875            NFY{A,B,C}.p2  \n",
        "116                    76  0.010211              2274                  ZEB1.p2  \n",
        "60                     76  0.012274               554                ZNF384.p2  \n",
        "109                    76  0.013366              1701                  LMO2.p2  \n",
        "123                    76  0.015628              1951            ATF5_CREB3.p2  \n",
        "[[\"cmp\", [\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"down\"]]\n",
        "               motif_name          motif_name_x  count  total_genes_in_group  \\\n",
        "96   ELK1,4_GABP{A,B1}.p3  ELK1,4_GABP{A,B1}.p3   1076                  3338   \n",
        "155               NRF1.p2               NRF1.p2    752                  3338   \n",
        "16               TFDP1.p2              TFDP1.p2   1621                  3338   \n",
        "28                 SP1.p2                SP1.p2   2221                  3338   \n",
        "12                 EHF.p2                EHF.p2    733                  3338   \n",
        "68            ELF1,2,4.p2           ELF1,2,4.p2   1198                  3338   \n",
        "7               ZNF143.p2             ZNF143.p2    447                  3338   \n",
        "172             TFAP2B.p2             TFAP2B.p2   1264                  3338   \n",
        "57                HIC1.p2               HIC1.p2    970                  3338   \n",
        "128         TFAP2{A,C}.p2         TFAP2{A,C}.p2   1547                  3338   \n",
        "1                 KLF4.p3               KLF4.p3   1955                  3338   \n",
        "80              ZFP161.p2             ZFP161.p2    724                  3338   \n",
        "55                 MAZ.p2                MAZ.p2   1934                  3338   \n",
        "150               PAX5.p2               PAX5.p2   1133                  3338   \n",
        "164              PATZ1.p2              PATZ1.p2   1829                  3338   \n",
        "95             EGR1..3.p2            EGR1..3.p2    761                  3338   \n",
        "104                FEV.p2                FEV.p2    799                  3338   \n",
        "73              ETS1,2.p2             ETS1,2.p2    501                  3338   \n",
        "129         NFY{A,B,C}.p2         NFY{A,B,C}.p2    567                  3338   \n",
        "47      AHR_ARNT_ARNT2.p2     AHR_ARNT_ARNT2.p2    402                  3338   \n",
        "\n",
        "             pval  motif_gene_count          motif_name_y  \n",
        "96   1.625563e-60              4645  ELK1,4_GABP{A,B1}.p3  \n",
        "155  4.106108e-53              3000               NRF1.p2  \n",
        "16   8.072881e-53              8067              TFDP1.p2  \n",
        "28   3.789479e-51             12044                SP1.p2  \n",
        "12   1.106859e-45              3024                EHF.p2  \n",
        "68   9.071453e-44              5707           ELF1,2,4.p2  \n",
        "7    2.280495e-39              1642             ZNF143.p2  \n",
        "172  2.878921e-39              6217             TFAP2B.p2  \n",
        "57   1.432830e-37              4518               HIC1.p2  \n",
        "128  1.798724e-35              8076         TFAP2{A,C}.p2  \n",
        "1    4.760335e-30             10899               KLF4.p3  \n",
        "80   9.451829e-29              3321             ZFP161.p2  \n",
        "55   8.351352e-27             10878                MAZ.p2  \n",
        "150  5.823536e-26              5810               PAX5.p2  \n",
        "164  5.486922e-23             10334              PATZ1.p2  \n",
        "95   1.176486e-16              3885            EGR1..3.p2  \n",
        "104  1.564528e-14              4192                FEV.p2  \n",
        "73   1.475258e-13              2463             ETS1,2.p2  \n",
        "129  9.134318e-13              2875         NFY{A,B,C}.p2  \n",
        "47   9.301654e-11              1983     AHR_ARNT_ARNT2.p2  "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"down\"]]\n",
        "            motif_name       motif_name_x  count  total_genes_in_group  \\\n",
        "17            TFDP1.p2           TFDP1.p2    805                  1683   \n",
        "172          TFAP2B.p2          TFAP2B.p2    633                  1683   \n",
        "129      TFAP2{A,C}.p2      TFAP2{A,C}.p2    775                  1683   \n",
        "29              SP1.p2             SP1.p2   1075                  1683   \n",
        "58             HIC1.p2            HIC1.p2    466                  1683   \n",
        "96          EGR1..3.p2         EGR1..3.p2    410                  1683   \n",
        "145           PATZ1.p2           PATZ1.p2    920                  1683   \n",
        "150            PAX5.p2            PAX5.p2    555                  1683   \n",
        "82           ZFP161.p2          ZFP161.p2    338                  1683   \n",
        "56              MAZ.p2             MAZ.p2    944                  1683   \n",
        "1              KLF4.p3            KLF4.p3    937                  1683   \n",
        "174            MTF1.p2            MTF1.p2    305                  1683   \n",
        "30            GTF2I.p2           GTF2I.p2    608                  1683   \n",
        "48   AHR_ARNT_ARNT2.p2  AHR_ARNT_ARNT2.p2    200                  1683   \n",
        "108            MAFB.p2            MAFB.p2    311                  1683   \n",
        "155            NRF1.p2            NRF1.p2    280                  1683   \n",
        "7            ZNF143.p2          ZNF143.p2    164                  1683   \n",
        "184     bHLH_family.p2     bHLH_family.p2    235                  1683   \n",
        "107          ZNF148.p2          ZNF148.p2    272                  1683   \n",
        "176         E2F1..5.p2         E2F1..5.p2     56                  1683   \n",
        "\n",
        "             pval  motif_gene_count       motif_name_y  \n",
        "17   1.267879e-22              8067           TFDP1.p2  \n",
        "172  3.439279e-18              6217          TFAP2B.p2  \n",
        "129  1.975036e-16              8076      TFAP2{A,C}.p2  \n",
        "29   1.829579e-15             12044             SP1.p2  \n",
        "58   1.688551e-13              4518            HIC1.p2  \n",
        "96   3.311743e-13              3885         EGR1..3.p2  \n",
        "145  2.933514e-11             10334           PATZ1.p2  \n",
        "150  2.826399e-10              5810            PAX5.p2  \n",
        "82   5.787758e-09              3321          ZFP161.p2  \n",
        "56   7.870792e-09             10878             MAZ.p2  \n",
        "1    9.158677e-08             10899            KLF4.p3  \n",
        "174  3.524783e-06              3146            MTF1.p2  \n",
        "30   4.978784e-06              6874           GTF2I.p2  \n",
        "48   2.237619e-05              1983  AHR_ARNT_ARNT2.p2  \n",
        "108  1.112607e-04              3359            MAFB.p2  \n",
        "155  1.540419e-04              3000            NRF1.p2  \n",
        "7    2.076007e-04              1642          ZNF143.p2  \n",
        "184  7.188711e-04              2529     bHLH_family.p2  \n",
        "107  8.192776e-04              2982          ZNF148.p2  \n",
        "176  1.049334e-03               480         E2F1..5.p2  \n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"any\"]]\n",
        "               motif_name          motif_name_x  count  total_genes_in_group  \\\n",
        "17               TFDP1.p2              TFDP1.p2   1538                  2980   \n",
        "173             TFAP2B.p2             TFAP2B.p2   1219                  2980   \n",
        "28                 SP1.p2                SP1.p2   2017                  2980   \n",
        "130         TFAP2{A,C}.p2         TFAP2{A,C}.p2   1475                  2980   \n",
        "146              PATZ1.p2              PATZ1.p2   1719                  2980   \n",
        "151               PAX5.p2               PAX5.p2   1070                  2980   \n",
        "57                 MAZ.p2                MAZ.p2   1784                  2980   \n",
        "97             EGR1..3.p2            EGR1..3.p2    772                  2980   \n",
        "1                 KLF4.p3               KLF4.p3   1784                  2980   \n",
        "59                HIC1.p2               HIC1.p2    867                  2980   \n",
        "83              ZFP161.p2             ZFP161.p2    659                  2980   \n",
        "109               MAFB.p2               MAFB.p2    651                  2980   \n",
        "167         ATF5_CREB3.p2         ATF5_CREB3.p2    418                  2980   \n",
        "70            ELF1,2,4.p2           ELF1,2,4.p2    987                  2980   \n",
        "29               GTF2I.p2              GTF2I.p2   1141                  2980   \n",
        "175               MTF1.p2               MTF1.p2    582                  2980   \n",
        "49      AHR_ARNT_ARNT2.p2     AHR_ARNT_ARNT2.p2    394                  2980   \n",
        "153               PAX2.p2               PAX2.p2    345                  2980   \n",
        "156               NRF1.p2               NRF1.p2    548                  2980   \n",
        "98   ELK1,4_GABP{A,B1}.p3  ELK1,4_GABP{A,B1}.p3    795                  2980   \n",
        "\n",
        "             pval  motif_gene_count          motif_name_y  \n",
        "17   7.375352e-72              8067              TFDP1.p2  \n",
        "173  8.323489e-58              6217             TFAP2B.p2  \n",
        "28   5.112842e-54             12044                SP1.p2  \n",
        "130  2.574002e-53              8076         TFAP2{A,C}.p2  \n",
        "146  1.409200e-36             10334              PATZ1.p2  \n",
        "151  5.093879e-35              5810               PAX5.p2  \n",
        "57   8.693720e-35             10878                MAZ.p2  \n",
        "97   3.141378e-34              3885            EGR1..3.p2  \n",
        "1    3.431916e-34             10899               KLF4.p3  \n",
        "59   2.510212e-33              4518               HIC1.p2  \n",
        "83   2.374613e-28              3321             ZFP161.p2  \n",
        "109  4.743462e-25              3359               MAFB.p2  \n",
        "167  7.465878e-24              1951         ATF5_CREB3.p2  \n",
        "70   3.045974e-21              5707           ELF1,2,4.p2  \n",
        "29   8.975623e-19              6874              GTF2I.p2  \n",
        "175  1.752052e-17              3146               MTF1.p2  \n",
        "49   1.437761e-16              1983     AHR_ARNT_ARNT2.p2  \n",
        "153  2.706949e-16              1690               PAX2.p2  \n",
        "156  2.641390e-15              3000               NRF1.p2  \n",
        "98   3.265509e-15              4645  ELK1,4_GABP{A,B1}.p3  "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"any\"]]\n",
        "               motif_name          motif_name_x  count  total_genes_in_group  \\\n",
        "16               TFDP1.p2              TFDP1.p2   2388                  4727   \n",
        "96   ELK1,4_GABP{A,B1}.p3  ELK1,4_GABP{A,B1}.p3   1532                  4727   \n",
        "28                 SP1.p2                SP1.p2   3201                  4727   \n",
        "172             TFAP2B.p2             TFAP2B.p2   1858                  4727   \n",
        "128         TFAP2{A,C}.p2         TFAP2{A,C}.p2   2277                  4727   \n",
        "68            ELF1,2,4.p2           ELF1,2,4.p2   1720                  4727   \n",
        "12                 EHF.p2                EHF.p2   1024                  4727   \n",
        "155               NRF1.p2               NRF1.p2   1013                  4727   \n",
        "57                HIC1.p2               HIC1.p2   1392                  4727   \n",
        "1                 KLF4.p3               KLF4.p3   2823                  4727   \n",
        "150               PAX5.p2               PAX5.p2   1668                  4727   \n",
        "55                 MAZ.p2                MAZ.p2   2793                  4727   \n",
        "145              PATZ1.p2              PATZ1.p2   2673                  4727   \n",
        "81              ZFP161.p2             ZFP161.p2   1041                  4727   \n",
        "166         ATF5_CREB3.p2         ATF5_CREB3.p2    677                  4727   \n",
        "95             EGR1..3.p2            EGR1..3.p2   1151                  4727   \n",
        "7               ZNF143.p2             ZNF143.p2    569                  4727   \n",
        "107               MAFB.p2               MAFB.p2    992                  4727   \n",
        "154              CREB1.p2              CREB1.p2    409                  4727   \n",
        "47      AHR_ARNT_ARNT2.p2     AHR_ARNT_ARNT2.p2    637                  4727   \n",
        "\n",
        "              pval  motif_gene_count          motif_name_y  \n",
        "16   1.636115e-107              8067              TFDP1.p2  \n",
        "96    2.500814e-95              4645  ELK1,4_GABP{A,B1}.p3  \n",
        "28    4.687757e-93             12044                SP1.p2  \n",
        "172   1.813149e-77              6217             TFAP2B.p2  \n",
        "128   1.774750e-74              8076         TFAP2{A,C}.p2  \n",
        "68    7.906920e-73              5707           ELF1,2,4.p2  \n",
        "12    3.050967e-65              3024                EHF.p2  \n",
        "155   1.017087e-63              3000               NRF1.p2  \n",
        "57    6.244297e-62              4518               HIC1.p2  \n",
        "1     1.156673e-56             10899               KLF4.p3  \n",
        "150   5.646193e-53              5810               PAX5.p2  \n",
        "55    5.359037e-51             10878                MAZ.p2  \n",
        "145   2.814602e-50             10334              PATZ1.p2  \n",
        "81    1.408429e-47              3321             ZFP161.p2  \n",
        "166   1.943342e-45              1951         ATF5_CREB3.p2  \n",
        "95    5.094276e-40              3885            EGR1..3.p2  \n",
        "7     1.152374e-37              1642             ZNF143.p2  \n",
        "107   4.004996e-33              3359               MAFB.p2  \n",
        "154   6.327255e-32              1125              CREB1.p2  \n",
        "47    3.947223e-31              1983     AHR_ARNT_ARNT2.p2  \n",
        "[[\"cmp\", [\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"up\"]]\n",
        "               motif_name          motif_name_x  count  total_genes_in_group  \\\n",
        "162         ATF5_CREB3.p2         ATF5_CREB3.p2    370                  1413   \n",
        "150              CREB1.p2              CREB1.p2    213                  1413   \n",
        "17               TFDP1.p2              TFDP1.p2    776                  1413   \n",
        "47                ATF4.p2               ATF4.p2    185                  1413   \n",
        "64                 JUN.p2                JUN.p2    229                  1413   \n",
        "125         TFAP2{A,C}.p2         TFAP2{A,C}.p2    743                  1413   \n",
        "30                 SP1.p2                SP1.p2    995                  1413   \n",
        "167             TFAP2B.p2             TFAP2B.p2    606                  1413   \n",
        "104               MAFB.p2               MAFB.p2    373                  1413   \n",
        "148               PAX2.p2               PAX2.p2    229                  1413   \n",
        "94   ELK1,4_GABP{A,B1}.p3  ELK1,4_GABP{A,B1}.p3    465                  1413   \n",
        "141              PATZ1.p2              PATZ1.p2    857                  1413   \n",
        "146               PAX5.p2               PAX5.p2    544                  1413   \n",
        "93             EGR1..3.p2            EGR1..3.p2    398                  1413   \n",
        "1                 KLF4.p3               KLF4.p3    882                  1413   \n",
        "68            ELF1,2,4.p2           ELF1,2,4.p2    532                  1413   \n",
        "49      AHR_ARNT_ARNT2.p2     AHR_ARNT_ARNT2.p2    239                  1413   \n",
        "57                 MAZ.p2                MAZ.p2    873                  1413   \n",
        "59                HIC1.p2               HIC1.p2    429                  1413   \n",
        "128               ATF6.p2               ATF6.p2    124                  1413   \n",
        "\n",
        "             pval  motif_gene_count          motif_name_y  \n",
        "162  7.387844e-89              1951         ATF5_CREB3.p2  \n",
        "150  8.669441e-49              1125              CREB1.p2  \n",
        "17   2.969305e-47              8067              TFDP1.p2  \n",
        "47   2.212795e-43               959               ATF4.p2  \n",
        "64   7.775682e-38              1473                JUN.p2  \n",
        "125  2.573794e-36              8076         TFAP2{A,C}.p2  \n",
        "30   9.354414e-36             12044                SP1.p2  \n",
        "167  3.472166e-34              6217             TFAP2B.p2  \n",
        "104  3.516363e-29              3359               MAFB.p2  \n",
        "148  1.165571e-28              1690               PAX2.p2  \n",
        "94   9.217744e-27              4645  ELK1,4_GABP{A,B1}.p3  \n",
        "141  1.082015e-26             10334              PATZ1.p2  \n",
        "146  6.115925e-25              5810               PAX5.p2  \n",
        "93   3.723957e-24              3885            EGR1..3.p2  \n",
        "1    5.793301e-24             10899               KLF4.p3  \n",
        "68   9.469722e-24              5707           ELF1,2,4.p2  \n",
        "49   1.226055e-22              1983     AHR_ARNT_ARNT2.p2  \n",
        "57   4.027788e-22             10878                MAZ.p2  \n",
        "59   1.111639e-19              4518               HIC1.p2  \n",
        "128  1.189393e-17               862               ATF6.p2  "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"up\"]]\n",
        "               motif_name          motif_name_x  count  total_genes_in_group  \\\n",
        "17               TFDP1.p2              TFDP1.p2    739                  1307   \n",
        "157         ATF5_CREB3.p2         ATF5_CREB3.p2    288                  1307   \n",
        "29                 SP1.p2                SP1.p2    950                  1307   \n",
        "163             TFAP2B.p2             TFAP2B.p2    592                  1307   \n",
        "122         TFAP2{A,C}.p2         TFAP2{A,C}.p2    708                  1307   \n",
        "30                KLF4.p3               KLF4.p3    855                  1307   \n",
        "56                 MAZ.p2                MAZ.p2    847                  1307   \n",
        "138              PATZ1.p2              PATZ1.p2    806                  1307   \n",
        "142               PAX5.p2               PAX5.p2    520                  1307   \n",
        "63                 JUN.p2                JUN.p2    194                  1307   \n",
        "103               MAFB.p2               MAFB.p2    343                  1307   \n",
        "145              CREB1.p2              CREB1.p2    161                  1307   \n",
        "67            ELF1,2,4.p2           ELF1,2,4.p2    501                  1307   \n",
        "46                ATF4.p2               ATF4.p2    141                  1307   \n",
        "80              ZFP161.p2             ZFP161.p2    324                  1307   \n",
        "92             EGR1..3.p2            EGR1..3.p2    364                  1307   \n",
        "93   ELK1,4_GABP{A,B1}.p3  ELK1,4_GABP{A,B1}.p3    417                  1307   \n",
        "143               PAX2.p2               PAX2.p2    197                  1307   \n",
        "58                HIC1.p2               HIC1.p2    406                  1307   \n",
        "31               GTF2I.p2              GTF2I.p2    537                  1307   \n",
        "\n",
        "             pval  motif_gene_count          motif_name_y  \n",
        "17   2.766183e-51              8067              TFDP1.p2  \n",
        "157  7.511132e-51              1951         ATF5_CREB3.p2  \n",
        "29   4.580839e-43             12044                SP1.p2  \n",
        "163  8.404339e-42              6217             TFAP2B.p2  \n",
        "122  4.021575e-40              8076         TFAP2{A,C}.p2  \n",
        "30   6.460663e-33             10899               KLF4.p3  \n",
        "56   6.830916e-31             10878                MAZ.p2  \n",
        "138  2.947481e-28             10334              PATZ1.p2  \n",
        "142  1.110680e-27              5810               PAX5.p2  \n",
        "63   7.338497e-27              1473                JUN.p2  \n",
        "103  2.531844e-26              3359               MAFB.p2  \n",
        "145  3.946564e-26              1125              CREB1.p2  \n",
        "67   2.514609e-24              5707           ELF1,2,4.p2  \n",
        "46   4.756208e-24               959               ATF4.p2  \n",
        "80   2.340293e-21              3321             ZFP161.p2  \n",
        "92   3.405817e-21              3885            EGR1..3.p2  \n",
        "93   3.659889e-21              4645  ELK1,4_GABP{A,B1}.p3  \n",
        "143  8.196294e-21              1690               PAX2.p2  \n",
        "58   1.419015e-20              4518               HIC1.p2  \n",
        "31   6.106786e-15              6874              GTF2I.p2  \n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"any\"]]\n",
        "               motif_name          motif_name_x  count  total_genes_in_group  \\\n",
        "17               TFDP1.p2              TFDP1.p2   2247                  4511   \n",
        "130         TFAP2{A,C}.p2         TFAP2{A,C}.p2   2208                  4511   \n",
        "173             TFAP2B.p2             TFAP2B.p2   1789                  4511   \n",
        "29                 SP1.p2                SP1.p2   3008                  4511   \n",
        "151               PAX5.p2               PAX5.p2   1624                  4511   \n",
        "59                HIC1.p2               HIC1.p2   1311                  4511   \n",
        "146              PATZ1.p2              PATZ1.p2   2543                  4511   \n",
        "57                 MAZ.p2                MAZ.p2   2647                  4511   \n",
        "1                 KLF4.p3               KLF4.p3   2645                  4511   \n",
        "83              ZFP161.p2             ZFP161.p2    980                  4511   \n",
        "97             EGR1..3.p2            EGR1..3.p2   1103                  4511   \n",
        "109               MAFB.p2               MAFB.p2    965                  4511   \n",
        "167         ATF5_CREB3.p2         ATF5_CREB3.p2    612                  4511   \n",
        "98   ELK1,4_GABP{A,B1}.p3  ELK1,4_GABP{A,B1}.p3   1228                  4511   \n",
        "70            ELF1,2,4.p2           ELF1,2,4.p2   1458                  4511   \n",
        "156               NRF1.p2               NRF1.p2    841                  4511   \n",
        "49      AHR_ARNT_ARNT2.p2     AHR_ARNT_ARNT2.p2    588                  4511   \n",
        "175               MTF1.p2               MTF1.p2    864                  4511   \n",
        "69                MZF1.p2               MZF1.p2   2015                  4511   \n",
        "30               GTF2I.p2              GTF2I.p2   1651                  4511   \n",
        "\n",
        "             pval  motif_gene_count          motif_name_y  \n",
        "17   1.379576e-91              8067              TFDP1.p2  \n",
        "130  9.551311e-80              8076         TFAP2{A,C}.p2  \n",
        "173  1.455436e-77              6217             TFAP2B.p2  \n",
        "29   1.602122e-74             12044                SP1.p2  \n",
        "151  7.394485e-58              5810               PAX5.p2  \n",
        "59   9.611257e-54              4518               HIC1.p2  \n",
        "146  7.434821e-46             10334              PATZ1.p2  \n",
        "57   2.395488e-44             10878                MAZ.p2  \n",
        "1    4.482023e-43             10899               KLF4.p3  \n",
        "83   2.255376e-41              3321             ZFP161.p2  \n",
        "97   7.606363e-39              3885            EGR1..3.p2  \n",
        "109  1.797565e-35              3359               MAFB.p2  \n",
        "167  1.172732e-32              1951         ATF5_CREB3.p2  \n",
        "98   1.836830e-28              4645  ELK1,4_GABP{A,B1}.p3  \n",
        "70   2.647970e-27              5707           ELF1,2,4.p2  \n",
        "156  1.249092e-26              3000               NRF1.p2  \n",
        "49   2.184007e-24              1983     AHR_ARNT_ARNT2.p2  \n",
        "175  2.436411e-24              3146               MTF1.p2  \n",
        "69   2.761562e-20              8515               MZF1.p2  \n",
        "30   4.128660e-18              6874              GTF2I.p2  "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]], [\"ct\", \"cp73\"], [\"dir\", \"any\"]]\n",
        "        motif_name   motif_name_x  count  total_genes_in_group          pval  \\\n",
        "153  ATF5_CREB3.p2  ATF5_CREB3.p2     84                   498  6.573432e-09   \n",
        "119  TFAP2{A,C}.p2  TFAP2{A,C}.p2    240                   498  7.673493e-08   \n",
        "27          SP1.p2         SP1.p2    330                   498  8.129220e-08   \n",
        "60          JUN.p2         JUN.p2     65                   498  1.811081e-07   \n",
        "15        TFDP1.p2       TFDP1.p2    236                   498  4.746567e-07   \n",
        "53          MAZ.p2         MAZ.p2    297                   498  2.460972e-06   \n",
        "44         ATF4.p2        ATF4.p2     45                   498  3.412928e-06   \n",
        "160      TFAP2B.p2      TFAP2B.p2    182                   498  3.031184e-05   \n",
        "135       PATZ1.p2       PATZ1.p2    277                   498  5.595770e-05   \n",
        "55         HIC1.p2        HIC1.p2    138                   498  6.637967e-05   \n",
        "42        EP300.p2       EP300.p2     57                   498  9.964664e-05   \n",
        "1          KLF4.p3        KLF4.p3    287                   498  1.517365e-04   \n",
        "99         MAFB.p2        MAFB.p2    106                   498  1.841743e-04   \n",
        "140        PAX5.p2        PAX5.p2    163                   498  8.939717e-04   \n",
        "8     MYFfamily.p2   MYFfamily.p2     84                   498  1.097839e-03   \n",
        "142       CREB1.p2       CREB1.p2     41                   498  1.909658e-03   \n",
        "47          MYB.p2         MYB.p2     29                   498  2.123022e-03   \n",
        "63         MZF1.p2        MZF1.p2    221                   498  5.140350e-03   \n",
        "58         TP53.p2        TP53.p2      8                   498  5.175432e-03   \n",
        "9          NFE2.p2        NFE2.p2     19                   498  5.262990e-03   \n",
        "\n",
        "     motif_gene_count   motif_name_y  \n",
        "153              1951  ATF5_CREB3.p2  \n",
        "119              8076  TFAP2{A,C}.p2  \n",
        "27              12044         SP1.p2  \n",
        "60               1473         JUN.p2  \n",
        "15               8067       TFDP1.p2  \n",
        "53              10878         MAZ.p2  \n",
        "44                959        ATF4.p2  \n",
        "160              6217      TFAP2B.p2  \n",
        "135             10334       PATZ1.p2  \n",
        "55               4518        HIC1.p2  \n",
        "42               1508       EP300.p2  \n",
        "1               10899        KLF4.p3  \n",
        "99               3359        MAFB.p2  \n",
        "140              5810        PAX5.p2  \n",
        "8                2672   MYFfamily.p2  \n",
        "142              1125       CREB1.p2  \n",
        "47                720         MYB.p2  \n",
        "63               8515        MZF1.p2  \n",
        "58                117        TP53.p2  \n",
        "9                 436        NFE2.p2  \n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]], [\"ct\", \"cp73\"], [\"dir\", \"up\"]]\n",
        "                    motif_name               motif_name_x  count  \\\n",
        "56                      JUN.p2                     JUN.p2     54   \n",
        "147              ATF5_CREB3.p2              ATF5_CREB3.p2     64   \n",
        "25                      SP1.p2                     SP1.p2    210   \n",
        "113              TFAP2{A,C}.p2              TFAP2{A,C}.p2    151   \n",
        "41                     ATF4.p2                    ATF4.p2     33   \n",
        "14                    TFDP1.p2                   TFDP1.p2    147   \n",
        "50                      MAZ.p2                     MAZ.p2    185   \n",
        "130                   PATZ1.p2                   PATZ1.p2    176   \n",
        "39                    EP300.p2                   EP300.p2     40   \n",
        "1                      KLF4.p3                    KLF4.p3    179   \n",
        "52                     HIC1.p2                    HIC1.p2     87   \n",
        "136                   CREB1.p2                   CREB1.p2     30   \n",
        "95                     ATF2.p2                    ATF2.p2     20   \n",
        "26                    GTF2I.p2                   GTF2I.p2    120   \n",
        "152                  TFAP2B.p2                  TFAP2B.p2    110   \n",
        "9                      NFE2.p2                    NFE2.p2     15   \n",
        "134                    PAX5.p2                    PAX5.p2    102   \n",
        "21   FOS_FOS{B,L1}_JUN{B,D}.p2  FOS_FOS{B,L1}_JUN{B,D}.p2     20   \n",
        "58                     MZF1.p2                    MZF1.p2    140   \n",
        "55                    FOXN1.p2                   FOXN1.p2     11   \n",
        "\n",
        "     total_genes_in_group          pval  motif_gene_count  \\\n",
        "56                    300  2.320714e-11              1473   \n",
        "147                   300  2.761372e-11              1951   \n",
        "25                    300  3.753100e-08             12044   \n",
        "113                   300  8.567260e-07              8076   \n",
        "41                    300  1.094971e-06               959   \n",
        "14                    300  7.406257e-06              8067   \n",
        "50                    300  1.218175e-05             10878   \n",
        "130                   300  2.821462e-05             10334   \n",
        "39                    300  4.150171e-05              1508   \n",
        "1                     300  2.484429e-04             10899   \n",
        "52                    300  2.841070e-04              4518   \n",
        "136                   300  3.686455e-04              1125   \n",
        "95                    300  6.636135e-04               651   \n",
        "26                    300  7.597837e-04              6874   \n",
        "152                   300  9.011819e-04              6217   \n",
        "9                     300  1.024873e-03               436   \n",
        "134                   300  2.046832e-03              5810   \n",
        "21                    300  2.706194e-03               733   \n",
        "58                    300  2.826689e-03              8515   \n",
        "55                    300  3.614202e-03               309   \n",
        "\n",
        "                  motif_name_y  \n",
        "56                      JUN.p2  \n",
        "147              ATF5_CREB3.p2  \n",
        "25                      SP1.p2  \n",
        "113              TFAP2{A,C}.p2  \n",
        "41                     ATF4.p2  \n",
        "14                    TFDP1.p2  \n",
        "50                      MAZ.p2  \n",
        "130                   PATZ1.p2  \n",
        "39                    EP300.p2  \n",
        "1                      KLF4.p3  \n",
        "52                     HIC1.p2  \n",
        "136                   CREB1.p2  \n",
        "95                     ATF2.p2  \n",
        "26                    GTF2I.p2  \n",
        "152                  TFAP2B.p2  \n",
        "9                      NFE2.p2  \n",
        "134                    PAX5.p2  \n",
        "21   FOS_FOS{B,L1}_JUN{B,D}.p2  \n",
        "58                     MZF1.p2  \n",
        "55                    FOXN1.p2  "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, acuteHigh\"]], [\"ct\", \"cp73\"], [\"dir\", \"any\"]]\n",
        "               motif_name          motif_name_x  count  total_genes_in_group  \\\n",
        "16               TFDP1.p2              TFDP1.p2   2305                  4491   \n",
        "28                 SP1.p2                SP1.p2   3048                  4491   \n",
        "128         TFAP2{A,C}.p2         TFAP2{A,C}.p2   2206                  4491   \n",
        "171             TFAP2B.p2             TFAP2B.p2   1788                  4491   \n",
        "1                 KLF4.p3               KLF4.p3   2708                  4491   \n",
        "58                HIC1.p2               HIC1.p2   1313                  4491   \n",
        "144              PATZ1.p2              PATZ1.p2   2571                  4491   \n",
        "96   ELK1,4_GABP{A,B1}.p3  ELK1,4_GABP{A,B1}.p3   1330                  4491   \n",
        "149               PAX5.p2               PAX5.p2   1592                  4491   \n",
        "56                 MAZ.p2                MAZ.p2   2657                  4491   \n",
        "81              ZFP161.p2             ZFP161.p2    988                  4491   \n",
        "68            ELF1,2,4.p2           ELF1,2,4.p2   1529                  4491   \n",
        "154               NRF1.p2               NRF1.p2    903                  4491   \n",
        "48      AHR_ARNT_ARNT2.p2     AHR_ARNT_ARNT2.p2    622                  4491   \n",
        "12                 EHF.p2                EHF.p2    857                  4491   \n",
        "107               MAFB.p2               MAFB.p2    925                  4491   \n",
        "95             EGR1..3.p2            EGR1..3.p2   1045                  4491   \n",
        "165         ATF5_CREB3.p2         ATF5_CREB3.p2    583                  4491   \n",
        "157            NHLH1,2.p2            NHLH1,2.p2    796                  4491   \n",
        "67                MZF1.p2               MZF1.p2   2016                  4491   \n",
        "\n",
        "              pval  motif_gene_count          motif_name_y  \n",
        "16   1.504400e-112              8067              TFDP1.p2  \n",
        "28    2.018764e-89             12044                SP1.p2  \n",
        "128   1.705104e-81              8076         TFAP2{A,C}.p2  \n",
        "171   3.844446e-79              6217             TFAP2B.p2  \n",
        "1     3.027605e-59             10899               KLF4.p3  \n",
        "58    1.642059e-55              4518               HIC1.p2  \n",
        "144   4.062624e-54             10334              PATZ1.p2  \n",
        "96    3.061402e-52              4645  ELK1,4_GABP{A,B1}.p3  \n",
        "149   1.913971e-51              5810               PAX5.p2  \n",
        "56    1.162823e-48             10878                MAZ.p2  \n",
        "81    2.358256e-44              3321             ZFP161.p2  \n",
        "68    2.564819e-42              5707           ELF1,2,4.p2  \n",
        "154   2.684227e-42              3000               NRF1.p2  \n",
        "48    8.456296e-34              1983     AHR_ARNT_ARNT2.p2  \n",
        "12    1.446055e-29              3024                EHF.p2  \n",
        "107   2.257914e-27              3359               MAFB.p2  \n",
        "95    3.303631e-27              3885            EGR1..3.p2  \n",
        "165   1.430825e-25              1951         ATF5_CREB3.p2  \n",
        "157   1.934550e-23              2886            NHLH1,2.p2  \n",
        "67    1.467763e-21              8515               MZF1.p2  \n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"up\"]]\n",
        "               motif_name          motif_name_x  count  total_genes_in_group  \\\n",
        "163         ATF5_CREB3.p2         ATF5_CREB3.p2    387                  1871   \n",
        "17               TFDP1.p2              TFDP1.p2    998                  1871   \n",
        "28                 SP1.p2                SP1.p2   1333                  1871   \n",
        "126         TFAP2{A,C}.p2         TFAP2{A,C}.p2    996                  1871   \n",
        "168             TFAP2B.p2             TFAP2B.p2    799                  1871   \n",
        "1                 KLF4.p3               KLF4.p3   1192                  1871   \n",
        "148               PAX5.p2               PAX5.p2    732                  1871   \n",
        "55                 MAZ.p2                MAZ.p2   1176                  1871   \n",
        "143              PATZ1.p2              PATZ1.p2   1127                  1871   \n",
        "45                ATF4.p2               ATF4.p2    194                  1871   \n",
        "62                 JUN.p2                JUN.p2    259                  1871   \n",
        "57                HIC1.p2               HIC1.p2    585                  1871   \n",
        "151              CREB1.p2              CREB1.p2    214                  1871   \n",
        "68            ELF1,2,4.p2           ELF1,2,4.p2    689                  1871   \n",
        "105               MAFB.p2               MAFB.p2    456                  1871   \n",
        "94   ELK1,4_GABP{A,B1}.p3  ELK1,4_GABP{A,B1}.p3    584                  1871   \n",
        "93             EGR1..3.p2            EGR1..3.p2    496                  1871   \n",
        "81              ZFP161.p2             ZFP161.p2    434                  1871   \n",
        "149               PAX2.p2               PAX2.p2    259                  1871   \n",
        "67                MZF1.p2               MZF1.p2    901                  1871   \n",
        "\n",
        "             pval  motif_gene_count          motif_name_y  \n",
        "163  8.871256e-62              1951         ATF5_CREB3.p2  \n",
        "17   2.562365e-53              8067              TFDP1.p2  \n",
        "28   8.042536e-53             12044                SP1.p2  \n",
        "126  2.087644e-52              8076         TFAP2{A,C}.p2  \n",
        "168  1.172758e-44              6217             TFAP2B.p2  \n",
        "1    3.781178e-38             10899               KLF4.p3  \n",
        "148  2.163138e-36              5810               PAX5.p2  \n",
        "55   2.543508e-34             10878                MAZ.p2  \n",
        "143  1.801720e-33             10334              PATZ1.p2  \n",
        "45   3.553025e-31               959               ATF4.p2  \n",
        "62   3.905084e-31              1473                JUN.p2  \n",
        "57   1.515735e-30              4518               HIC1.p2  \n",
        "151  1.946884e-30              1125              CREB1.p2  \n",
        "68   1.206812e-27              5707           ELF1,2,4.p2  \n",
        "105  2.869131e-27              3359               MAFB.p2  \n",
        "94   5.773587e-27              4645  ELK1,4_GABP{A,B1}.p3  \n",
        "93   1.083340e-23              3885            EGR1..3.p2  \n",
        "81   2.301248e-22              3321             ZFP161.p2  \n",
        "149  6.037605e-22              1690               PAX2.p2  \n",
        "67   2.356899e-18              8515               MZF1.p2  "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"up\"]]\n",
        "            motif_name       motif_name_x  count  total_genes_in_group  \\\n",
        "48              JUN.p2             JUN.p2     18                   103   \n",
        "80             MAFB.p2            MAFB.p2     29                   103   \n",
        "81             ATF2.p2            ATF2.p2     10                   103   \n",
        "44             HIC1.p2            HIC1.p2     35                   103   \n",
        "127      ATF5_CREB3.p2      ATF5_CREB3.p2     19                   103   \n",
        "13            TFDP1.p2           TFDP1.p2     51                   103   \n",
        "53             REST.p3            REST.p3      3                   103   \n",
        "87       RXR{A,B,G}.p2      RXR{A,B,G}.p2     10                   103   \n",
        "140     bHLH_family.p2     bHLH_family.p2     20                   103   \n",
        "131          TFAP2B.p2          TFAP2B.p2     40                   103   \n",
        "97       TFAP2{A,C}.p2      TFAP2{A,C}.p2     49                   103   \n",
        "117           NFIL3.p2           NFIL3.p2      3                   103   \n",
        "8              NFE2.p2            NFE2.p2      6                   103   \n",
        "36   AHR_ARNT_ARNT2.p2  AHR_ARNT_ARNT2.p2     16                   103   \n",
        "136           HIF1A.p2           HIF1A.p2      8                   103   \n",
        "22            GTF2I.p2           GTF2I.p2     41                   103   \n",
        "101           MYOD1.p2           MYOD1.p2     11                   103   \n",
        "83         GTF2A1,2.p2        GTF2A1,2.p2      5                   103   \n",
        "45            FOSL2.p2           FOSL2.p2      5                   103   \n",
        "115            PAX5.p2            PAX5.p2     35                   103   \n",
        "\n",
        "         pval  motif_gene_count       motif_name_y  \n",
        "48   0.000155              1473             JUN.p2  \n",
        "80   0.000584              3359            MAFB.p2  \n",
        "81   0.000958               651            ATF2.p2  \n",
        "44   0.001005              4518            HIC1.p2  \n",
        "127  0.001651              1951      ATF5_CREB3.p2  \n",
        "13   0.005073              8067           TFDP1.p2  \n",
        "53   0.005880                78            REST.p3  \n",
        "87   0.007389               866      RXR{A,B,G}.p2  \n",
        "140  0.012807              2529     bHLH_family.p2  \n",
        "131  0.013009              6217          TFAP2B.p2  \n",
        "97   0.015234              8076      TFAP2{A,C}.p2  \n",
        "117  0.015348               111           NFIL3.p2  \n",
        "8    0.016722               436            NFE2.p2  \n",
        "36   0.021861              1983  AHR_ARNT_ARNT2.p2  \n",
        "136  0.038864               821           HIF1A.p2  \n",
        "22   0.040178              6874           GTF2I.p2  \n",
        "101  0.043113              1312           MYOD1.p2  \n",
        "83   0.049160               424        GTF2A1,2.p2  \n",
        "45   0.049568               425           FOSL2.p2  \n",
        "115  0.053660              5810            PAX5.p2  \n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"any\"]]\n",
        "            motif_name       motif_name_x  count  total_genes_in_group  \\\n",
        "25              SP1.p2             SP1.p2    296                   408   \n",
        "154          TFAP2B.p2          TFAP2B.p2    185                   408   \n",
        "14            TFDP1.p2           TFDP1.p2    216                   408   \n",
        "115      TFAP2{A,C}.p2      TFAP2{A,C}.p2    214                   408   \n",
        "1              KLF4.p3            KLF4.p3    266                   408   \n",
        "72           ZFP161.p2          ZFP161.p2    111                   408   \n",
        "51              MAZ.p2             MAZ.p2    264                   408   \n",
        "130           PATZ1.p2           PATZ1.p2    249                   408   \n",
        "52             HIC1.p2            HIC1.p2    132                   408   \n",
        "135            PAX5.p2            PAX5.p2    158                   408   \n",
        "26            GTF2I.p2           GTF2I.p2    176                   408   \n",
        "60             MZF1.p2            MZF1.p2    208                   408   \n",
        "149      ATF5_CREB3.p2      ATF5_CREB3.p2     68                   408   \n",
        "124            HES1.p2            HES1.p2     55                   408   \n",
        "97             MAFB.p2            MAFB.p2    100                   408   \n",
        "96           ZNF148.p2          ZNF148.p2     90                   408   \n",
        "85          EGR1..3.p2         EGR1..3.p2    106                   408   \n",
        "57              JUN.p2             JUN.p2     50                   408   \n",
        "43   AHR_ARNT_ARNT2.p2  AHR_ARNT_ARNT2.p2     62                   408   \n",
        "41             ATF4.p2            ATF4.p2     36                   408   \n",
        "\n",
        "             pval  motif_gene_count       motif_name_y  \n",
        "25   6.475633e-14             12044             SP1.p2  \n",
        "154  9.580949e-14              6217          TFAP2B.p2  \n",
        "14   1.056250e-11              8067           TFDP1.p2  \n",
        "115  4.602741e-11              8076      TFAP2{A,C}.p2  \n",
        "1    8.933986e-11             10899            KLF4.p3  \n",
        "72   1.519666e-10              3321          ZFP161.p2  \n",
        "51   2.598962e-10             10878             MAZ.p2  \n",
        "130  6.086178e-09             10334           PATZ1.p2  \n",
        "52   1.124119e-08              4518            HIC1.p2  \n",
        "135  2.812130e-08              5810            PAX5.p2  \n",
        "26   2.268056e-07              6874           GTF2I.p2  \n",
        "60   2.559966e-07              8515            MZF1.p2  \n",
        "149  2.828976e-07              1951      ATF5_CREB3.p2  \n",
        "124  4.101099e-07              1458            HES1.p2  \n",
        "97   6.044684e-07              3359            MAFB.p2  \n",
        "96   1.503784e-06              2982          ZNF148.p2  \n",
        "85   1.458209e-05              3885         EGR1..3.p2  \n",
        "57   2.609123e-05              1473             JUN.p2  \n",
        "43   3.180658e-05              1983  AHR_ARNT_ARNT2.p2  \n",
        "41   5.232512e-05               959            ATF4.p2  "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"down\"]]\n",
        "            motif_name       motif_name_x  count  total_genes_in_group  \\\n",
        "72       NFY{A,B,C}.p2      NFY{A,B,C}.p2     14                    51   \n",
        "18            GTF2I.p2           GTF2I.p2     25                    51   \n",
        "90          NHLH1,2.p2         NHLH1,2.p2     13                    51   \n",
        "13            RREB1.p2           RREB1.p2     18                    51   \n",
        "10            TFDP1.p2           TFDP1.p2     26                    51   \n",
        "51          GLI1..3.p2         GLI1..3.p2      4                    51   \n",
        "73           ZFP161.p2          ZFP161.p2     13                    51   \n",
        "26   AHR_ARNT_ARNT2.p2  AHR_ARNT_ARNT2.p2      9                    51   \n",
        "1              KLF4.p3            KLF4.p3     32                    51   \n",
        "75             HES1.p2            HES1.p2      7                    51   \n",
        "38             MZF1.p2            MZF1.p2     26                    51   \n",
        "100            MTF1.p2            MTF1.p2     12                    51   \n",
        "19            ESRRA.p2           ESRRA.p2      3                    51   \n",
        "57           ZNF148.p2          ZNF148.p2     11                    51   \n",
        "58             MAFB.p2            MAFB.p2     12                    51   \n",
        "6         MYFfamily.p2       MYFfamily.p2     10                    51   \n",
        "34             HIC1.p2            HIC1.p2     15                    51   \n",
        "80            PATZ1.p2           PATZ1.p2     29                    51   \n",
        "84             PAX5.p2            PAX5.p2     18                    51   \n",
        "98           TFAP2B.p2          TFAP2B.p2     19                    51   \n",
        "\n",
        "         pval  motif_gene_count       motif_name_y  \n",
        "72   0.004720              2875      NFY{A,B,C}.p2  \n",
        "18   0.005964              6874           GTF2I.p2  \n",
        "90   0.012595              2886         NHLH1,2.p2  \n",
        "13   0.013843              4649           RREB1.p2  \n",
        "10   0.025513              8067           TFDP1.p2  \n",
        "51   0.030283               510         GLI1..3.p2  \n",
        "73   0.036420              3321          ZFP161.p2  \n",
        "26   0.036921              1983  AHR_ARNT_ARNT2.p2  \n",
        "1    0.039800             10899            KLF4.p3  \n",
        "75   0.049929              1458            HES1.p2  \n",
        "38   0.050039              8515            MZF1.p2  \n",
        "100  0.052702              3146            MTF1.p2  \n",
        "19   0.060560               387           ESRRA.p2  \n",
        "57   0.076854              2982          ZNF148.p2  \n",
        "58   0.079238              3359            MAFB.p2  \n",
        "6    0.083914              2672       MYFfamily.p2  \n",
        "34   0.085017              4518            HIC1.p2  \n",
        "80   0.101001             10334           PATZ1.p2  \n",
        "84   0.102162              5810            PAX5.p2  \n",
        "98   0.103522              6217          TFAP2B.p2  \n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"any\"]]\n",
        "            motif_name       motif_name_x  count  total_genes_in_group  \\\n",
        "83             MAFB.p2            MAFB.p2     41                   154   \n",
        "46             HIC1.p2            HIC1.p2     50                   154   \n",
        "14            TFDP1.p2           TFDP1.p2     77                   154   \n",
        "24            GTF2I.p2           GTF2I.p2     66                   154   \n",
        "69             REST.p3            REST.p3      4                   154   \n",
        "37   AHR_ARNT_ARNT2.p2  AHR_ARNT_ARNT2.p2     25                   154   \n",
        "136          TFAP2B.p2          TFAP2B.p2     59                   154   \n",
        "101      TFAP2{A,C}.p2      TFAP2{A,C}.p2     72                   154   \n",
        "50              JUN.p2             JUN.p2     19                   154   \n",
        "9              NFE2.p2            NFE2.p2      8                   154   \n",
        "84             ATF2.p2            ATF2.p2     10                   154   \n",
        "119            PAX5.p2            PAX5.p2     53                   154   \n",
        "52             MZF1.p2            MZF1.p2     73                   154   \n",
        "132      ATF5_CREB3.p2      ATF5_CREB3.p2     22                   154   \n",
        "105           MYOD1.p2           MYOD1.p2     16                   154   \n",
        "61           ZFP161.p2          ZFP161.p2     33                   154   \n",
        "141           HIF1A.p2           HIF1A.p2     11                   154   \n",
        "91       RXR{A,B,G}.p2      RXR{A,B,G}.p2     11                   154   \n",
        "81            NFIL3.p2           NFIL3.p2      3                   154   \n",
        "125         NHLH1,2.p2         NHLH1,2.p2     28                   154   \n",
        "\n",
        "         pval  motif_gene_count       motif_name_y  \n",
        "83   0.000189              3359            MAFB.p2  \n",
        "46   0.000338              4518            HIC1.p2  \n",
        "14   0.000472              8067           TFDP1.p2  \n",
        "24   0.001525              6874           GTF2I.p2  \n",
        "69   0.002204                78            REST.p3  \n",
        "37   0.002800              1983  AHR_ARNT_ARNT2.p2  \n",
        "136  0.004336              6217          TFAP2B.p2  \n",
        "101  0.006559              8076      TFAP2{A,C}.p2  \n",
        "50   0.007261              1473             JUN.p2  \n",
        "9    0.011865               436            NFE2.p2  \n",
        "84   0.016491               651            ATF2.p2  \n",
        "119  0.016787              5810            PAX5.p2  \n",
        "52   0.016860              8515            MZF1.p2  \n",
        "132  0.017398              1951      ATF5_CREB3.p2  \n",
        "105  0.021656              1312           MYOD1.p2  \n",
        "61   0.021826              3321          ZFP161.p2  \n",
        "141  0.029790               821           HIF1A.p2  \n",
        "91   0.041298               866      RXR{A,B,G}.p2  \n",
        "81   0.043147               111           NFIL3.p2  \n",
        "125  0.044726              2886         NHLH1,2.p2  "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"down\"]]\n",
        "                  motif_name             motif_name_x  count  \\\n",
        "134                TFAP2B.p2                TFAP2B.p2     69   \n",
        "23                  GTF2I.p2                 GTF2I.p2     72   \n",
        "44                   HIC1.p2                  HIC1.p2     51   \n",
        "100            TFAP2{A,C}.p2            TFAP2{A,C}.p2     80   \n",
        "63                 ZFP161.p2                ZFP161.p2     39   \n",
        "129            ATF5_CREB3.p2            ATF5_CREB3.p2     26   \n",
        "82                    SRY.p2                   SRY.p2     16   \n",
        "66   HBP1_HMGB_SSRP1_UBTF.p2  HBP1_HMGB_SSRP1_UBTF.p2     23   \n",
        "119                 CREB1.p2                 CREB1.p2     17   \n",
        "13                  TFDP1.p2                 TFDP1.p2     77   \n",
        "52                   SOX5.p2                  SOX5.p2     11   \n",
        "70                   REST.p3                  REST.p3      3   \n",
        "35         AHR_ARNT_ARNT2.p2        AHR_ARNT_ARNT2.p2     23   \n",
        "48                    JUN.p2                   JUN.p2     18   \n",
        "50                   MZF1.p2                  MZF1.p2     76   \n",
        "84                   MAFB.p2                  MAFB.p2     34   \n",
        "106                  HES1.p2                  HES1.p2     17   \n",
        "112                 PATZ1.p2                 PATZ1.p2     89   \n",
        "68                    SRF.p3                   SRF.p3      3   \n",
        "43                    MAZ.p2                   MAZ.p2     93   \n",
        "\n",
        "     total_genes_in_group      pval  motif_gene_count             motif_name_y  \n",
        "134                   166  0.000152              6217                TFAP2B.p2  \n",
        "23                    166  0.000642              6874                 GTF2I.p2  \n",
        "44                    166  0.001226              4518                  HIC1.p2  \n",
        "100                   166  0.001550              8076            TFAP2{A,C}.p2  \n",
        "63                    166  0.002766              3321                ZFP161.p2  \n",
        "129                   166  0.003098              1951            ATF5_CREB3.p2  \n",
        "82                    166  0.003393               989                   SRY.p2  \n",
        "66                    166  0.004399              1696  HBP1_HMGB_SSRP1_UBTF.p2  \n",
        "119                   166  0.005047              1125                 CREB1.p2  \n",
        "13                    166  0.006265              8067                 TFDP1.p2  \n",
        "52                    166  0.009557               641                  SOX5.p2  \n",
        "70                    166  0.021247                78                  REST.p3  \n",
        "35                    166  0.025200              1983        AHR_ARNT_ARNT2.p2  \n",
        "48                    166  0.029494              1473                   JUN.p2  \n",
        "50                    166  0.036841              8515                  MZF1.p2  \n",
        "84                    166  0.042542              3359                  MAFB.p2  \n",
        "106                   166  0.049128              1458                  HES1.p2  \n",
        "112                   166  0.050316             10334                 PATZ1.p2  \n",
        "68                    166  0.050725               110                   SRF.p3  \n",
        "43                    166  0.052117             10878                   MAZ.p2  \n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, acuteHigh\"]], [\"ct\", \"cp73\"], [\"dir\", \"up\"]]\n",
        "               motif_name          motif_name_x  count  total_genes_in_group  \\\n",
        "17               TFDP1.p2              TFDP1.p2   1358                  2681   \n",
        "29                 SP1.p2                SP1.p2   1826                  2681   \n",
        "95   ELK1,4_GABP{A,B1}.p3  ELK1,4_GABP{A,B1}.p3    860                  2681   \n",
        "170             TFAP2B.p2             TFAP2B.p2   1071                  2681   \n",
        "127         TFAP2{A,C}.p2         TFAP2{A,C}.p2   1295                  2681   \n",
        "153               NRF1.p2               NRF1.p2    594                  2681   \n",
        "68            ELF1,2,4.p2           ELF1,2,4.p2    962                  2681   \n",
        "58                HIC1.p2               HIC1.p2    798                  2681   \n",
        "1                 KLF4.p3               KLF4.p3   1621                  2681   \n",
        "148               PAX5.p2               PAX5.p2    963                  2681   \n",
        "13                 EHF.p2                EHF.p2    561                  2681   \n",
        "81              ZFP161.p2             ZFP161.p2    593                  2681   \n",
        "143              PATZ1.p2              PATZ1.p2   1508                  2681   \n",
        "56                 MAZ.p2                MAZ.p2   1568                  2681   \n",
        "156            NHLH1,2.p2            NHLH1,2.p2    480                  2681   \n",
        "106               MAFB.p2               MAFB.p2    530                  2681   \n",
        "48      AHR_ARNT_ARNT2.p2     AHR_ARNT_ARNT2.p2    334                  2681   \n",
        "103                FEV.p2                FEV.p2    634                  2681   \n",
        "121               SPI1.p2               SPI1.p2    811                  2681   \n",
        "94             EGR1..3.p2            EGR1..3.p2    590                  2681   \n",
        "\n",
        "             pval  motif_gene_count          motif_name_y  \n",
        "17   3.115482e-56              8067              TFDP1.p2  \n",
        "29   3.189842e-51             12044                SP1.p2  \n",
        "95   5.877703e-46              4645  ELK1,4_GABP{A,B1}.p3  \n",
        "170  2.561836e-44              6217             TFAP2B.p2  \n",
        "127  2.097643e-39              8076         TFAP2{A,C}.p2  \n",
        "153  2.697524e-38              3000               NRF1.p2  \n",
        "68   3.032877e-34              5707           ELF1,2,4.p2  \n",
        "58   4.697150e-34              4518               HIC1.p2  \n",
        "1    6.362342e-34             10899               KLF4.p3  \n",
        "148  2.798827e-31              5810               PAX5.p2  \n",
        "13   5.100019e-28              3024                EHF.p2  \n",
        "81   2.654049e-25              3321             ZFP161.p2  \n",
        "143  6.514221e-25             10334              PATZ1.p2  \n",
        "56   8.573421e-24             10878                MAZ.p2  \n",
        "156  2.947329e-14              2886            NHLH1,2.p2  \n",
        "106  1.191038e-11              3359               MAFB.p2  \n",
        "48   1.163590e-10              1983     AHR_ARNT_ARNT2.p2  \n",
        "103  1.604910e-10              4192                FEV.p2  \n",
        "121  2.316868e-10              5555               SPI1.p2  \n",
        "94   4.628869e-10              3885            EGR1..3.p2  "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"down\"]]\n",
        "                           motif_name                       motif_name_x  \\\n",
        "0                            TEAD1.p2                           TEAD1.p2   \n",
        "87                      ATF5_CREB3.p2                      ATF5_CREB3.p2   \n",
        "23                  AHR_ARNT_ARNT2.p2                  AHR_ARNT_ARNT2.p2   \n",
        "69                            ATF6.p2                            ATF6.p2   \n",
        "35                            REST.p3                            REST.p3   \n",
        "57                            MAFB.p2                            MAFB.p2   \n",
        "45                             SRF.p3                             SRF.p3   \n",
        "31                          ZNF423.p2                          ZNF423.p2   \n",
        "65                  NFKB1_REL_RELA.p2                  NFKB1_REL_RELA.p2   \n",
        "79                            PAX2.p2                            PAX2.p2   \n",
        "43            HBP1_HMGB_SSRP1_UBTF.p2            HBP1_HMGB_SSRP1_UBTF.p2   \n",
        "40                          ZNF384.p2                          ZNF384.p2   \n",
        "90                             DBP.p2                             DBP.p2   \n",
        "80                           CREB1.p2                           CREB1.p2   \n",
        "16                        SREBF1,2.p2                        SREBF1,2.p2   \n",
        "34                            SOX5.p2                            SOX5.p2   \n",
        "42                      TCF4_dimer.p2                      TCF4_dimer.p2   \n",
        "7                        MYFfamily.p2                       MYFfamily.p2   \n",
        "4   ARNT_ARNT2_BHLHB2_MAX_MYC_USF1.p2  ARNT_ARNT2_BHLHB2_MAX_MYC_USF1.p2   \n",
        "15          FOS_FOS{B,L1}_JUN{B,D}.p2          FOS_FOS{B,L1}_JUN{B,D}.p2   \n",
        "\n",
        "    count  total_genes_in_group      pval  motif_gene_count  \\\n",
        "0       2                    26  0.037534               260   \n",
        "87      5                    26  0.074934              1951   \n",
        "23      5                    26  0.079209              1983   \n",
        "69      3                    26  0.079931               862   \n",
        "35      1                    26  0.088258                78   \n",
        "57      7                    26  0.089768              3359   \n",
        "45      1                    26  0.122256               110   \n",
        "31      1                    26  0.127457               115   \n",
        "65      2                    26  0.127828               527   \n",
        "79      4                    26  0.134809              1690   \n",
        "43      4                    26  0.136086              1696   \n",
        "40      2                    26  0.138620               554   \n",
        "90      2                    26  0.142670               564   \n",
        "80      3                    26  0.145403              1125   \n",
        "16      3                    26  0.162080              1184   \n",
        "34      2                    26  0.174660               641   \n",
        "42      8                    26  0.176189              4716   \n",
        "7       5                    26  0.201621              2672   \n",
        "4       3                    26  0.213797              1356   \n",
        "15      2                    26  0.214293               733   \n",
        "\n",
        "                         motif_name_y  \n",
        "0                            TEAD1.p2  \n",
        "87                      ATF5_CREB3.p2  \n",
        "23                  AHR_ARNT_ARNT2.p2  \n",
        "69                            ATF6.p2  \n",
        "35                            REST.p3  \n",
        "57                            MAFB.p2  \n",
        "45                             SRF.p3  \n",
        "31                          ZNF423.p2  \n",
        "65                  NFKB1_REL_RELA.p2  \n",
        "79                            PAX2.p2  \n",
        "43            HBP1_HMGB_SSRP1_UBTF.p2  \n",
        "40                          ZNF384.p2  \n",
        "90                             DBP.p2  \n",
        "80                           CREB1.p2  \n",
        "16                        SREBF1,2.p2  \n",
        "34                            SOX5.p2  \n",
        "42                      TCF4_dimer.p2  \n",
        "7                        MYFfamily.p2  \n",
        "4   ARNT_ARNT2_BHLHB2_MAX_MYC_USF1.p2  \n",
        "15          FOS_FOS{B,L1}_JUN{B,D}.p2  \n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"up\"]]\n",
        "            motif_name       motif_name_x  count  total_genes_in_group  \\\n",
        "100           PATZ1.p2           PATZ1.p2     40                    50   \n",
        "19              SP1.p2             SP1.p2     43                    50   \n",
        "11            TFDP1.p2           TFDP1.p2     34                    50   \n",
        "39              MAZ.p2             MAZ.p2     38                    50   \n",
        "118          TFAP2B.p2          TFAP2B.p2     27                    50   \n",
        "0              KLF4.p3            KLF4.p3     38                    50   \n",
        "88       TFAP2{A,C}.p2      TFAP2{A,C}.p2     31                    50   \n",
        "73             MAFB.p2            MAFB.p2     18                    50   \n",
        "37             SPZ1.p2            SPZ1.p2     11                    50   \n",
        "84          NR5A1,2.p2         NR5A1,2.p2     10                    50   \n",
        "46             MZF1.p2            MZF1.p2     30                    50   \n",
        "115     SOX{8,9,10}.p2     SOX{8,9,10}.p2      8                    50   \n",
        "106           CREB1.p2           CREB1.p2      8                    50   \n",
        "94       FOX{D1,D2}.p2      FOX{D1,D2}.p2      5                    50   \n",
        "89       NFY{A,B,C}.p2      NFY{A,B,C}.p2     14                    50   \n",
        "108            ZEB1.p2            ZEB1.p2     12                    50   \n",
        "31   AHR_ARNT_ARNT2.p2  AHR_ARNT_ARNT2.p2     11                    50   \n",
        "45              JUN.p2             JUN.p2      9                    50   \n",
        "111           SOX17.p2           SOX17.p2      9                    50   \n",
        "23             GFI1.p2            GFI1.p2     10                    50   \n",
        "\n",
        "         pval  motif_gene_count       motif_name_y  \n",
        "100  0.000002             10334           PATZ1.p2  \n",
        "19   0.000003             12044             SP1.p2  \n",
        "11   0.000007              8067           TFDP1.p2  \n",
        "39   0.000111             10878             MAZ.p2  \n",
        "118  0.000113              6217          TFAP2B.p2  \n",
        "0    0.000117             10899            KLF4.p3  \n",
        "88   0.000242              8076      TFAP2{A,C}.p2  \n",
        "73   0.000254              3359            MAFB.p2  \n",
        "37   0.001609              1750            SPZ1.p2  \n",
        "84   0.001696              1493         NR5A1,2.p2  \n",
        "46   0.001835              8515            MZF1.p2  \n",
        "115  0.002765              1077     SOX{8,9,10}.p2  \n",
        "106  0.003611              1125           CREB1.p2  \n",
        "94   0.003683               456      FOX{D1,D2}.p2  \n",
        "89   0.003883              2875      NFY{A,B,C}.p2  \n",
        "108  0.004177              2274            ZEB1.p2  \n",
        "31   0.004287              1983  AHR_ARNT_ARNT2.p2  \n",
        "45   0.005452              1473             JUN.p2  \n",
        "111  0.005696              1483           SOX17.p2  \n",
        "23   0.007364              1832            GFI1.p2  "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"down\"]]\n",
        "                    motif_name               motif_name_x  count  \\\n",
        "137                  TFAP2B.p2                  TFAP2B.p2     69   \n",
        "101              TFAP2{A,C}.p2              TFAP2{A,C}.p2     82   \n",
        "85                     MAFB.p2                    MAFB.p2     41   \n",
        "14                    TFDP1.p2                   TFDP1.p2     76   \n",
        "48                     HIC1.p2                    HIC1.p2     48   \n",
        "24                    GTF2I.p2                   GTF2I.p2     66   \n",
        "126                 NHLH1,2.p2                 NHLH1,2.p2     34   \n",
        "116                    LMO2.p2                    LMO2.p2     23   \n",
        "65               TCF4_dimer.p2              TCF4_dimer.p2     48   \n",
        "39           AHR_ARNT_ARNT2.p2          AHR_ARNT_ARNT2.p2     25   \n",
        "62                 SNAI1..3.p2                SNAI1..3.p2     41   \n",
        "148             bHLH_family.p2             bHLH_family.p2     28   \n",
        "54                     MZF1.p2                    MZF1.p2     73   \n",
        "64                   ZFP161.p2                  ZFP161.p2     34   \n",
        "46                      MAZ.p2                     MAZ.p2     89   \n",
        "109                    HES1.p2                    HES1.p2     18   \n",
        "19   FOS_FOS{B,L1}_JUN{B,D}.p2  FOS_FOS{B,L1}_JUN{B,D}.p2     11   \n",
        "135                  POU5F1.p2                  POU5F1.p2     12   \n",
        "23                      SP1.p2                     SP1.p2     95   \n",
        "139                    MTF1.p2                    MTF1.p2     31   \n",
        "\n",
        "     total_genes_in_group      pval  motif_gene_count  \\\n",
        "137                   150  0.000003              6217   \n",
        "101                   150  0.000005              8076   \n",
        "85                    150  0.000101              3359   \n",
        "14                    150  0.000307              8067   \n",
        "48                    150  0.000632              4518   \n",
        "24                    150  0.000664              6874   \n",
        "126                   150  0.000891              2886   \n",
        "116                   150  0.001211              1701   \n",
        "65                    150  0.001656              4716   \n",
        "39                    150  0.001940              1983   \n",
        "62                    150  0.002497              3921   \n",
        "148                   150  0.006520              2529   \n",
        "54                    150  0.008063              8515   \n",
        "64                    150  0.008795              3321   \n",
        "46                    150  0.009310             10878   \n",
        "109                   150  0.010510              1458   \n",
        "19                    150  0.011882               733   \n",
        "135                   150  0.016545               873   \n",
        "23                    150  0.020183             12044   \n",
        "139                   150  0.020727              3146   \n",
        "\n",
        "                  motif_name_y  \n",
        "137                  TFAP2B.p2  \n",
        "101              TFAP2{A,C}.p2  \n",
        "85                     MAFB.p2  \n",
        "14                    TFDP1.p2  \n",
        "48                     HIC1.p2  \n",
        "24                    GTF2I.p2  \n",
        "126                 NHLH1,2.p2  \n",
        "116                    LMO2.p2  \n",
        "65               TCF4_dimer.p2  \n",
        "39           AHR_ARNT_ARNT2.p2  \n",
        "62                 SNAI1..3.p2  \n",
        "148             bHLH_family.p2  \n",
        "54                     MZF1.p2  \n",
        "64                   ZFP161.p2  \n",
        "46                      MAZ.p2  \n",
        "109                    HES1.p2  \n",
        "19   FOS_FOS{B,L1}_JUN{B,D}.p2  \n",
        "135                  POU5F1.p2  \n",
        "23                      SP1.p2  \n",
        "139                    MTF1.p2  \n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]], [\"ct\", \"cp73\"], [\"dir\", \"down\"]]\n",
        "                  motif_name             motif_name_x  count  \\\n",
        "92                   MAFB.p2                  MAFB.p2     46   \n",
        "75   HBP1_HMGB_SSRP1_UBTF.p2  HBP1_HMGB_SSRP1_UBTF.p2     27   \n",
        "8               MYFfamily.p2             MYFfamily.p2     37   \n",
        "69                   ALX1.p2                  ALX1.p2      5   \n",
        "79                GLI1..3.p2               GLI1..3.p2     11   \n",
        "149                TFAP2B.p2                TFAP2B.p2     72   \n",
        "16                  TFDP1.p2                 TFDP1.p2     89   \n",
        "111            TFAP2{A,C}.p2            TFAP2{A,C}.p2     89   \n",
        "61               ELF1,2,4.p2              ELF1,2,4.p2     66   \n",
        "62                   SOX5.p2                  SOX5.p2     12   \n",
        "73             TCF4_dimer.p2            TCF4_dimer.p2     55   \n",
        "55                   TP53.p2                  TP53.p2      4   \n",
        "51                    MAZ.p2                   MAZ.p2    112   \n",
        "45                    MYB.p2                   MYB.p2     12   \n",
        "47                PRRX1,2.p2               PRRX1,2.p2     35   \n",
        "85                   ESR1.p2                  ESR1.p2      7   \n",
        "18        RXRA_VDR{dimer}.p2       RXRA_VDR{dimer}.p2      2   \n",
        "52                   HIC1.p2                  HIC1.p2     51   \n",
        "103                 NR1H4.p2                 NR1H4.p2      4   \n",
        "117            FOX{D1,D2}.p2            FOX{D1,D2}.p2      8   \n",
        "\n",
        "     total_genes_in_group      pval  motif_gene_count             motif_name_y  \n",
        "92                    198  0.001983              3359                  MAFB.p2  \n",
        "75                    198  0.002709              1696  HBP1_HMGB_SSRP1_UBTF.p2  \n",
        "8                     198  0.004924              2672             MYFfamily.p2  \n",
        "69                    198  0.004977               122                  ALX1.p2  \n",
        "79                    198  0.006683               510               GLI1..3.p2  \n",
        "149                   198  0.007842              6217                TFAP2B.p2  \n",
        "16                    198  0.009880              8067                 TFDP1.p2  \n",
        "111                   198  0.010206              8076            TFAP2{A,C}.p2  \n",
        "61                    198  0.012065              5707              ELF1,2,4.p2  \n",
        "62                    198  0.013574               641                  SOX5.p2  \n",
        "73                    198  0.020212              4716            TCF4_dimer.p2  \n",
        "55                    198  0.021469               117                  TP53.p2  \n",
        "51                    198  0.026001             10878                   MAZ.p2  \n",
        "45                    198  0.030181               720                   MYB.p2  \n",
        "47                    198  0.041091              2894               PRRX1,2.p2  \n",
        "85                    198  0.042716               356                  ESR1.p2  \n",
        "18                    198  0.043686                37       RXRA_VDR{dimer}.p2  \n",
        "52                    198  0.043786              4518                  HIC1.p2  \n",
        "103                   198  0.047807               151                 NR1H4.p2  \n",
        "117                   198  0.054795               456            FOX{D1,D2}.p2  "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"up\"]]\n",
        "        motif_name   motif_name_x  count  total_genes_in_group          pval  \\\n",
        "24          SP1.p2         SP1.p2    196                   243  1.449396e-17   \n",
        "1          KLF4.p3        KLF4.p3    176                   243  2.444616e-13   \n",
        "49          MAZ.p2         MAZ.p2    172                   243  9.335942e-12   \n",
        "14        TFDP1.p2       TFDP1.p2    140                   243  2.123488e-11   \n",
        "146      TFAP2B.p2      TFAP2B.p2    117                   243  3.291692e-11   \n",
        "129        PAX5.p2        PAX5.p2    108                   243  8.544552e-10   \n",
        "125       PATZ1.p2       PATZ1.p2    161                   243  8.803213e-10   \n",
        "111  TFAP2{A,C}.p2  TFAP2{A,C}.p2    135                   243  1.482120e-09   \n",
        "70       ZFP161.p2      ZFP161.p2     73                   243  2.216096e-09   \n",
        "112  NFY{A,B,C}.p2  NFY{A,B,C}.p2     62                   243  1.146821e-07   \n",
        "58         MZF1.p2        MZF1.p2    133                   243  2.643552e-07   \n",
        "93         MAFB.p2        MAFB.p2     67                   243  5.623453e-07   \n",
        "119        HES1.p2        HES1.p2     38                   243  6.856488e-07   \n",
        "92       ZNF148.p2      ZNF148.p2     61                   243  9.407918e-07   \n",
        "50         HIC1.p2        HIC1.p2     81                   243  1.961127e-06   \n",
        "82      EGR1..3.p2     EGR1..3.p2     72                   243  2.944742e-06   \n",
        "135     NHLH1,2.p2     NHLH1,2.p2     56                   243  1.481063e-05   \n",
        "133        ZEB1.p2        ZEB1.p2     47                   243  1.767771e-05   \n",
        "142  ATF5_CREB3.p2  ATF5_CREB3.p2     42                   243  2.091524e-05   \n",
        "148        MTF1.p2        MTF1.p2     59                   243  2.280708e-05   \n",
        "\n",
        "     motif_gene_count   motif_name_y  \n",
        "24              12044         SP1.p2  \n",
        "1               10899        KLF4.p3  \n",
        "49              10878         MAZ.p2  \n",
        "14               8067       TFDP1.p2  \n",
        "146              6217      TFAP2B.p2  \n",
        "129              5810        PAX5.p2  \n",
        "125             10334       PATZ1.p2  \n",
        "111              8076  TFAP2{A,C}.p2  \n",
        "70               3321      ZFP161.p2  \n",
        "112              2875  NFY{A,B,C}.p2  \n",
        "58               8515        MZF1.p2  \n",
        "93               3359        MAFB.p2  \n",
        "119              1458        HES1.p2  \n",
        "92               2982      ZNF148.p2  \n",
        "50               4518        HIC1.p2  \n",
        "82               3885     EGR1..3.p2  \n",
        "135              2886     NHLH1,2.p2  \n",
        "133              2274        ZEB1.p2  \n",
        "142              1951  ATF5_CREB3.p2  \n",
        "148              3146        MTF1.p2  \n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, acuteHigh\"]], [\"ct\", \"cp73\"], [\"dir\", \"down\"]]\n",
        "            motif_name       motif_name_x  count  total_genes_in_group  \\\n",
        "16            TFDP1.p2           TFDP1.p2    969                  1852   \n",
        "129      TFAP2{A,C}.p2      TFAP2{A,C}.p2    932                  1852   \n",
        "182             SP1.p2             SP1.p2   1249                  1852   \n",
        "170          TFAP2B.p2          TFAP2B.p2    739                  1852   \n",
        "162           PATZ1.p2           PATZ1.p2   1087                  1852   \n",
        "164      ATF5_CREB3.p2      ATF5_CREB3.p2    299                  1852   \n",
        "48   AHR_ARNT_ARNT2.p2  AHR_ARNT_ARNT2.p2    296                  1852   \n",
        "56              MAZ.p2             MAZ.p2   1113                  1852   \n",
        "29            GTF2I.p2           GTF2I.p2    764                  1852   \n",
        "1              KLF4.p3            KLF4.p3   1111                  1852   \n",
        "67             MZF1.p2            MZF1.p2    897                  1852   \n",
        "58             HIC1.p2            HIC1.p2    532                  1852   \n",
        "149            PAX5.p2            PAX5.p2    649                  1852   \n",
        "95          EGR1..3.p2         EGR1..3.p2    464                  1852   \n",
        "81           ZFP161.p2          ZFP161.p2    408                  1852   \n",
        "152           CREB1.p2           CREB1.p2    174                  1852   \n",
        "107            MAFB.p2            MAFB.p2    405                  1852   \n",
        "46             ATF4.p2            ATF4.p2    146                  1852   \n",
        "172            MTF1.p2            MTF1.p2    369                  1852   \n",
        "151            PAX2.p2            PAX2.p2    217                  1852   \n",
        "\n",
        "             pval  motif_gene_count       motif_name_y  \n",
        "16   8.572635e-47              8067           TFDP1.p2  \n",
        "129  5.724898e-36              8076      TFAP2{A,C}.p2  \n",
        "182  2.285206e-31             12044             SP1.p2  \n",
        "170  1.082588e-29              6217          TFAP2B.p2  \n",
        "162  2.968391e-26             10334           PATZ1.p2  \n",
        "164  3.556266e-26              1951      ATF5_CREB3.p2  \n",
        "48   5.006481e-24              1983  AHR_ARNT_ARNT2.p2  \n",
        "56   4.732741e-22             10878             MAZ.p2  \n",
        "29   1.311548e-21              6874           GTF2I.p2  \n",
        "1    2.756503e-21             10899            KLF4.p3  \n",
        "67   3.870983e-19              8515            MZF1.p2  \n",
        "58   1.055470e-18              4518            HIC1.p2  \n",
        "149  6.220501e-18              5810            PAX5.p2  \n",
        "95   3.941245e-17              3885         EGR1..3.p2  \n",
        "81   6.550405e-17              3321          ZFP161.p2  \n",
        "152  1.159899e-15              1125           CREB1.p2  \n",
        "107  1.983111e-15              3359            MAFB.p2  \n",
        "46   9.546676e-13               959            ATF4.p2  \n",
        "172  2.707394e-12              3146            MTF1.p2  \n",
        "151  9.095598e-11              1690            PAX2.p2  "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"down\"]]\n",
        "               motif_name          motif_name_x  count  total_genes_in_group  \\\n",
        "16               TFDP1.p2              TFDP1.p2   1262                  2668   \n",
        "172             TFAP2B.p2             TFAP2B.p2   1001                  2668   \n",
        "129         TFAP2{A,C}.p2         TFAP2{A,C}.p2   1227                  2668   \n",
        "28                 SP1.p2                SP1.p2   1694                  2668   \n",
        "58                HIC1.p2               HIC1.p2    735                  2668   \n",
        "150               PAX5.p2               PAX5.p2    903                  2668   \n",
        "82              ZFP161.p2             ZFP161.p2    551                  2668   \n",
        "96             EGR1..3.p2            EGR1..3.p2    613                  2668   \n",
        "145              PATZ1.p2              PATZ1.p2   1433                  2668   \n",
        "56                 MAZ.p2                MAZ.p2   1489                  2668   \n",
        "155               NRF1.p2               NRF1.p2    479                  2668   \n",
        "7               ZNF143.p2             ZNF143.p2    287                  2668   \n",
        "174               MTF1.p2               MTF1.p2    495                  2668   \n",
        "108               MAFB.p2               MAFB.p2    519                  2668   \n",
        "1                 KLF4.p3               KLF4.p3   1471                  2668   \n",
        "48      AHR_ARNT_ARNT2.p2     AHR_ARNT_ARNT2.p2    322                  2668   \n",
        "107             ZNF148.p2             ZNF148.p2    440                  2668   \n",
        "97   ELK1,4_GABP{A,B1}.p3  ELK1,4_GABP{A,B1}.p3    654                  2668   \n",
        "158            NHLH1,2.p2            NHLH1,2.p2    423                  2668   \n",
        "69            ELF1,2,4.p2           ELF1,2,4.p2    781                  2668   \n",
        "\n",
        "             pval  motif_gene_count          motif_name_y  \n",
        "16   2.401273e-33              8067              TFDP1.p2  \n",
        "172  1.178356e-28              6217             TFAP2B.p2  \n",
        "129  6.538212e-26              8076         TFAP2{A,C}.p2  \n",
        "28   1.094470e-22             12044                SP1.p2  \n",
        "58   1.153060e-20              4518               HIC1.p2  \n",
        "150  6.079912e-20              5810               PAX5.p2  \n",
        "82   1.084339e-16              3321             ZFP161.p2  \n",
        "96   5.296523e-14              3885            EGR1..3.p2  \n",
        "145  6.267173e-14             10334              PATZ1.p2  \n",
        "56   1.292221e-12             10878                MAZ.p2  \n",
        "155  1.222110e-11              3000               NRF1.p2  \n",
        "7    3.090739e-11              1642             ZNF143.p2  \n",
        "174  5.164735e-11              3146               MTF1.p2  \n",
        "108  2.742261e-10              3359               MAFB.p2  \n",
        "1    3.919890e-10             10899               KLF4.p3  \n",
        "48   9.134365e-09              1983     AHR_ARNT_ARNT2.p2  \n",
        "107  2.198902e-06              2982             ZNF148.p2  \n",
        "97   3.455691e-06              4645  ELK1,4_GABP{A,B1}.p3  \n",
        "158  7.168112e-06              2886            NHLH1,2.p2  \n",
        "69   1.881445e-05              5707           ELF1,2,4.p2  \n"
       ]
      }
     ],
     "prompt_number": 539
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for k in top_tfs.keys():\n",
      "    if top_tfs[k] is not None:\n",
      "        top_tfs[k][\"pval\"] = top_tfs[k].merge(tf_target_counts, left_on=\"tf\", right_index=True).apply( lambda x: (\n",
      "         scipy.stats.hypergeom.sf(\n",
      "              x.ix[\"c[ount\"]-1, # number of differentially expressed genes in set\n",
      "              total_genes,           # total number of genes\n",
      "              x.ix[\"chea_target_count\"],   # number of genes in current set\n",
      "              x.ix[\"total_genes_in_group\"])),     # total number of genes in test set\n",
      "            axis=1 )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 468
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for k in top_motifs.keys():\n",
      "    if top_motifs[k] is not None:\n",
      "        top_motifs[k][\"pval\"] = top_motifs[k].merge(mm9_motif_gene_counts, left_on=\"motif_name\", right_index=True).apply( lambda x: (\n",
      "         scipy.stats.hypergeom.sf(\n",
      "              x.ix[\"count\"]-1, # number of differentially expressed genes in set\n",
      "              total_genes,           # total number of genes\n",
      "              x.ix[\"motif_gene_count\"],   # number of genes in current set\n",
      "              x.ix[\"total_genes_in_group\"])),     # total number of genes in test set\n",
      "            axis=1 )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 528
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "\"_\".join( [str(z[1]) for z in pda.tm.d(k)] )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 481,
       "text": [
        "\"[u'6-OHDA, chronicSaline', u'Ascorbate, chronicSaline']_cp73_up\""
       ]
      }
     ],
     "prompt_number": 481
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "e = pd.ExcelWriter(\"/data/adrian/Dropbox/pd_tf_enrichment_20130820.xls\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 469
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "e = open(\"/data/adrian/Dropbox/pd_tf_enrichment_20130821.html\",\"w\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 555
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for k in top_tfs.keys():\n",
      "    if top_tfs[k] is not None:\n",
      "        print k\n",
      "        t = top_tfs[k].sort_index(by=\"pval\", ascending=True)[0:30].merge(tf_target_counts, left_on=\"tf\", right_index=True).drop([\"tf_x\", \"tf_y\"], axis=1)\n",
      "        e.write(\"<hr>\")\n",
      "        e.write(\"<h2>\" + k + \"</h2>\")\n",
      "        \n",
      "        m = top_motifs[k].merge(mm9_motif_gene_counts, left_on=\"motif_name\", right_index=True).sort_index(by=\"pval\", ascending=True)[0:30].drop([\"motif_name_x\", \"motif_name_y\"], axis=1)\n",
      "        e.write(\"<table>\")\n",
      "        e.write(\"<tr><th> top 30 overrepresented associations from CHEA </th> <th> top 30 over-represented tf binding motifs (SwissRegulon) </th></tr>\")\n",
      "        e.write(\"<tr><td>\" + t.to_html() + \"</td>\" + \"<td style='padding-left: 25px'>\" + m.to_html() + \"</td></tr></table>\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"up\"]]\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"any\"]]\n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"any\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"down\"]]\n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"down\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"any\"]]\n",
        "[[\"cmp\", [\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"any\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"up\"]]\n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"up\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"any\"]]\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]], [\"ct\", \"cp73\"], [\"dir\", \"any\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]], [\"ct\", \"cp73\"], [\"dir\", \"up\"]]\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, acuteHigh\"]], [\"ct\", \"cp73\"], [\"dir\", \"any\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"up\"]]\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"up\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"any\"]]\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"down\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"any\"]]\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"down\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, acuteHigh\"]], [\"ct\", \"cp73\"], [\"dir\", \"up\"]]\n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"down\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"up\"]]\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"down\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]], [\"ct\", \"cp73\"], [\"dir\", \"down\"]]\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"dir\", \"up\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, acuteHigh\"]], [\"ct\", \"cp73\"], [\"dir\", \"down\"]]\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"dir\", \"down\"]]"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n"
       ]
      }
     ],
     "prompt_number": 556
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "e.close()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 557
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "patterns = cp_both_wide_info[ [c for c in pda.tm.m([(\"cmp\"), \n",
      "          (\"st\", \"pval\"), \n",
      "          (\"mc\", \"bh\"), \n",
      "          (\"tt\", \"welch ttest\")], cp_both_wide_info.columns)] ]  < 0.10"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "not an encoded tag list\n",
        "not an encoded tag list\n",
        "not an encoded tag list\n"
       ]
      }
     ],
     "prompt_number": 29
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "[c for c in pda.tm.m([(\"cmp\"), \n",
      "          (\"st\", \"pval\"), \n",
      "          (\"mc\", \"bh\"), \n",
      "          (\"tt\", \"welch ttest\")], cp_both_wide_info.columns)]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "not an encoded tag list\n",
        "not an encoded tag list\n",
        "not an encoded tag list\n"
       ]
      },
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 329,
       "text": [
        "['[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, acuteHigh\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]']"
       ]
      }
     ],
     "prompt_number": 329
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "table_fc_threshold = log2(1.5)\n",
      "cn_cp73_bh_ttest_pval = pda.tm.e([(\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]), \n",
      "        (\"ct\", \"cp73\"), (\"mc\", \"bh\"), (\"st\", \"pval\"), (\"tt\", \"welch ttest\")])\n",
      "cn_cp73_median_foldchange = pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]), \n",
      "        (\"ct\", \"cp73\"), (\"st\", \"fc_medians\")])\n",
      "\n",
      "cp73_dopamine_depletion_up = (cp_both_wide_info.select(\n",
      "    lambda x: cp_both_wide_info.ix[x, cn_cp73_bh_ttest_pval] < 0.10\n",
      "    and cp_both_wide_info.ix[x, cn_cp73_median_foldchange] >= table_fc_threshold\n",
      "))[[\"probe_id\", \"symbol\", \"gene_name\", \n",
      "    cn_cp73_bh_ttest_pval,\n",
      "    cn_cp73_median_foldchange\n",
      "    ]].sort_index(by=cn_cp73_median_foldchange, ascending=False)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 117
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cn_cp73_chronicHigh_bh_ttest_pval = pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"mc\", \"bh\"),\n",
      "            (\"st\", \"pval\"),\n",
      "            (\"tt\", \"welch ttest\")\n",
      "        ])\n",
      "\n",
      "cn_cp73_chronicHigh_foldchange = pda.tm.e([  \n",
      "            (\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"st\", \"fc_medians\")\n",
      "        ])\n",
      "\n",
      "cp73_chronicHigh_up = (cp_both_wide_info.select(\n",
      "    lambda x: (cp_both_wide_info.ix[x, cn_cp73_chronicHigh_bh_ttest_pval] < 0.10)\n",
      "    and cp_both_wide_info.ix[x, cn_cp73_chronicHigh_foldchange] >= table_fc_threshold\n",
      "))\n",
      "\n",
      "cp73_chronicHigh_down = (cp_both_wide_info.select(\n",
      "    lambda x: (cp_both_wide_info.ix[x, cn_cp73_chronicHigh_bh_ttest_pval] < 0.10)\n",
      "    and cp_both_wide_info.ix[x, cn_cp73_chronicHigh_foldchange] <= -table_fc_threshold\n",
      "))[[\"symbol\", cn_cp73_chronicHigh_bh_ttest_pval, cn_cp73_chronicHigh_foldchange]]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 577
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp73_chronicHigh_down"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<pre>\n",
        "&ltclass 'pandas.core.frame.DataFrame'&gt\n",
        "Index: 4038 entries, 1415682_at to 1460728_s_at\n",
        "Data columns (total 3 columns):\n",
        "symbol                                                                                                                              3799  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]    4038  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"fc_medians\"]]                                   4038  non-null values\n",
        "dtypes: float64(2), object(1)\n",
        "</pre>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 578,
       "text": [
        "<class 'pandas.core.frame.DataFrame'>\n",
        "Index: 4038 entries, 1415682_at to 1460728_s_at\n",
        "Data columns (total 3 columns):\n",
        "symbol                                                                                                                              3799  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]    4038  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"fc_medians\"]]                                   4038  non-null values\n",
        "dtypes: float64(2), object(1)"
       ]
      }
     ],
     "prompt_number": 578
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "[c for c in pda.tm.m([(\"cmp\"), \n",
      "          (\"st\", \"pval\"), \n",
      "          (\"mc\", \"bh\"), \n",
      "          (\"tt\", \"welch ttest\")], cp_both_wide_info.columns)]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "not an encoded tag list\n",
        "not an encoded tag list\n",
        "not an encoded tag list\n"
       ]
      },
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 324,
       "text": [
        "['[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, acuteHigh\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]',\n",
        " '[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]']"
       ]
      }
     ],
     "prompt_number": 324
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "patterns"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<pre>\n",
        "&ltclass 'pandas.core.frame.DataFrame'&gt\n",
        "Index: 45101 entries, 1415670_at to AFFX-TrpnX-M_at\n",
        "Data columns (total 10 columns):\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, chronicHigh\"]             45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]             45101  non-null values\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "dtypes: bool(10)\n",
        "</pre>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 321,
       "text": [
        "<class 'pandas.core.frame.DataFrame'>\n",
        "Index: 45101 entries, 1415670_at to AFFX-TrpnX-M_at\n",
        "Data columns (total 10 columns):\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, chronicHigh\"]             45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]             45101  non-null values\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "dtypes: bool(10)"
       ]
      },
      {
       "html": [
        "<pre>\n",
        "&ltclass 'pandas.core.frame.DataFrame'&gt\n",
        "Index: 45101 entries, 1415670_at to AFFX-TrpnX-M_at\n",
        "Data columns (total 10 columns):\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, chronicHigh\"]             45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]             45101  non-null values\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "dtypes: bool(10)\n",
        "</pre>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 322,
       "text": [
        "<class 'pandas.core.frame.DataFrame'>\n",
        "Index: 45101 entries, 1415670_at to AFFX-TrpnX-M_at\n",
        "Data columns (total 10 columns):\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, chronicHigh\"]             45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]             45101  non-null values\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "dtypes: bool(10)"
       ]
      }
     ],
     "prompt_number": 322
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "patterns.columns = [[pda.tm.e(val) for (t, val) in pda.tm.d( c ) if t == \"cmp\"][0] for c in patterns.columns]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 41
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "patterns"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<pre>\n",
        "&ltclass 'pandas.core.frame.DataFrame'&gt\n",
        "Index: 45101 entries, 1415670_at to AFFX-TrpnX-M_at\n",
        "Data columns (total 10 columns):\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, chronicHigh\"]             45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]             45101  non-null values\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "dtypes: bool(10)\n",
        "</pre>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 49,
       "text": [
        "<class 'pandas.core.frame.DataFrame'>\n",
        "Index: 45101 entries, 1415670_at to AFFX-TrpnX-M_at\n",
        "Data columns (total 10 columns):\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, chronicHigh\"]             45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]             45101  non-null values\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "dtypes: bool(10)"
       ]
      }
     ],
     "prompt_number": 49
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "patterns.dtypes"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<pre>\n",
        "&ltclass 'pandas.core.frame.DataFrame'&gt\n",
        "Index: 45101 entries, 1415670_at to AFFX-TrpnX-M_at\n",
        "Data columns (total 10 columns):\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, chronicHigh\"]             45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]             45101  non-null values\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "dtypes: bool(10)\n",
        "</pre>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 66,
       "text": [
        "<class 'pandas.core.frame.DataFrame'>\n",
        "Index: 45101 entries, 1415670_at to AFFX-TrpnX-M_at\n",
        "Data columns (total 10 columns):\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, chronicHigh\"]             45101  non-null values\n",
        "[\"6-OHDA, acuteHigh\", \"6-OHDA, acuteSaline\"]             45101  non-null values\n",
        "[\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]    45101  non-null values\n",
        "[\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]          45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]         45101  non-null values\n",
        "[\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]            45101  non-null values\n",
        "dtypes: bool(10)"
       ]
      }
     ],
     "prompt_number": 66
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "r = "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 74
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def patternGroup(x):\n",
      "    r = patterns.ix[ x, :]\n",
      "    return int(\"\".join([str(int(a)) for a in r]), 2)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 89
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "g = patterns.groupby(patternGroup)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 327
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len(g.groups.items())"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 328,
       "text": [
        "231"
       ]
      }
     ],
     "prompt_number": 328
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for (k, v) in g.groups.items():\n",
      "    print k, len(v)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "0 19529\n",
        "512 78\n",
        "2 88\n",
        "4 7\n",
        "6 1\n",
        "8 22\n",
        "10 5\n",
        "514 5\n",
        "16 1639\n",
        "18 30\n",
        "20 1\n",
        "22 2\n",
        "24 6\n",
        "768 11\n",
        "32 3406\n",
        "688 25\n",
        "34 34\n",
        "36 1\n",
        "38 1\n",
        "40 3\n",
        "42 1\n",
        "690 5\n",
        "48 2151\n",
        "520 1\n",
        "50 58\n",
        "52 1\n",
        "54 6\n",
        "56 10\n",
        "58 4\n",
        "522 2\n",
        "62 1\n",
        "64 87\n",
        "608 2\n",
        "578 1\n",
        "1008 37\n",
        "70 1\n",
        "72 1\n",
        "866 1\n",
        "80 19\n",
        "440 24\n",
        "88 1\n",
        "90 1\n",
        "954 2\n",
        "96 25\n",
        "528 15\n",
        "98 1\n",
        "992 19\n",
        "444 1\n",
        "618 1\n",
        "786 1\n",
        "624 1\n",
        "986 1\n",
        "114 1\n",
        "754 2\n",
        "720 5\n",
        "128 2721\n",
        "1010 8\n",
        "642 3\n",
        "132 1\n",
        "136 9\n",
        "138 2\n",
        "962 1\n",
        "656 13\n",
        "146 9\n",
        "150 2\n",
        "152 4\n",
        "666 1\n",
        "704 2\n",
        "282 1\n",
        "160 1052\n",
        "880 2\n",
        "162 20\n",
        "960 1\n",
        "166 2\n",
        "168 9\n",
        "170 2\n",
        "172 1\n",
        "176 710\n",
        "178 20\n",
        "182 4\n",
        "184 2\n",
        "192 163\n",
        "544 37\n",
        "706 1\n",
        "944 99\n",
        "552 2\n",
        "200 1\n",
        "546 3\n",
        "208 127\n",
        "210 7\n",
        "212 1\n",
        "214 1\n",
        "216 2\n",
        "672 26\n",
        "734 1\n",
        "224 147\n",
        "464 319\n",
        "738 1\n",
        "230 1\n",
        "232 4\n",
        "234 1\n",
        "236 1\n",
        "978 3\n",
        "752 8\n",
        "758 2\n",
        "400 1800\n",
        "246 1\n",
        "760 3\n",
        "250 1\n",
        "530 1\n",
        "298 1\n",
        "946 12\n",
        "256 1099\n",
        "896 73\n",
        "258 9\n",
        "260 1\n",
        "982 2\n",
        "264 6\n",
        "898 7\n",
        "272 148\n",
        "984 1\n",
        "274 6\n",
        "278 1\n",
        "674 1\n",
        "640 21\n",
        "248 2\n",
        "288 262\n",
        "560 49\n",
        "290 9\n",
        "292 1\n",
        "294 1\n",
        "112 9\n",
        "810 1\n",
        "562 7\n",
        "304 243\n",
        "904 3\n",
        "306 5\n",
        "310 2\n",
        "314 1\n",
        "828 1\n",
        "914 3\n",
        "320 37\n",
        "736 7\n",
        "834 1\n",
        "130 15\n",
        "226 5\n",
        "336 8\n",
        "568 1\n",
        "850 1\n",
        "352 20\n",
        "144 844\n",
        "354 1\n",
        "870 1\n",
        "360 1\n",
        "658 3\n",
        "368 20\n",
        "488 17\n",
        "370 2\n",
        "374 1\n",
        "950 1\n",
        "490 2\n",
        "384 2129\n",
        "832 1\n",
        "386 54\n",
        "388 3\n",
        "406 5\n",
        "390 1\n",
        "392 9\n",
        "394 3\n",
        "912 60\n",
        "920 1\n",
        "402 46\n",
        "404 6\n",
        "918 2\n",
        "408 11\n",
        "410 7\n",
        "922 1\n",
        "928 113\n",
        "240 124\n",
        "930 11\n",
        "420 2\n",
        "422 8\n",
        "424 27\n",
        "426 6\n",
        "242 8\n",
        "432 1775\n",
        "434 82\n",
        "436 2\n",
        "438 5\n",
        "952 10\n",
        "500 2\n",
        "442 4\n",
        "956 1\n",
        "958 2\n",
        "816 7\n",
        "448 37\n",
        "416 1876\n",
        "450 3\n",
        "1014 1\n",
        "454 1\n",
        "456 1\n",
        "418 85\n",
        "976 22\n",
        "1016 12\n",
        "466 21\n",
        "468 1\n",
        "470 5\n",
        "472 4\n",
        "474 6\n",
        "1018 1\n",
        "990 1\n",
        "800 8\n",
        "480 204\n",
        "482 8\n",
        "484 1\n",
        "486 1\n",
        "1000 5\n",
        "1002 4\n",
        "496 437\n",
        "936 8\n",
        "498 47\n",
        "1012 2\n",
        "1022 2\n",
        "502 13\n",
        "784 3\n",
        "504 17\n",
        "576 1\n",
        "506 4\n",
        "508 1\n",
        "938 2\n",
        "510 3\n"
       ]
      }
     ],
     "prompt_number": 97
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "figsize(10,10)\n",
      "imshow( g.get_group(656).as_matrix(), interpolation=\"nearest\", cmap=get_cmap(\"gray\"))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 110,
       "text": [
        "<matplotlib.image.AxesImage at 0x95d3790>"
       ]
      },
      {
       "metadata": {},
       "output_type": "display_data",
       "png": "iVBORw0KGgoAAAANSUhEUgAAAcoAAAJICAYAAADhDKSSAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAFMpJREFUeJzt3V1onXcdwPHfqYkIlra2rClLOip7MS/rS2pL6UA503Ub\ng9bSRlm7UenLdiHCNoY3IjQONB0VZuf0ZuDcVNaJF26WkosoC5sljpIxy5x0zAbT1g6GtFIzlzUc\nL8R0ZeZne3rOznni5wOFJM3558e/z8O3z3NOklKlUqkEAPBfzWn0AADQzIQSABJCCQAJoQSAhFAC\nQEIoASDRUq+FS6VSvZYGgJqb6bsl63pFWalUav5n7969NV+zSIqyp0Xb1yKpx79Vvf7AbODWKwAk\nhBIAEoULZblcbvQIs449BZhZqVKnJxJKpVJhnqMo0guPirKnEcXa1yJxDEB9zHRuFe6KEgA+SkIJ\nAAmhBICEUAJAQigBICGUAJAQSgBICCUAJIQSABJCCQAJoQSAhFACQKLqUA4ODkZnZ2fceOON8eij\nj9ZyJgBoGlX99pCpqan4zGc+E0NDQ9He3h5r166NZ599Nrq6ui4u7LeH1EVR9jSiWPtaJI4BqI+a\n/vaQV155JW644YZYtmxZtLa2xt133x3PP//8VQ0IAM2opZoHnTp1KpYuXTr9fkdHR/z+97//0Of1\n9/dPv10ul/2CYAAKp6pQXu7tlA+GEgCKqKpbr+3t7TE+Pj79/vj4eHR0dNRsKABoFlWFcs2aNfHm\nm2/G2NhYTE5OxnPPPRebNm2q9WwA0HBV3XptaWmJJ554Iu64446YmpqK3bt3X/KKVwCYLar69pDL\nWti3h9RFUfY0olj7WiSOAaiPmn57CAD8vxBKAEgIJQAkhBIAEkIJAAmhBICEUAJAQigBICGUAJAQ\nSgBICCUAJIQSABJCCQAJoQSAhFACQEIoASAhlACQEEoASAglACSEEgASQgkACaEEgIRQAkBCKAEg\nIZQAkBBKAEgIJQAkhBIAEkIJAAmhBICEUAJAQigBICGUAJAQSgBICCUAJIQSABJCCQAJoQSAhFAC\nQEIoASAhlACQEEoASAglACSEEgASQgkACaEEgIRQAkBCKAEgIZQAkBBKAEgIJQAkhBIAEkIJAAmh\nBICEUAJAQigBICGUAJAQSgBICCUAJIQSABJCCQAJoQSAhFACQEIoASAhlACQEEoASAglACSEEgAS\nQgkACaEEgIRQAkBCKAEgIZQAkBBKAEgIJQAkhBIAEkIJAImqQjk+Ph633npr9PT0xM033xyPP/54\nrecCgKZQqlQqlSt90JkzZ+LMmTOxatWqOH/+fHz2s5+NX/3qV9HV1XVx4VIpqli6IUqlUqNHuGxF\n2dOIYu1rkTgGoD5mOrequqJcsmRJrFq1KiIi5s6dG11dXXH69OnqpwOAJtVytQuMjY3Fq6++GuvW\nrfvQ3/X390+/XS6Xo1wuX+2XA4CPVFW3Xv/j/PnzUS6X41vf+lZs3rz50oXdeq2LouxpRLH2tUgc\nA1AfNb31GhHx/vvvx9atW+Pee+/9UCQBYLao6oqyUqnEV7/61Vi0aFE89thj/31hV5R1UZQ9jSjW\nvhaJYwDqY6Zzq6pQvvzyy/H5z38+VqxYMX0iDAwMxJ133nlxYaGsi6LsaUSx9rVIHANQHzUN5eUQ\nyvooyp5GFGtfi8QxAPVR8+coAeD/gVACQEIoASAhlACQEEoASAglACSEEgASQgkACaEEgIRQAkBC\nKAEgIZQAkBBKAEgIJQAkhBIAEkIJAAmhBICEUAJAQigBICGUAJAQSgBICCUAJIQSABJCCQAJoQSA\nhFACQEIoASAhlACQEEoASAglACSEEgASQgkACaEEgIRQAkBCKAEgIZQAkBBKAEgIJQAkhBIAEkIJ\nAAmhBICEUAJAQigBICGUAJAQSgBICCUAJIQSABJCCQAJoQSAhFACQEIoASAhlACQEEoASAglACSE\nEgASQgkACaEEgIRQAkBCKAEgIZQAkBBKAEgIJQAkhBIAEkIJAAmhBICEUAJAQigBICGUAJAQSgBI\nCCUAJIQSABJCCQAJoQSAhFACQEIoASAhlACQuKpQTk1NRW9vb2zcuLFW8wBAU7mqUB44cCC6u7uj\nVCrVah4AaCpVh/LkyZNx+PDh2LNnT1QqlVrOBABNo+pQPvTQQ7F///6YM8fTnADMXi3VPOjQoUOx\nePHi6O3tjRdffHHGz+vv759+u1wuR7lcrubLAUDDlCpV3Df95je/GT/96U+jpaUl/vnPf8bf//73\n2Lp1azzzzDMXFy6VCnNLtkjPsRZlTyOKta9F4hiA+pjp3KoqlB80PDwc3/ve9+LXv/71pQsLZV0U\nZU8jirWvReIYgPqY6dyqyROMTgYAZqurvqKccWFXlHVRlD2NKNa+FoljAOqjrleUADBbCSUAJIQS\nABJCCQAJoQSAhFACQEIoASAhlACQEEoASAglACSEEgASQgkACaEEgIRQAkBCKAEgIZQAkBBKAEgI\nJQAkhBIAEkIJAAmhBICEUAJAQigBICGUAJAQSgBICCUAJIQSABJCCQAJoQSAhFACQEIoASAhlACQ\nEEoASAglACSEEgASQgkACaEEgIRQAkBCKAEgIZQAkBBKAEgIJQAkhBIAEkIJAAmhBICEUAJAQigB\nICGUAJAQSgBICCUAJIQSABJCCQAJoQSAhFACQEIoASAhlACQEEoASAglACSEEgASQgkACaEEgIRQ\nAkBCKAEgIZQAkBBKAEgIJQAkhBIAEkIJAAmhBICEUAJAQigBICGUAJAQSgBICCUAJIQSABJCCQAJ\noQSARNWhPHv2bPT19UVXV1d0d3fHyMhILecCgKbQUu0DH3jggbjrrrvil7/8ZVy4cCH+8Y9/1HIu\nAGgKpUqlUrnSB507dy56e3vjz3/+88wLl0pRxdINUSqVGj3CZSvKnkYUa1+LxDEA9THTuVXVrdcT\nJ07ENddcEzt37ozVq1fHfffdFxMTE1c1IAA0o6quKI8ePRrr16+PI0eOxNq1a+PBBx+MefPmxSOP\nPHJx4VIp9u7dO/1+uVyOcrlck6FrrUj/63U1gWMA6mOmc6uqUJ45cybWr18fJ06ciIiIl19+Ofbt\n2xeHDh26uLBbr3VRlD2NKNa+FoljAOqjprdelyxZEkuXLo3jx49HRMTQ0FD09PRUPx0ANKmqrigj\nIl577bXYs2dPTE5OxvXXXx9PPfVUzJ8//+LCrijroih7GlGsfS0SxwDUR01vvV4OoayPouxpRLH2\ntUgcA1AfNb31CgD/L4QSABJCCQAJoQSAhFACQEIoASAhlACQEEoASAglACSEEgASQgkACaEEgIRQ\nAkBCKAEgIZQAkBBKAEgIJQAkhBIAEkIJAAmhBICEUAJAQigBICGUAJAQSgBICCUAJIQSABJCCQAJ\noQSAhFACQEIoASAhlACQEEoASAglACSEEgASQgkACaEEgIRQAkBCKAEgIZQAkBBKAEgIJQAkhBIA\nEkIJAAmhBICEUAJAQigBICGUAJAQSgBICCUAJIQSABJCCQAJoQSAhFACQEIoASAhlACQEEoASAgl\nACSEEgASQgkACaEEgIRQAkBCKAEgIZQAkBBKAEgIJQAkhBIAEkIJAAmhBICEUAJAQigBICGUAJAQ\nSgBICCUAJIQSABJCCQAJoQSAhFACQKLqUA4MDERPT08sX748tm/fHu+9914t5wKAplBVKMfGxuLJ\nJ5+M0dHROHbsWExNTcXBgwdrPRsANFxLNQ+aN29etLa2xsTERHzsYx+LiYmJaG9vr/VsANBwVV1R\nLly4MB5++OG47rrr4tprr40FCxbEbbfdVuvZAKDhSpVKpXKlD3rrrbdi48aN8dJLL8X8+fPjy1/+\ncvT19cU999xzceFSKfbu3Tv9frlcjnK5XJOha61UKjV6hMtWxT9XwxRpX4vEMQD1MdO5VdWt16NH\nj8Ytt9wSixYtioiILVu2xJEjRy4JZUREf39/NcsDQNOo6tZrZ2dnjIyMxLvvvhuVSiWGhoaiu7u7\n1rMBQMNVFcqVK1fGjh07Ys2aNbFixYqIiLj//vtrOhgANIOqnqO8rIVLpcI8l1Kk51GKsqcRxdrX\nInEMQH3MdG75yTwAkBBKAEgIJQAkhBIAEkIJAAmhBICEUAJAQigBICGUAJAQSgBICCUAJIQSABJC\nCQAJoQSAhFACQEIoASAhlACQEEoASAglACSEEgASQgkACaEEgIRQAkBCKAEgIZQAkBBKAEgIJQAk\nhBIAEkIJAAmhBICEUAJAQigBICGUAJAQSgBICCUAJIQSABJCCQAJoQSAhFACQEIoASAhlACQEEoA\nSAglACSEEgASQgkACaEEgIRQAkBCKAEgIZQAkBBKAEgIJQAkhBIAEkIJAAmhBICEUAJAQigBICGU\nAJAQSgBICCUAJIQSABJCCQAJoQSAhFACQEIoASAhlACQEEoASAglACSEEgASQgkACaEEgIRQAkBC\nKAEgIZQAkBBKAEgIJQAkhBIAEkIJAIk0lLt27Yq2trZYvnz59Mf+9re/xYYNG+Kmm26K22+/Pc6e\nPVv3IQGgUdJQ7ty5MwYHBy/52L59+2LDhg1x/Pjx+OIXvxj79u2r64AA0EilSqVSyT5hbGwsNm7c\nGMeOHYuIiM7OzhgeHo62trY4c+ZMlMvl+NOf/vThhUul+B9LN41SqdToES5bUfY0olj7WiSOAaiP\nmc6tlitd6O233462traIiGhra4u33357xs/t7++ffrtcLke5XL7SLwcADXXFofygUqmU/o/xg6EE\ngCK64le9/ueWa0TEX//611i8eHHNhwKAZnHFody0aVM8/fTTERHx9NNPx+bNm2s+FAA0i/TFPNu2\nbYvh4eF45513oq2tLR555JH40pe+FF/5ylfiL3/5Syxbtix+8YtfxIIFCz68sBfz1EVR9jSiWPta\nJI4BqI+Zzq3/+arXagllfRRlTyOKta9F4hiA+pjp3PKTeQAgIZQAkBBKAEgIJQAkhBIAEkIJAAmh\nBICEUAJAQigBICGUAJAQSgBICCUAJIQSABJCCQAJoQSAhFACQEIoASAhlACQEEoASAglACSEEgAS\nQgkACaEEgIRQAkBCKAEgIZQAkBBKAEgIJQAkhBIAEkIJAAmhBICEUAJAQigBICGUAJAQSgBICCUA\nJIQSABJCCQAJoQSAhFACQEIoASAhlACQEEoASAglACSEEgASQgkACaEEgIRQAkBCKAEgIZQAkBBK\nAEgIJQAkhBIAEkIJAAmhBICEUAJAQigBICGUAJAQSgBICCUAJIQSABJCCQAJoQSAhFACQEIoASAh\nlACQEEoASAglACSEEgASQgkACaEEgIRQAkBCKAEgIZQAkBBKAEgIJQAkhBIAEmkod+3aFW1tbbF8\n+fLpj33jG9+Irq6uWLlyZWzZsiXOnTtX9yEBoFHSUO7cuTMGBwcv+djtt98er7/+erz22mtx0003\nxcDAQF0HBIBGSkP5uc99Lj71qU9d8rENGzbEnDn/fti6devi5MmT9ZsOABqs5Woe/OMf/zi2bds2\n49/39/dPv10ul6NcLl/NlwOAj1zVofzOd74TH//4x2P79u0zfs4HQwkARVRVKH/yk5/E4cOH4ze/\n+U2t5wGApnLFoRwcHIz9+/fH8PBwfOITn6jHTADQNEqVSqUy019u27YthoeH45133om2trb49re/\nHQMDAzE5ORkLFy6MiIj169fHj370ow8vXCpFsnRTKZVKjR7hshVlTyOKta9F4hiA+pjp3EpDeTWE\nsj6KsqcRxdrXInEMQH3MdG75yTwAkBBKAEgIJQAkhBIAEkIJAAmhBICEUAJAQigBICGUAJAQSgBI\nCCUAJIQSABKFC+WLL77Y6BFmHXsKMDOhxJ4CJAoXSgD4KAklACTq+oubAaAoZsphy0f9BQGgSNx6\nBYCEUAJAojChHBwcjM7Ozrjxxhvj0UcfbfQ4s8L4+Hjceuut0dPTEzfffHM8/vjjjR5pVpmamore\n3t7YuHFjo0eZNc6ePRt9fX3R1dUV3d3dMTIy0uiRCm9gYCB6enpi+fLlsX379njvvfcaPVLTKUQo\np6am4utf/3oMDg7GH//4x3j22WfjjTfeaPRYhdfa2hqPPfZYvP766zEyMhI//OEP7WsNHThwILq7\nu72wrYYeeOCBuOuuu+KNN96IP/zhD9HV1dXokQptbGwsnnzyyRgdHY1jx47F1NRUHDx4sNFjNZ1C\nhPKVV16JG264IZYtWxatra1x9913x/PPP9/osQpvyZIlsWrVqoiImDt3bnR1dcXp06cbPNXscPLk\nyTh8+HDs2bPHC9tq5Ny5c/HSSy/Frl27IiKipaUl5s+f3+Cpim3evHnR2toaExMTceHChZiYmIj2\n9vZGj9V0ChHKU6dOxdKlS6ff7+joiFOnTjVwotlnbGwsXn311Vi3bl2jR5kVHnroodi/f3/MmVOI\nU6wQTpw4Eddcc03s3LkzVq9eHffdd19MTEw0eqxCW7hwYTz88MNx3XXXxbXXXhsLFiyI2267rdFj\nNZ1CnMVuXdXX+fPno6+vLw4cOBBz585t9DiFd+jQoVi8eHH09va6mqyhCxcuxOjoaHzta1+L0dHR\n+OQnPxn79u1r9FiF9tZbb8X3v//9GBsbi9OnT8f58+fj5z//eaPHajqFCGV7e3uMj49Pvz8+Ph4d\nHR0NnGj2eP/992Pr1q1x7733xubNmxs9zqxw5MiReOGFF+LTn/50bNu2LX7729/Gjh07Gj1W4XV0\ndERHR0esXbs2IiL6+vpidHS0wVMV29GjR+OWW26JRYsWRUtLS2zZsiWOHDnS6LGaTiFCuWbNmnjz\nzTdjbGwsJicn47nnnotNmzY1eqzCq1QqsXv37uju7o4HH3yw0ePMGt/97ndjfHw8Tpw4EQcPHowv\nfOEL8cwzzzR6rMJbsmRJLF26NI4fPx4REUNDQ9HT09PgqYqts7MzRkZG4t13341KpRJDQ0PR3d3d\n6LGaTt1+Mk8ttbS0xBNPPBF33HFHTE1Nxe7du73arQZ+97vfxc9+9rNYsWJF9Pb2RsS/Xyp+5513\nNniy2cVTB7Xzgx/8IO65556YnJyM66+/Pp566qlGj1RoK1eujB07dsSaNWtizpw5sXr16rj//vsb\nPVbTqdvPegWA2aAQt14BoFGEEgASQgkACaEEgIRQAkBCKAEg8S+8A81h+ix/9QAAAABJRU5ErkJg\ngg==\n",
       "text": [
        "<matplotlib.figure.Figure at 0xa148090>"
       ]
      }
     ],
     "prompt_number": 110
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def uniqueSym(symbolSeries):\n",
      "    symbolSeries = symbolSeries.fillna(\"-\")\n",
      "    r = set(symbolSeries)\n",
      "    r.discard(nan)\n",
      "    r.discard(\"-\")\n",
      "    return(r)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 120
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "targets = cz.target.unique()\n",
      "tfs = cz.tf.unique()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "chea = pd.read_table(\"/data/adrian/Dropbox/Data/chea/chea-background.csv\", header=None, \n",
      "                     names=[\"chea_id\", \"tf\", \"tf_pmid\", \"target\", \n",
      "                            \"pmid\", \"exp_type\", \"cell_type\", \"organism\", \"date\"], sep=\",\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 112
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "chea = chea.astype('unicode')"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 113
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "czm = zeros( [len(targets), len(tfs)] )\n",
      "target_index = dict( (a[1], a[0]) for a in enumerate(targets) )\n",
      "tf_index = dict( (a[1], a[0]) for a in enumerate(tfs) )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "pandasql.sqldf(\"select tf, count(*) c from cz group by tf order by c desc\", locals())[0:10]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for i in cz.index:\n",
      "    czm[target_index[cz.ix[i, \"target\"]], tf_index[cz.ix[i, \"tf\"]]] = 1"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "czm_df = pandas.DataFrame(data=czm, index=target_index, columns=tf_index)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 184
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 185
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import IPython"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 136
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "IPython.display.HTML( cz[0:5].to_html() )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp73_dopamine_depletion_up"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<pre>\n",
        "&ltclass 'pandas.core.frame.DataFrame'&gt\n",
        "Index: 92 entries, 1438967_x_at to 1426734_at\n",
        "Data columns (total 5 columns):\n",
        "probe_id                                                                                                                                 88  non-null values\n",
        "symbol                                                                                                                                   88  non-null values\n",
        "gene_name                                                                                                                                88  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]    92  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"fc_medians\"]]                                   92  non-null values\n",
        "dtypes: float64(2), object(3)\n",
        "</pre>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 179,
       "text": [
        "<class 'pandas.core.frame.DataFrame'>\n",
        "Index: 92 entries, 1438967_x_at to 1426734_at\n",
        "Data columns (total 5 columns):\n",
        "probe_id                                                                                                                                 88  non-null values\n",
        "symbol                                                                                                                                   88  non-null values\n",
        "gene_name                                                                                                                                88  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]    92  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"fc_medians\"]]                                   92  non-null values\n",
        "dtypes: float64(2), object(3)"
       ]
      }
     ],
     "prompt_number": 179
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import itertools"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 141
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "177 * 177 * 177"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 156,
       "text": [
        "5545233"
       ]
      }
     ],
     "prompt_number": 156
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len( cz.groupby(\"target\").groups.items() )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 186,
       "text": [
        "1763"
       ]
      }
     ],
     "prompt_number": 186
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len(  list( itertools.combinations( cz.ix[g, \"tf\"].unique(), 3) ) )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 206,
       "text": [
        "2925"
       ]
      }
     ],
     "prompt_number": 206
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def countTFcombos( genegroup, combination_size ):\n",
      "    z = [a.upper() for a in set( uniqueSym( genegroup.symbol ))]\n",
      "    z_set = set(z)\n",
      "    cz= chea.select( lambda x: chea.ix[x, \"target\"] in z_set)\n",
      "    \n",
      "    print(len(cz.index))\n",
      "    \n",
      "    pair_counts = {}\n",
      "    for (target_group, g) in cz.groupby(\"target\").groups.items():\n",
      "        group_tfs = cz.ix[g, \"tf\"].unique()\n",
      "        group_tfs.sort()\n",
      "#        for combo in itertools.combinations( group_tfs , combination_size):\n",
      "#            pair_counts[combo] = pair_counts.setdefault(combo, 0) + 1\n",
      "        for tf in group_tfs:\n",
      "            pair_counts[tf] = pair_counts.setdefault(tf, 0) + 1\n",
      "    result = pd.DataFrame( pair_counts.items(), columns=[\"tf\", \"count\"]).sort_index(by=\"count\", ascending=False) \n",
      "    result[\"total_genes_in_group\"] = len(z)\n",
      "    #ret\n",
      "    return result"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 429
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "g"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<pre>\n",
        "&ltclass 'pandas.core.frame.DataFrame'&gt\n",
        "Int64Index: 2529 entries, 102 to 276607\n",
        "Data columns (total 6 columns):\n",
        "genename           2529  non-null values\n",
        "chrom              2529  non-null values\n",
        "motif_type         2529  non-null values\n",
        "motif_name         2529  non-null values\n",
        "max_motif_score    2529  non-null values\n",
        "motif_count        2529  non-null values\n",
        "dtypes: float64(1), int64(1), object(4)\n",
        "</pre>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 558,
       "text": [
        "<class 'pandas.core.frame.DataFrame'>\n",
        "Int64Index: 2529 entries, 102 to 276607\n",
        "Data columns (total 6 columns):\n",
        "genename           2529  non-null values\n",
        "chrom              2529  non-null values\n",
        "motif_type         2529  non-null values\n",
        "motif_name         2529  non-null values\n",
        "max_motif_score    2529  non-null values\n",
        "motif_count        2529  non-null values\n",
        "dtypes: float64(1), int64(1), object(4)"
       ]
      }
     ],
     "prompt_number": 558
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def countMotifCombos( genegroup, combination_size ):\n",
      "    z = [a for a in set( uniqueSym( genegroup.symbol ))]\n",
      "    z_set = set(z)\n",
      "    cz= mm9_gene_motifs.select( lambda x: mm9_gene_motifs.ix[x, \"genename\"] in z_set)\n",
      "    \n",
      "    print(len(cz.index))\n",
      "    \n",
      "    pair_counts = {}\n",
      "    for (target_group, g) in cz.groupby(\"genename\").groups.items():\n",
      "        group_motifs = cz.ix[g, \"motif_name\"].unique()\n",
      "        group_motifs.sort()\n",
      "#        for combo in itertools.combinations( group_tfs , combination_size):\n",
      "#            pair_counts[combo] = pair_counts.setdefault(combo, 0) + 1\n",
      "        for motif_name in group_motifs:\n",
      "            pair_counts[motif_name] = pair_counts.setdefault(motif_name, 0) + 1\n",
      "    result = pd.DataFrame( pair_counts.items(), columns=[\"motif_name\", \"count\"]).sort_index(by=\"count\", ascending=False) \n",
      "    result[\"total_genes_in_group\"] = len(z)\n",
      "    #ret\n",
      "    return result\n",
      "\n",
      "\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 500
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def getMotifMatches( genegroup, combination_size ):\n",
      "#    z = [a for a in set( uniqueSym( genegroup.symbol ))]\n",
      "#    z_set = set(z)\n",
      "    cz= mm9_gene_motifs.merge(genegroup, left_on=\"genename\", right_on=\"symbol\") \n",
      "    #select( lambda x: mm9_gene_motifs.ix[x, \"genename\"] in z_set)\n",
      "    \n",
      "    print(len(cz.index))\n",
      "    \n",
      "    result = []\n",
      "    for k, z in cz.groupby(\"motif_name\"):\n",
      "        result.append( { \"motif_name\"  : k,\n",
      "                        \"\n",
      "                         \"motif_occurrences\"\n",
      "                        }  )\n",
      "    \n",
      "    \n",
      "    for (target_group, g) in cz.groupby(\"genename\").groups.items():\n",
      "        group_motifs = cz.ix[g, \"motif_name\"].unique()\n",
      "        group_motifs.sort()\n",
      "#        for combo in itertools.combinations( group_tfs , combination_size):\n",
      "#            pair_counts[combo] = pair_counts.setdefault(combo, 0) + 1\n",
      "        for motif_name in group_motifs:\n",
      "            pair_counts[motif_name] = pair_counts.setdefault(motif_name, 0) + 1\n",
      "    result = pd.DataFrame( pair_counts.items(), columns=[\"motif_name\", \"count\"]).sort_index(by=\"count\", ascending=False) \n",
      "    result[\"total_genes_in_group\"] = len(z)\n",
      "    #ret\n",
      "    return result\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 584
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "r, c = getMotifMatches( cp73_chronicHigh_down, 1)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "53342\n"
       ]
      }
     ],
     "prompt_number": 585
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "c"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<pre>\n",
        "&ltclass 'pandas.core.frame.DataFrame'&gt\n",
        "Int64Index: 53342 entries, 0 to 53341\n",
        "Data columns (total 9 columns):\n",
        "genename                                                                                                                            53342  non-null values\n",
        "chrom                                                                                                                               53342  non-null values\n",
        "motif_type                                                                                                                          53342  non-null values\n",
        "motif_name                                                                                                                          53342  non-null values\n",
        "max_motif_score                                                                                                                     53342  non-null values\n",
        "motif_count                                                                                                                         53342  non-null values\n",
        "symbol                                                                                                                              53342  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]    53342  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"fc_medians\"]]                                   53342  non-null values\n",
        "dtypes: float64(3), int64(1), object(5)\n",
        "</pre>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 586,
       "text": [
        "<class 'pandas.core.frame.DataFrame'>\n",
        "Int64Index: 53342 entries, 0 to 53341\n",
        "Data columns (total 9 columns):\n",
        "genename                                                                                                                            53342  non-null values\n",
        "chrom                                                                                                                               53342  non-null values\n",
        "motif_type                                                                                                                          53342  non-null values\n",
        "motif_name                                                                                                                          53342  non-null values\n",
        "max_motif_score                                                                                                                     53342  non-null values\n",
        "motif_count                                                                                                                         53342  non-null values\n",
        "symbol                                                                                                                              53342  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]    53342  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"fc_medians\"]]                                   53342  non-null values\n",
        "dtypes: float64(3), int64(1), object(5)"
       ]
      }
     ],
     "prompt_number": 586
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "IPython.display.HTML( c.sort_index(by=c.columns[8], ascending=False).to_html() )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "KeyboardInterrupt",
       "evalue": "",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
        "\u001b[1;32m<ipython-input-582-49a39dd90e1e>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mIPython\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdisplay\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mHTML\u001b[0m\u001b[1;33m(\u001b[0m \u001b[0mc\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msort_index\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mby\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mc\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m8\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mascending\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mFalse\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mto_html\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/frame.pyc\u001b[0m in \u001b[0;36mto_html\u001b[1;34m(self, buf, columns, col_space, colSpace, header, index, na_rep, formatters, float_format, sparsify, index_names, justify, force_unicode, bold_rows, classes, escape)\u001b[0m\n\u001b[0;32m   1530\u001b[0m                                            \u001b[0mbold_rows\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mbold_rows\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1531\u001b[0m                                            escape=escape)\n\u001b[1;32m-> 1532\u001b[1;33m         \u001b[0mformatter\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mto_html\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mclasses\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mclasses\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1533\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1534\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mbuf\u001b[0m \u001b[1;32mis\u001b[0m \u001b[0mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/format.pyc\u001b[0m in \u001b[0;36mto_html\u001b[1;34m(self, classes)\u001b[0m\n\u001b[0;32m    410\u001b[0m         \"\"\"\n\u001b[0;32m    411\u001b[0m         \u001b[0mhtml_renderer\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mHTMLFormatter\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mclasses\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mclasses\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 412\u001b[1;33m         \u001b[0mhtml_renderer\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mwrite_result\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbuf\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    413\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    414\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_get_formatted_column_labels\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/format.pyc\u001b[0m in \u001b[0;36mwrite_result\u001b[1;34m(self, buf)\u001b[0m\n\u001b[0;32m    589\u001b[0m             \u001b[0mindent\u001b[0m \u001b[1;33m+=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mindent_delta\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    590\u001b[0m             \u001b[0mindent\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_write_header\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mindent\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 591\u001b[1;33m             \u001b[0mindent\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_write_body\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mindent\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    592\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    593\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mwrite\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'</table>'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindent\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/format.pyc\u001b[0m in \u001b[0;36m_write_body\u001b[1;34m(self, indent)\u001b[0m\n\u001b[0;32m    682\u001b[0m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_write_hierarchical_rows\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfmt_values\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindent\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    683\u001b[0m             \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 684\u001b[1;33m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_write_regular_rows\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfmt_values\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindent\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    685\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    686\u001b[0m             \u001b[1;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mframe\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/format.pyc\u001b[0m in \u001b[0;36m_write_regular_rows\u001b[1;34m(self, fmt_values, indent)\u001b[0m\n\u001b[0;32m    708\u001b[0m             \u001b[0mrow\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mextend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfmt_values\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mj\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0mj\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mncols\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    709\u001b[0m             self.write_tr(row, indent, self.indent_delta, tags=None,\n\u001b[1;32m--> 710\u001b[1;33m                           nindex_levels=1)\n\u001b[0m\u001b[0;32m    711\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    712\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_write_hierarchical_rows\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfmt_values\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindent\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/format.pyc\u001b[0m in \u001b[0;36mwrite_tr\u001b[1;34m(self, line, indent, indent_delta, header, align, tags, nindex_levels)\u001b[0m\n\u001b[0;32m    558\u001b[0m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mwrite_th\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0ms\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindent\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtags\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mval_tag\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    559\u001b[0m             \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 560\u001b[1;33m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mwrite_td\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0ms\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindent\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtags\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mval_tag\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    561\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    562\u001b[0m         \u001b[0mindent\u001b[0m \u001b[1;33m-=\u001b[0m \u001b[0mindent_delta\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/format.pyc\u001b[0m in \u001b[0;36mwrite_td\u001b[1;34m(self, s, indent, tags)\u001b[0m\n\u001b[0;32m    523\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    524\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mwrite_td\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0ms\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindent\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtags\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mNone\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 525\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_write_cell\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0ms\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mkind\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'td'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindent\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mindent\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtags\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mtags\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    526\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    527\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_write_cell\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0ms\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mkind\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'td'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindent\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtags\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mNone\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/format.pyc\u001b[0m in \u001b[0;36m_write_cell\u001b[1;34m(self, s, kind, indent, tags)\u001b[0m\n\u001b[0;32m    540\u001b[0m         \u001b[0mrs\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpprint_thing\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0ms\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mescape_chars\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mesc\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    541\u001b[0m         self.write(\n\u001b[1;32m--> 542\u001b[1;33m             '%s%s</%s>' % (start_tag, rs, kind), indent)\n\u001b[0m\u001b[0;32m    543\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    544\u001b[0m     def write_tr(self, line, indent=0, indent_delta=4, header=False,\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/format.pyc\u001b[0m in \u001b[0;36mwrite\u001b[1;34m(self, s, indent)\u001b[0m\n\u001b[0;32m    510\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mescape\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfmt\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mkwds\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'escape'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mTrue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    511\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 512\u001b[1;33m     \u001b[1;32mdef\u001b[0m \u001b[0mwrite\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0ms\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindent\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    513\u001b[0m         \u001b[0mrs\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpprint_thing\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0ms\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    514\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0melements\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m' '\u001b[0m \u001b[1;33m*\u001b[0m \u001b[0mindent\u001b[0m \u001b[1;33m+\u001b[0m \u001b[0mrs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
       ]
      }
     ],
     "prompt_number": 582
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for k, g in cz.groupby(\"tf\"):\n",
      "    print k, len(g.target.unique())"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "AHR 87\n",
        "AR 545\n",
        "ARNT 116\n",
        "ASH2L 394\n",
        "ATF3 268\n",
        "BACH1 121\n",
        "BCL11B 4\n",
        "BCL3 113\n",
        "BMI1 143\n",
        "BP1 3\n",
        "CCND1 247\n",
        "CDKN2AIP 21\n",
        "CDX2 81\n",
        "CEBPA 64\n",
        "CEBPB 204\n",
        "CEBPD 57\n",
        "CHD1 108\n",
        "CHD7 9\n",
        "CIITA 5\n",
        "CLOCK 86\n",
        "CNOT3 157\n",
        "CREB1 701\n",
        "CREM 868\n",
        "CRX 80\n",
        "CTCF 152\n",
        "CTNNB1 150\n",
        "CUX1 189\n",
        "DACH1 193\n",
        "DCP1A 94\n",
        "DMRT1 266\n",
        "DNAJC2 78\n",
        "DROSHA 36\n",
        "E2F1 612\n",
        "E2F4 371\n",
        "E2F7 9\n",
        "EBNA2 112\n",
        "EED 59\n",
        "EGR1 653\n",
        "EKLF 164\n",
        "ELF1 91\n",
        "ELF5 125\n",
        "ELK1 204\n",
        "EOMES 292\n",
        "EP300 248\n",
        "ERG 337\n",
        "ESR1 229\n",
        "ESR2 53\n",
        "ESRRB 155\n",
        "EST1 55\n",
        "ETS1 149\n",
        "ETS2 29\n",
        "EWS-ERG 83\n",
        "EWS-FLI1 49\n",
        "EZH2 123\n",
        "FLI1 725\n",
        "FOXA2 293\n",
        "FOXM1 31\n",
        "FOXO1 36\n",
        "FOXO3 312\n",
        "FOXP1 410\n",
        "FOXP2 214\n",
        "FOXP3 155\n",
        "GABP 322\n",
        "GATA1 436\n",
        "GATA2 431\n",
        "GATA3 89\n",
        "GATA4 203\n",
        "GBX2 25\n",
        "GFI1B 235\n",
        "GLI1 2\n",
        "HCFC1 23\n",
        "HIF1A 51\n",
        "HNF4A 625\n",
        "HOXA2 5\n",
        "HOXB4 201\n",
        "HOXD13 17\n",
        "HSF1 97\n",
        "HTT 30\n",
        "IKZF1 8\n",
        "IRF1 48\n",
        "IRF8 87\n",
        "JARID1A 217\n",
        "JARID2 169\n",
        "JUN 138\n",
        "KDM5B 426\n",
        "KDM6A 57\n",
        "KLF1 134\n",
        "KLF2 16\n",
        "KLF4 513\n",
        "KLF5 16\n",
        "LMO2 173\n",
        "LYL1 88\n",
        "MECOM 236\n",
        "MEF2A 94\n",
        "MEIS1 189\n",
        "MITF 578\n",
        "MTF2 327\n",
        "MYB 134\n",
        "MYBL1 18\n",
        "MYBL2 200\n",
        "MYC 860\n",
        "MYCN 295\n",
        "NACC1 77\n",
        "NANOG 641\n",
        "NFE2L2 174\n",
        "NKX2-5 145\n",
        "NOTCH1 39\n",
        "NR0B1 177\n",
        "NR1H3 76\n",
        "NR1I2 83\n",
        "NR3C1 201\n",
        "NR4A2 19\n",
        "NRF2 133\n",
        "OLIG2 227\n",
        "PADI4 112\n",
        "PAX3-FKHR 112\n",
        "PAX6 72\n",
        "PBX1 155\n",
        "PDX1 67\n",
        "PHC1 67\n",
        "PHF8 180\n",
        "POU3F2 140\n",
        "POU5F1 595\n",
        "PPARD 387\n",
        "PPARG 556\n",
        "PRDM14 228\n",
        "PRDM16 15\n",
        "PRDM5 110\n",
        "RAD21 181\n",
        "RARG 55\n",
        "RBPJ 160\n",
        "RCOR1 136\n",
        "RCOR2 44\n",
        "RCOR3 329\n",
        "REST 467\n",
        "RNF2 171\n",
        "RUNX1 761\n",
        "RUNX2 291\n",
        "SALL1 11\n",
        "SALL4 306\n",
        "SCL 191\n",
        "SETDB1 371\n",
        "SFPI1 328\n",
        "SIN3A 140\n",
        "SIN3B 454\n",
        "SMAD 12\n",
        "SMAD1 64\n",
        "SMAD2 230\n",
        "SMAD3 335\n",
        "SMAD4 464\n",
        "SMARCA4 375\n",
        "SOX11 94\n",
        "SOX17 183\n",
        "SOX2 778\n",
        "SOX9 5\n",
        "SPI1 780\n",
        "SREBP1 81\n",
        "SREBP2 109\n",
        "SRF 143\n",
        "SRY 13\n",
        "STAT1 37\n",
        "STAT3 617\n",
        "STAT4 171\n",
        "STAT5 118\n",
        "STAT6 47\n",
        "SUZ12 485\n",
        "TAF7L 97\n",
        "TAL1 462\n",
        "TBP 100\n",
        "TBX3 87\n",
        "TBX5 208\n",
        "TCF21 12\n",
        "TCF3 381\n",
        "TCF4 380\n",
        "TCF7 146\n",
        "TCF7L2 64\n",
        "TCFAP2C 291\n",
        "TCFCP2L1 220\n",
        "TEAD4 221\n",
        "TET1 405\n",
        "TFAP2A 168\n",
        "TFAP2C 122\n",
        "TFEB 96\n",
        "THAP11 72\n",
        "THRA 19\n",
        "TP53 540\n",
        "TP63 605\n",
        "TRIM28 429\n",
        "TRP63 20\n",
        "TTF2 203\n",
        "VDR 300\n",
        "WT1 260\n",
        "XRN2 178\n",
        "YAP1 263\n",
        "YY1 142\n",
        "ZFP281 238\n",
        "ZFP42 144\n",
        "ZFX 360\n",
        "ZIC3 42\n",
        "ZNF263 12\n",
        "ZNF274 17\n",
        "ZNF652 6\n"
       ]
      }
     ],
     "prompt_number": 567
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cz.groupby(\"motif_name\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "KeyError",
       "evalue": "u'no item named motif_name'",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
        "\u001b[1;32m<ipython-input-560-67798268a3ef>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mcz\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mgroupby\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"motif_name\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/generic.pyc\u001b[0m in \u001b[0;36mgroupby\u001b[1;34m(self, by, axis, level, as_index, sort, group_keys)\u001b[0m\n\u001b[0;32m    151\u001b[0m         \u001b[0maxis\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_get_axis_number\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    152\u001b[0m         return groupby(self, by, axis=axis, level=level, as_index=as_index,\n\u001b[1;32m--> 153\u001b[1;33m                        sort=sort, group_keys=group_keys)\n\u001b[0m\u001b[0;32m    154\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    155\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0masfreq\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfreq\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mNone\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mhow\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mNone\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnormalize\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mFalse\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/groupby.pyc\u001b[0m in \u001b[0;36mgroupby\u001b[1;34m(obj, by, **kwds)\u001b[0m\n\u001b[0;32m    535\u001b[0m         \u001b[1;32mraise\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'invalid type: %s'\u001b[0m \u001b[1;33m%\u001b[0m \u001b[0mtype\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    536\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 537\u001b[1;33m     \u001b[1;32mreturn\u001b[0m \u001b[0mklass\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mby\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    538\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    539\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/groupby.pyc\u001b[0m in \u001b[0;36m__init__\u001b[1;34m(self, obj, keys, axis, level, grouper, exclusions, selection, as_index, sort, group_keys)\u001b[0m\n\u001b[0;32m    193\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mgrouper\u001b[0m \u001b[1;32mis\u001b[0m \u001b[0mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    194\u001b[0m             grouper, exclusions = _get_grouper(obj, keys, axis=axis,\n\u001b[1;32m--> 195\u001b[1;33m                                                level=level, sort=sort)\n\u001b[0m\u001b[0;32m    196\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    197\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mgrouper\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mgrouper\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/groupby.pyc\u001b[0m in \u001b[0;36m_get_grouper\u001b[1;34m(obj, key, axis, level, sort)\u001b[0m\n\u001b[0;32m   1318\u001b[0m             \u001b[0mexclusions\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mgpr\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1319\u001b[0m             \u001b[0mname\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mgpr\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1320\u001b[1;33m             \u001b[0mgpr\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mobj\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mgpr\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1321\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1322\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0misinstance\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mgpr\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mCategorical\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mand\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mgpr\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m!=\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/frame.pyc\u001b[0m in \u001b[0;36m__getitem__\u001b[1;34m(self, key)\u001b[0m\n\u001b[0;32m   1926\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1927\u001b[0m             \u001b[1;31m# get column\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1928\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_get_item_cache\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1929\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1930\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_getitem_slice\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/generic.pyc\u001b[0m in \u001b[0;36m_get_item_cache\u001b[1;34m(self, item)\u001b[0m\n\u001b[0;32m    568\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mcache\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mitem\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    569\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mException\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 570\u001b[1;33m             \u001b[0mvalues\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    571\u001b[0m             \u001b[0mres\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_box_item_values\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvalues\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    572\u001b[0m             \u001b[0mcache\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mitem\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mres\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/internals.pyc\u001b[0m in \u001b[0;36mget\u001b[1;34m(self, item)\u001b[0m\n\u001b[0;32m   1381\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1382\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mitem\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1383\u001b[1;33m         \u001b[0m_\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mblock\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_find_block\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1384\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mblock\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1385\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/internals.pyc\u001b[0m in \u001b[0;36m_find_block\u001b[1;34m(self, item)\u001b[0m\n\u001b[0;32m   1523\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1524\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_find_block\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mitem\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1525\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_check_have\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1526\u001b[0m         \u001b[1;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mblock\u001b[0m \u001b[1;32min\u001b[0m \u001b[0menumerate\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mblocks\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1527\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mitem\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mblock\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/internals.pyc\u001b[0m in \u001b[0;36m_check_have\u001b[1;34m(self, item)\u001b[0m\n\u001b[0;32m   1530\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_check_have\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mitem\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1531\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mitem\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1532\u001b[1;33m             \u001b[1;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'no item named %s'\u001b[0m \u001b[1;33m%\u001b[0m \u001b[0mcom\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpprint_thing\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1533\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1534\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mreindex_axis\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnew_axis\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mNone\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mTrue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;31mKeyError\u001b[0m: u'no item named motif_name'"
       ]
      }
     ],
     "prompt_number": 560
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "countMotifCombos( cp73_chronicHigh_up, 1)[0:20]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "29880\n"
       ]
      },
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>motif_name</th>\n",
        "      <th>count</th>\n",
        "      <th>total_genes_in_group</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>28 </th>\n",
        "      <td>               SP1.p2</td>\n",
        "      <td> 1333</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1  </th>\n",
        "      <td>              KLF4.p3</td>\n",
        "      <td> 1192</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>55 </th>\n",
        "      <td>               MAZ.p2</td>\n",
        "      <td> 1176</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>143</th>\n",
        "      <td>             PATZ1.p2</td>\n",
        "      <td> 1127</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>17 </th>\n",
        "      <td>             TFDP1.p2</td>\n",
        "      <td>  998</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>126</th>\n",
        "      <td>        TFAP2{A,C}.p2</td>\n",
        "      <td>  996</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>67 </th>\n",
        "      <td>              MZF1.p2</td>\n",
        "      <td>  901</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>168</th>\n",
        "      <td>            TFAP2B.p2</td>\n",
        "      <td>  799</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>29 </th>\n",
        "      <td>             GTF2I.p2</td>\n",
        "      <td>  744</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>148</th>\n",
        "      <td>              PAX5.p2</td>\n",
        "      <td>  732</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>68 </th>\n",
        "      <td>          ELF1,2,4.p2</td>\n",
        "      <td>  689</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>120</th>\n",
        "      <td>              SPI1.p2</td>\n",
        "      <td>  614</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>57 </th>\n",
        "      <td>              HIC1.p2</td>\n",
        "      <td>  585</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>94 </th>\n",
        "      <td> ELK1,4_GABP{A,B1}.p3</td>\n",
        "      <td>  584</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>154</th>\n",
        "      <td>               YY1.p2</td>\n",
        "      <td>  561</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>93 </th>\n",
        "      <td>           EGR1..3.p2</td>\n",
        "      <td>  496</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>23 </th>\n",
        "      <td>             RREB1.p2</td>\n",
        "      <td>  474</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>102</th>\n",
        "      <td>               FEV.p2</td>\n",
        "      <td>  458</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>105</th>\n",
        "      <td>              MAFB.p2</td>\n",
        "      <td>  456</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>82 </th>\n",
        "      <td>        TCF4_dimer.p2</td>\n",
        "      <td>  454</td>\n",
        "      <td> 1871</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 504,
       "text": [
        "               motif_name  count  total_genes_in_group\n",
        "28                 SP1.p2   1333                  1871\n",
        "1                 KLF4.p3   1192                  1871\n",
        "55                 MAZ.p2   1176                  1871\n",
        "143              PATZ1.p2   1127                  1871\n",
        "17               TFDP1.p2    998                  1871\n",
        "126         TFAP2{A,C}.p2    996                  1871\n",
        "67                MZF1.p2    901                  1871\n",
        "168             TFAP2B.p2    799                  1871\n",
        "29               GTF2I.p2    744                  1871\n",
        "148               PAX5.p2    732                  1871\n",
        "68            ELF1,2,4.p2    689                  1871\n",
        "120               SPI1.p2    614                  1871\n",
        "57                HIC1.p2    585                  1871\n",
        "94   ELK1,4_GABP{A,B1}.p3    584                  1871\n",
        "154                YY1.p2    561                  1871\n",
        "93             EGR1..3.p2    496                  1871\n",
        "23               RREB1.p2    474                  1871\n",
        "102                FEV.p2    458                  1871\n",
        "105               MAFB.p2    456                  1871\n",
        "82          TCF4_dimer.p2    454                  1871"
       ]
      }
     ],
     "prompt_number": 504
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "test = countTFcombos( cp73_chronicHigh_up, 1)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "47501\n"
       ]
      }
     ],
     "prompt_number": 430
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "test = pd.DataFrame( countTFcombos( cp73_chronicHigh_up, 1).items(), columns=[\"tf\", \"count\"]).sort_index(by=\"count\", ascending=False) "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "AttributeError",
       "evalue": "'DataFrame' object has no attribute 'items'",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
        "\u001b[1;32m<ipython-input-416-9b1a8a1ebaee>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mtest\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[1;33m(\u001b[0m \u001b[0mcountTFcombos\u001b[0m\u001b[1;33m(\u001b[0m \u001b[0mcp73_chronicHigh_up\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;36m1\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"tf\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"count\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msort_index\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mby\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m\"count\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mascending\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mFalse\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[1;32m/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/frame.pyc\u001b[0m in \u001b[0;36m__getattr__\u001b[1;34m(self, name)\u001b[0m\n\u001b[0;32m   2005\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2006\u001b[0m         raise AttributeError(\"'%s' object has no attribute '%s'\" %\n\u001b[1;32m-> 2007\u001b[1;33m                              (type(self).__name__, name))\n\u001b[0m\u001b[0;32m   2008\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2009\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m__setattr__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;31mAttributeError\u001b[0m: 'DataFrame' object has no attribute 'items'"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "47501\n"
       ]
      }
     ],
     "prompt_number": 416
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "test[0:10].merge(tf_target_counts, left_on=\"tf\", right_index=True)[0:10]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>tf</th>\n",
        "      <th>tf_x</th>\n",
        "      <th>count</th>\n",
        "      <th>total_genes_in_group</th>\n",
        "      <th>chea_target_count</th>\n",
        "      <th>tf_y</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>93 </th>\n",
        "      <td>  CREM</td>\n",
        "      <td>  CREM</td>\n",
        "      <td> 868</td>\n",
        "      <td> 1871</td>\n",
        "      <td> 5776</td>\n",
        "      <td>  CREM</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>18 </th>\n",
        "      <td>   MYC</td>\n",
        "      <td>   MYC</td>\n",
        "      <td> 860</td>\n",
        "      <td> 1871</td>\n",
        "      <td> 9863</td>\n",
        "      <td>   MYC</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>75 </th>\n",
        "      <td>  SPI1</td>\n",
        "      <td>  SPI1</td>\n",
        "      <td> 780</td>\n",
        "      <td> 1871</td>\n",
        "      <td> 8273</td>\n",
        "      <td>  SPI1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>152</th>\n",
        "      <td>  SOX2</td>\n",
        "      <td>  SOX2</td>\n",
        "      <td> 778</td>\n",
        "      <td> 1871</td>\n",
        "      <td> 9261</td>\n",
        "      <td>  SOX2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>38 </th>\n",
        "      <td> RUNX1</td>\n",
        "      <td> RUNX1</td>\n",
        "      <td> 761</td>\n",
        "      <td> 1871</td>\n",
        "      <td> 7624</td>\n",
        "      <td> RUNX1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>25 </th>\n",
        "      <td>  FLI1</td>\n",
        "      <td>  FLI1</td>\n",
        "      <td> 725</td>\n",
        "      <td> 1871</td>\n",
        "      <td> 7063</td>\n",
        "      <td>  FLI1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>26 </th>\n",
        "      <td> CREB1</td>\n",
        "      <td> CREB1</td>\n",
        "      <td> 701</td>\n",
        "      <td> 1871</td>\n",
        "      <td> 4907</td>\n",
        "      <td> CREB1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>59 </th>\n",
        "      <td>  EGR1</td>\n",
        "      <td>  EGR1</td>\n",
        "      <td> 653</td>\n",
        "      <td> 1871</td>\n",
        "      <td> 7207</td>\n",
        "      <td>  EGR1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>114</th>\n",
        "      <td> NANOG</td>\n",
        "      <td> NANOG</td>\n",
        "      <td> 641</td>\n",
        "      <td> 1871</td>\n",
        "      <td> 8386</td>\n",
        "      <td> NANOG</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>87 </th>\n",
        "      <td> HNF4A</td>\n",
        "      <td> HNF4A</td>\n",
        "      <td> 625</td>\n",
        "      <td> 1871</td>\n",
        "      <td> 6749</td>\n",
        "      <td> HNF4A</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 425,
       "text": [
        "        tf   tf_x  count  total_genes_in_group  chea_target_count   tf_y\n",
        "93    CREM   CREM    868                  1871               5776   CREM\n",
        "18     MYC    MYC    860                  1871               9863    MYC\n",
        "75    SPI1   SPI1    780                  1871               8273   SPI1\n",
        "152   SOX2   SOX2    778                  1871               9261   SOX2\n",
        "38   RUNX1  RUNX1    761                  1871               7624  RUNX1\n",
        "25    FLI1   FLI1    725                  1871               7063   FLI1\n",
        "26   CREB1  CREB1    701                  1871               4907  CREB1\n",
        "59    EGR1   EGR1    653                  1871               7207   EGR1\n",
        "114  NANOG  NANOG    641                  1871               8386  NANOG\n",
        "87   HNF4A  HNF4A    625                  1871               6749  HNF4A"
       ]
      }
     ],
     "prompt_number": 425
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "test.merge(tf_target_counts, left_on=\"tf\", right_index=True)[0:10]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>tf</th>\n",
        "      <th>tf_x</th>\n",
        "      <th>count</th>\n",
        "      <th>total_genes</th>\n",
        "      <th>chea_target_count</th>\n",
        "      <th>tf_y</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>13</th>\n",
        "      <td>   CREB1</td>\n",
        "      <td>   CREB1</td>\n",
        "      <td> 9</td>\n",
        "      <td> 10</td>\n",
        "      <td> 4907</td>\n",
        "      <td>   CREB1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>54</th>\n",
        "      <td>    CREM</td>\n",
        "      <td>    CREM</td>\n",
        "      <td> 8</td>\n",
        "      <td> 10</td>\n",
        "      <td> 5776</td>\n",
        "      <td>    CREM</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>43</th>\n",
        "      <td>    SPI1</td>\n",
        "      <td>    SPI1</td>\n",
        "      <td> 6</td>\n",
        "      <td> 10</td>\n",
        "      <td> 8273</td>\n",
        "      <td>    SPI1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>8 </th>\n",
        "      <td>     MYC</td>\n",
        "      <td>     MYC</td>\n",
        "      <td> 6</td>\n",
        "      <td> 10</td>\n",
        "      <td> 9863</td>\n",
        "      <td>     MYC</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>12</th>\n",
        "      <td>    FLI1</td>\n",
        "      <td>    FLI1</td>\n",
        "      <td> 6</td>\n",
        "      <td> 10</td>\n",
        "      <td> 7063</td>\n",
        "      <td>    FLI1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>23</th>\n",
        "      <td>   KDM5B</td>\n",
        "      <td>   KDM5B</td>\n",
        "      <td> 5</td>\n",
        "      <td> 10</td>\n",
        "      <td> 3724</td>\n",
        "      <td>   KDM5B</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>22</th>\n",
        "      <td>   RUNX1</td>\n",
        "      <td>   RUNX1</td>\n",
        "      <td> 5</td>\n",
        "      <td> 10</td>\n",
        "      <td> 7624</td>\n",
        "      <td>   RUNX1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>84</th>\n",
        "      <td> SMARCA4</td>\n",
        "      <td> SMARCA4</td>\n",
        "      <td> 5</td>\n",
        "      <td> 10</td>\n",
        "      <td> 3481</td>\n",
        "      <td> SMARCA4</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>47</th>\n",
        "      <td>   PPARG</td>\n",
        "      <td>   PPARG</td>\n",
        "      <td> 5</td>\n",
        "      <td> 10</td>\n",
        "      <td> 4976</td>\n",
        "      <td>   PPARG</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>44</th>\n",
        "      <td>   NANOG</td>\n",
        "      <td>   NANOG</td>\n",
        "      <td> 4</td>\n",
        "      <td> 10</td>\n",
        "      <td> 8386</td>\n",
        "      <td>   NANOG</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 422,
       "text": [
        "         tf     tf_x  count  total_genes  chea_target_count     tf_y\n",
        "13    CREB1    CREB1      9           10               4907    CREB1\n",
        "54     CREM     CREM      8           10               5776     CREM\n",
        "43     SPI1     SPI1      6           10               8273     SPI1\n",
        "8       MYC      MYC      6           10               9863      MYC\n",
        "12     FLI1     FLI1      6           10               7063     FLI1\n",
        "23    KDM5B    KDM5B      5           10               3724    KDM5B\n",
        "22    RUNX1    RUNX1      5           10               7624    RUNX1\n",
        "84  SMARCA4  SMARCA4      5           10               3481  SMARCA4\n",
        "47    PPARG    PPARG      5           10               4976    PPARG\n",
        "44    NANOG    NANOG      4           10               8386    NANOG"
       ]
      }
     ],
     "prompt_number": 422
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len(z)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 409,
       "text": [
        "1871"
       ]
      }
     ],
     "prompt_number": 409
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "tf_target_counts = []\n",
      "for (k, g) in chea.groupby(\"tf\"):\n",
      "    tf_target_counts.append( { \"tf\" : k, \"chea_target_count\" : len( uniqueSym( g.target )) } )\n",
      "tf_target_counts = pd.DataFrame.from_dict(tf_target_counts)\n",
      "tf_target_counts.index = tf_target_counts.tf"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 378
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "mm9_motif_gene_counts = []\n",
      "for (k, g) in mm9_gene_motifs.groupby(\"motif_name\"):\n",
      "    mm9_motif_gene_counts.append( { \"motif_name\" : k, \"motif_gene_count\" : len(g.genename.unique()) } )\n",
      "mm9_motif_gene_counts = pd.DataFrame.from_dict( mm9_motif_gene_counts )\n",
      "mm9_motif_gene_counts.index = mm9_motif_gene_counts.motif_name\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 507
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "pd.set_printoptions(max_columns=100, max_rows=100)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "/data/adrian/anaconda/lib/python2.7/site-packages/pandas/core/format.py:1612: FutureWarning: set_printoptions is deprecated, use set_option instead\n",
        "  FutureWarning)\n"
       ]
      }
     ],
     "prompt_number": 513
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "IPython.display.HTML( mm9_motif_gene_counts.sort_index(by=\"motif_gene_count\", ascending=False).to_html() )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>motif_gene_count</th>\n",
        "      <th>motif_name</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>motif_name</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>SP1.p2</th>\n",
        "      <td> 12044</td>\n",
        "      <td>                            SP1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>KLF4.p3</th>\n",
        "      <td> 10899</td>\n",
        "      <td>                           KLF4.p3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MAZ.p2</th>\n",
        "      <td> 10878</td>\n",
        "      <td>                            MAZ.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PATZ1.p2</th>\n",
        "      <td> 10334</td>\n",
        "      <td>                          PATZ1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MZF1.p2</th>\n",
        "      <td>  8515</td>\n",
        "      <td>                           MZF1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TFAP2{A,C}.p2</th>\n",
        "      <td>  8076</td>\n",
        "      <td>                     TFAP2{A,C}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TFDP1.p2</th>\n",
        "      <td>  8067</td>\n",
        "      <td>                          TFDP1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>GTF2I.p2</th>\n",
        "      <td>  6874</td>\n",
        "      <td>                          GTF2I.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TFAP2B.p2</th>\n",
        "      <td>  6217</td>\n",
        "      <td>                         TFAP2B.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PAX5.p2</th>\n",
        "      <td>  5810</td>\n",
        "      <td>                           PAX5.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ELF1,2,4.p2</th>\n",
        "      <td>  5707</td>\n",
        "      <td>                       ELF1,2,4.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SPI1.p2</th>\n",
        "      <td>  5555</td>\n",
        "      <td>                           SPI1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>YY1.p2</th>\n",
        "      <td>  5456</td>\n",
        "      <td>                            YY1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TCF4_dimer.p2</th>\n",
        "      <td>  4716</td>\n",
        "      <td>                     TCF4_dimer.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>RREB1.p2</th>\n",
        "      <td>  4649</td>\n",
        "      <td>                          RREB1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ELK1,4_GABP{A,B1}.p3</th>\n",
        "      <td>  4645</td>\n",
        "      <td>              ELK1,4_GABP{A,B1}.p3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HIC1.p2</th>\n",
        "      <td>  4518</td>\n",
        "      <td>                           HIC1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HOX{A6,A7,B6,B7}.p2</th>\n",
        "      <td>  4193</td>\n",
        "      <td>               HOX{A6,A7,B6,B7}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FEV.p2</th>\n",
        "      <td>  4192</td>\n",
        "      <td>                            FEV.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>BPTF.p2</th>\n",
        "      <td>  4175</td>\n",
        "      <td>                           BPTF.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SPIB.p2</th>\n",
        "      <td>  3935</td>\n",
        "      <td>                           SPIB.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SNAI1..3.p2</th>\n",
        "      <td>  3921</td>\n",
        "      <td>                       SNAI1..3.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>EGR1..3.p2</th>\n",
        "      <td>  3885</td>\n",
        "      <td>                        EGR1..3.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOXD3.p2</th>\n",
        "      <td>  3629</td>\n",
        "      <td>                          FOXD3.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MAFB.p2</th>\n",
        "      <td>  3359</td>\n",
        "      <td>                           MAFB.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ZFP161.p2</th>\n",
        "      <td>  3321</td>\n",
        "      <td>                         ZFP161.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>GATA1..3.p2</th>\n",
        "      <td>  3297</td>\n",
        "      <td>                       GATA1..3.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MTF1.p2</th>\n",
        "      <td>  3146</td>\n",
        "      <td>                           MTF1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>EHF.p2</th>\n",
        "      <td>  3024</td>\n",
        "      <td>                            EHF.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NRF1.p2</th>\n",
        "      <td>  3000</td>\n",
        "      <td>                           NRF1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ZNF148.p2</th>\n",
        "      <td>  2982</td>\n",
        "      <td>                         ZNF148.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PRRX1,2.p2</th>\n",
        "      <td>  2894</td>\n",
        "      <td>                        PRRX1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NHLH1,2.p2</th>\n",
        "      <td>  2886</td>\n",
        "      <td>                        NHLH1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NFY{A,B,C}.p2</th>\n",
        "      <td>  2875</td>\n",
        "      <td>                     NFY{A,B,C}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOX{I1,J2}.p2</th>\n",
        "      <td>  2722</td>\n",
        "      <td>                     FOX{I1,J2}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MYFfamily.p2</th>\n",
        "      <td>  2672</td>\n",
        "      <td>                      MYFfamily.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>bHLH_family.p2</th>\n",
        "      <td>  2529</td>\n",
        "      <td>                    bHLH_family.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ETS1,2.p2</th>\n",
        "      <td>  2463</td>\n",
        "      <td>                         ETS1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>KLF12.p2</th>\n",
        "      <td>  2342</td>\n",
        "      <td>                          KLF12.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NFE2L1.p2</th>\n",
        "      <td>  2329</td>\n",
        "      <td>                         NFE2L1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ZEB1.p2</th>\n",
        "      <td>  2274</td>\n",
        "      <td>                           ZEB1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>AHR_ARNT_ARNT2.p2</th>\n",
        "      <td>  1983</td>\n",
        "      <td>                 AHR_ARNT_ARNT2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ATF5_CREB3.p2</th>\n",
        "      <td>  1951</td>\n",
        "      <td>                     ATF5_CREB3.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>GFI1.p2</th>\n",
        "      <td>  1832</td>\n",
        "      <td>                           GFI1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TFCP2.p2</th>\n",
        "      <td>  1777</td>\n",
        "      <td>                          TFCP2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>CDX1,2,4.p2</th>\n",
        "      <td>  1764</td>\n",
        "      <td>                       CDX1,2,4.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SPZ1.p2</th>\n",
        "      <td>  1750</td>\n",
        "      <td>                           SPZ1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>LMO2.p2</th>\n",
        "      <td>  1701</td>\n",
        "      <td>                           LMO2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HBP1_HMGB_SSRP1_UBTF.p2</th>\n",
        "      <td>  1696</td>\n",
        "      <td>           HBP1_HMGB_SSRP1_UBTF.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PAX2.p2</th>\n",
        "      <td>  1690</td>\n",
        "      <td>                           PAX2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ZNF143.p2</th>\n",
        "      <td>  1642</td>\n",
        "      <td>                         ZNF143.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>EP300.p2</th>\n",
        "      <td>  1508</td>\n",
        "      <td>                          EP300.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NR5A1,2.p2</th>\n",
        "      <td>  1493</td>\n",
        "      <td>                        NR5A1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAT2,4,6.p2</th>\n",
        "      <td>  1484</td>\n",
        "      <td>                      STAT2,4,6.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SOX17.p2</th>\n",
        "      <td>  1483</td>\n",
        "      <td>                          SOX17.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>JUN.p2</th>\n",
        "      <td>  1473</td>\n",
        "      <td>                            JUN.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ZBTB16.p2</th>\n",
        "      <td>  1466</td>\n",
        "      <td>                         ZBTB16.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HES1.p2</th>\n",
        "      <td>  1458</td>\n",
        "      <td>                           HES1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SOX2.p2</th>\n",
        "      <td>  1378</td>\n",
        "      <td>                           SOX2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ARNT_ARNT2_BHLHB2_MAX_MYC_USF1.p2</th>\n",
        "      <td>  1356</td>\n",
        "      <td> ARNT_ARNT2_BHLHB2_MAX_MYC_USF1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MYOD1.p2</th>\n",
        "      <td>  1312</td>\n",
        "      <td>                          MYOD1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>IKZF2.p2</th>\n",
        "      <td>  1230</td>\n",
        "      <td>                          IKZF2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SREBF1,2.p2</th>\n",
        "      <td>  1184</td>\n",
        "      <td>                       SREBF1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>CREB1.p2</th>\n",
        "      <td>  1125</td>\n",
        "      <td>                          CREB1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SOX{8,9,10}.p2</th>\n",
        "      <td>  1077</td>\n",
        "      <td>                    SOX{8,9,10}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HNF4A_NR2F1,2.p2</th>\n",
        "      <td>  1072</td>\n",
        "      <td>                  HNF4A_NR2F1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ZBTB6.p2</th>\n",
        "      <td>  1063</td>\n",
        "      <td>                          ZBTB6.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PITX1..3.p2</th>\n",
        "      <td>  1032</td>\n",
        "      <td>                       PITX1..3.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRY.p2</th>\n",
        "      <td>   989</td>\n",
        "      <td>                            SRY.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ATF4.p2</th>\n",
        "      <td>   959</td>\n",
        "      <td>                           ATF4.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TFAP4.p2</th>\n",
        "      <td>   896</td>\n",
        "      <td>                          TFAP4.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>POU5F1.p2</th>\n",
        "      <td>   873</td>\n",
        "      <td>                         POU5F1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>RXR{A,B,G}.p2</th>\n",
        "      <td>   866</td>\n",
        "      <td>                     RXR{A,B,G}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ATF6.p2</th>\n",
        "      <td>   862</td>\n",
        "      <td>                           ATF6.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TFEB.p2</th>\n",
        "      <td>   861</td>\n",
        "      <td>                           TFEB.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HMGA1,2.p2</th>\n",
        "      <td>   849</td>\n",
        "      <td>                        HMGA1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOXO1,3,4.p2</th>\n",
        "      <td>   843</td>\n",
        "      <td>                      FOXO1,3,4.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NKX3-2.p2</th>\n",
        "      <td>   827</td>\n",
        "      <td>                         NKX3-2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HIF1A.p2</th>\n",
        "      <td>   821</td>\n",
        "      <td>                          HIF1A.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>RFX1..5_RFXANK_RFXAP.p2</th>\n",
        "      <td>   792</td>\n",
        "      <td>           RFX1..5_RFXANK_RFXAP.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOS_FOS{B,L1}_JUN{B,D}.p2</th>\n",
        "      <td>   733</td>\n",
        "      <td>         FOS_FOS{B,L1}_JUN{B,D}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MYB.p2</th>\n",
        "      <td>   720</td>\n",
        "      <td>                            MYB.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TBP.p2</th>\n",
        "      <td>   696</td>\n",
        "      <td>                            TBP.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>CTCF.p2</th>\n",
        "      <td>   666</td>\n",
        "      <td>                           CTCF.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ATF2.p2</th>\n",
        "      <td>   651</td>\n",
        "      <td>                           ATF2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SOX5.p2</th>\n",
        "      <td>   641</td>\n",
        "      <td>                           SOX5.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TLX2.p2</th>\n",
        "      <td>   637</td>\n",
        "      <td>                           TLX2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NFATC1..3.p2</th>\n",
        "      <td>   616</td>\n",
        "      <td>                      NFATC1..3.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ZIC1..3.p2</th>\n",
        "      <td>   610</td>\n",
        "      <td>                        ZIC1..3.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ADNP_IRX_SIX_ZHX.p2</th>\n",
        "      <td>   595</td>\n",
        "      <td>               ADNP_IRX_SIX_ZHX.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HMX1.p2</th>\n",
        "      <td>   566</td>\n",
        "      <td>                           HMX1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>DBP.p2</th>\n",
        "      <td>   564</td>\n",
        "      <td>                            DBP.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ZNF384.p2</th>\n",
        "      <td>   554</td>\n",
        "      <td>                         ZNF384.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEF2{A,B,C,D}.p2</th>\n",
        "      <td>   548</td>\n",
        "      <td>                  MEF2{A,B,C,D}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NFKB1_REL_RELA.p2</th>\n",
        "      <td>   527</td>\n",
        "      <td>                 NFKB1_REL_RELA.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>GLI1..3.p2</th>\n",
        "      <td>   510</td>\n",
        "      <td>                        GLI1..3.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>EBF1.p2</th>\n",
        "      <td>   482</td>\n",
        "      <td>                           EBF1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>E2F1..5.p2</th>\n",
        "      <td>   480</td>\n",
        "      <td>                        E2F1..5.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SMAD1..7,9.p2</th>\n",
        "      <td>   466</td>\n",
        "      <td>                     SMAD1..7,9.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>AIRE.p2</th>\n",
        "      <td>   464</td>\n",
        "      <td>                           AIRE.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>EVI1.p2</th>\n",
        "      <td>   458</td>\n",
        "      <td>                           EVI1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOX{D1,D2}.p2</th>\n",
        "      <td>   456</td>\n",
        "      <td>                     FOX{D1,D2}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOXQ1.p2</th>\n",
        "      <td>   442</td>\n",
        "      <td>                          FOXQ1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>RUNX1..3.p2</th>\n",
        "      <td>   437</td>\n",
        "      <td>                       RUNX1..3.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NFE2.p2</th>\n",
        "      <td>   436</td>\n",
        "      <td>                           NFE2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>LEF1_TCF7_TCF7L1,2.p2</th>\n",
        "      <td>   431</td>\n",
        "      <td>             LEF1_TCF7_TCF7L1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOSL2.p2</th>\n",
        "      <td>   425</td>\n",
        "      <td>                          FOSL2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>GTF2A1,2.p2</th>\n",
        "      <td>   424</td>\n",
        "      <td>                       GTF2A1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TGIF1.p2</th>\n",
        "      <td>   411</td>\n",
        "      <td>                          TGIF1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PAX4.p2</th>\n",
        "      <td>   407</td>\n",
        "      <td>                           PAX4.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOXA2.p3</th>\n",
        "      <td>   388</td>\n",
        "      <td>                          FOXA2.p3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ESRRA.p2</th>\n",
        "      <td>   387</td>\n",
        "      <td>                          ESRRA.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TBX4,5.p2</th>\n",
        "      <td>   371</td>\n",
        "      <td>                         TBX4,5.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOXL1.p2</th>\n",
        "      <td>   367</td>\n",
        "      <td>                          FOXL1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ESR1.p2</th>\n",
        "      <td>   356</td>\n",
        "      <td>                           ESR1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>IRF1,2,7.p3</th>\n",
        "      <td>   347</td>\n",
        "      <td>                       IRF1,2,7.p3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HSF1,2.p2</th>\n",
        "      <td>   318</td>\n",
        "      <td>                         HSF1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOXN1.p2</th>\n",
        "      <td>   309</td>\n",
        "      <td>                          FOXN1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>POU2F1..3.p2</th>\n",
        "      <td>   292</td>\n",
        "      <td>                      POU2F1..3.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HOX{A4,D4}.p2</th>\n",
        "      <td>   290</td>\n",
        "      <td>                     HOX{A4,D4}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>BACH2.p2</th>\n",
        "      <td>   282</td>\n",
        "      <td>                          BACH2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOX{F1,F2,J1}.p2</th>\n",
        "      <td>   279</td>\n",
        "      <td>                  FOX{F1,F2,J1}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NKX6-1,2.p2</th>\n",
        "      <td>   273</td>\n",
        "      <td>                       NKX6-1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TLX1..3_NFIC{dimer}.p2</th>\n",
        "      <td>   269</td>\n",
        "      <td>            TLX1..3_NFIC{dimer}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TEAD1.p2</th>\n",
        "      <td>   260</td>\n",
        "      <td>                          TEAD1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NFE2L2.p2</th>\n",
        "      <td>   252</td>\n",
        "      <td>                         NFE2L2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ONECUT1,2.p2</th>\n",
        "      <td>   237</td>\n",
        "      <td>                      ONECUT1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>EWSR1-FLI1.p2</th>\n",
        "      <td>   236</td>\n",
        "      <td>                     EWSR1-FLI1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>RBPJ.p2</th>\n",
        "      <td>   230</td>\n",
        "      <td>                           RBPJ.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NKX2-3_NKX2-5.p2</th>\n",
        "      <td>   217</td>\n",
        "      <td>                  NKX2-3_NKX2-5.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PBX1.p2</th>\n",
        "      <td>   216</td>\n",
        "      <td>                           PBX1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>LHX3,4.p2</th>\n",
        "      <td>   213</td>\n",
        "      <td>                         LHX3,4.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>CRX.p2</th>\n",
        "      <td>   207</td>\n",
        "      <td>                            CRX.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>GATA6.p2</th>\n",
        "      <td>   207</td>\n",
        "      <td>                          GATA6.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>IKZF1.p2</th>\n",
        "      <td>   204</td>\n",
        "      <td>                          IKZF1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>POU1F1.p2</th>\n",
        "      <td>   197</td>\n",
        "      <td>                         POU1F1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ZNF238.p2</th>\n",
        "      <td>   196</td>\n",
        "      <td>                         ZNF238.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>POU5F1_SOX2{dimer}.p2</th>\n",
        "      <td>   185</td>\n",
        "      <td>             POU5F1_SOX2{dimer}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HNF1A.p2</th>\n",
        "      <td>   172</td>\n",
        "      <td>                          HNF1A.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>DMAP1_NCOR{1,2}_SMARC.p2</th>\n",
        "      <td>   159</td>\n",
        "      <td>          DMAP1_NCOR{1,2}_SMARC.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NR6A1.p2</th>\n",
        "      <td>   158</td>\n",
        "      <td>                          NR6A1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NR1H4.p2</th>\n",
        "      <td>   151</td>\n",
        "      <td>                          NR1H4.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>EOMES.p2</th>\n",
        "      <td>   148</td>\n",
        "      <td>                          EOMES.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NKX2-2,8.p2</th>\n",
        "      <td>   147</td>\n",
        "      <td>                       NKX2-2,8.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>XBP1.p3</th>\n",
        "      <td>   139</td>\n",
        "      <td>                           XBP1.p3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>RXRG_dimer.p3</th>\n",
        "      <td>   128</td>\n",
        "      <td>                     RXRG_dimer.p3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NFIX.p2</th>\n",
        "      <td>   127</td>\n",
        "      <td>                           NFIX.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PRDM1.p3</th>\n",
        "      <td>   123</td>\n",
        "      <td>                          PRDM1.p3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ALX1.p2</th>\n",
        "      <td>   122</td>\n",
        "      <td>                           ALX1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HLF.p2</th>\n",
        "      <td>   118</td>\n",
        "      <td>                            HLF.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TP53.p2</th>\n",
        "      <td>   117</td>\n",
        "      <td>                           TP53.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ZNF423.p2</th>\n",
        "      <td>   115</td>\n",
        "      <td>                         ZNF423.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MYBL2.p2</th>\n",
        "      <td>   114</td>\n",
        "      <td>                          MYBL2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NFIL3.p2</th>\n",
        "      <td>   111</td>\n",
        "      <td>                          NFIL3.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRF.p3</th>\n",
        "      <td>   110</td>\n",
        "      <td>                            SRF.p3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NKX3-1.p2</th>\n",
        "      <td>   109</td>\n",
        "      <td>                         NKX3-1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAT5{A,B}.p2</th>\n",
        "      <td>   107</td>\n",
        "      <td>                     STAT5{A,B}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NANOG.p2</th>\n",
        "      <td>   101</td>\n",
        "      <td>                          NANOG.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>CDC5L.p2</th>\n",
        "      <td>    92</td>\n",
        "      <td>                          CDC5L.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>GFI1B.p2</th>\n",
        "      <td>    82</td>\n",
        "      <td>                          GFI1B.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>EN1,2.p2</th>\n",
        "      <td>    79</td>\n",
        "      <td>                          EN1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HOX{A5,B5}.p2</th>\n",
        "      <td>    79</td>\n",
        "      <td>                     HOX{A5,B5}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>REST.p3</th>\n",
        "      <td>    78</td>\n",
        "      <td>                           REST.p3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>T.p2</th>\n",
        "      <td>    75</td>\n",
        "      <td>                              T.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HAND1,2.p2</th>\n",
        "      <td>    69</td>\n",
        "      <td>                        HAND1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>POU6F1.p2</th>\n",
        "      <td>    62</td>\n",
        "      <td>                         POU6F1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>CEBPA,B_DDIT3.p2</th>\n",
        "      <td>    59</td>\n",
        "      <td>                  CEBPA,B_DDIT3.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NR3C1.p2</th>\n",
        "      <td>    50</td>\n",
        "      <td>                          NR3C1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAT1,3.p3</th>\n",
        "      <td>    49</td>\n",
        "      <td>                        STAT1,3.p3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TAL1_TCF{3,4,12}.p2</th>\n",
        "      <td>    44</td>\n",
        "      <td>               TAL1_TCF{3,4,12}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>RORA.p2</th>\n",
        "      <td>    38</td>\n",
        "      <td>                           RORA.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HOXA9_MEIS1.p2</th>\n",
        "      <td>    38</td>\n",
        "      <td>                    HOXA9_MEIS1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>RXRA_VDR{dimer}.p2</th>\n",
        "      <td>    37</td>\n",
        "      <td>                RXRA_VDR{dimer}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>POU3F1..4.p2</th>\n",
        "      <td>    31</td>\n",
        "      <td>                      POU3F1..4.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOXP3.p2</th>\n",
        "      <td>    26</td>\n",
        "      <td>                          FOXP3.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ARID5B.p2</th>\n",
        "      <td>    25</td>\n",
        "      <td>                         ARID5B.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOX{C1,C2}.p2</th>\n",
        "      <td>    23</td>\n",
        "      <td>                     FOX{C1,C2}.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NKX2-1,4.p2</th>\n",
        "      <td>    21</td>\n",
        "      <td>                       NKX2-1,4.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>CUX2.p2</th>\n",
        "      <td>    15</td>\n",
        "      <td>                           CUX2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>AR.p2</th>\n",
        "      <td>    14</td>\n",
        "      <td>                             AR.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>VSX1,2.p2</th>\n",
        "      <td>    13</td>\n",
        "      <td>                         VSX1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PPARG.p2</th>\n",
        "      <td>    13</td>\n",
        "      <td>                          PPARG.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PAX6.p2</th>\n",
        "      <td>    11</td>\n",
        "      <td>                           PAX6.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MSX1,2.p2</th>\n",
        "      <td>     9</td>\n",
        "      <td>                         MSX1,2.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>GZF1.p2</th>\n",
        "      <td>     9</td>\n",
        "      <td>                           GZF1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PDX1.p2</th>\n",
        "      <td>     7</td>\n",
        "      <td>                           PDX1.p2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PAX3,7.p2</th>\n",
        "      <td>     1</td>\n",
        "      <td>                         PAX3,7.p2</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 520,
       "text": [
        "<IPython.core.display.HTML at 0x163e8da10>"
       ]
      }
     ],
     "prompt_number": 520
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "tf_target_counts.ix[\"CREM\", :]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 382,
       "text": [
        "chea_target_count    5776\n",
        "tf                   CREM\n",
        "Name: CREM, dtype: object"
       ]
      }
     ],
     "prompt_number": 382
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "chea.target.unique()[3000:3500]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 361,
       "text": [
        "array([u'SLC35A2', u'SLC35C2', u'SLC37A2', u'SLC37A4', u'SLC38A1',\n",
        "       u'SLC38A2', u'SLC38A5', u'SLC3A2', u'SLC41A2', u'SLC43A1',\n",
        "       u'SLC6A13', u'SLC6A14', u'SLC6A4', u'SLC6A5', u'SLC7A8', u'SLC8A3',\n",
        "       u'SLC9A6', u'SLITRK4', u'SLITRK5', u'SMAD4', u'SMAD9', u'SMARCD2',\n",
        "       u'SMC1L1', u'SMOX', u'SMPD3', u'SMPX', u'SMS', u'SMTN', u'SNAP25',\n",
        "       u'SNAP29', u'SNAPC1', u'SNF1LK2', u'SNRP70', u'SNRPF', u'SNRPN',\n",
        "       u'SNW1', u'SNX12', u'SNX19', u'SOX5', u'SP1', u'SPAG7', u'SPAG9',\n",
        "       u'SPATA13', u'SPATA20', u'SPG7', u'SPIRE1', u'SPRY3', u'SPTLC2',\n",
        "       u'SQRDL', u'SRP14', u'SRPR', u'SRXN1', u'SSH1', u'SSH2', u'SSRP1',\n",
        "       u'ST5', u'ST8SIA2', u'STAB2', u'STAC3', u'STARD13', u'STARD5',\n",
        "       u'STARD6', u'STAT2', u'STK24', u'STK33', u'STK6', u'STRA13',\n",
        "       u'STRN4', u'STX10', u'STX16', u'STX5A', u'SUHW2', u'SULF2',\n",
        "       u'SUPT4H1', u'SUZ12P', u'SV2B', u'SYBL1', u'SYCP2', u'SYNE2',\n",
        "       u'SYNJ1', u'SYT13', u'SYT16', u'SYTL2', u'SYTL4', u'TAF1', u'TAF10',\n",
        "       u'TANC2', u'TAOK2', u'TARSL2', u'TBC1D15', u'TBC1D21', u'TBCD',\n",
        "       u'TBX3', u'TBX5', u'TCEAL1', u'TCEAL8', u'TCF15', u'TCF2', u'TCF20',\n",
        "       u'TCL1B', u'TDG', u'TDRD9', u'TEAD2', u'TEAD4', u'TEKT3', u'TEP1',\n",
        "       u'TEX101', u'TEX2', u'TEX9', u'TGFB3', u'TGIF', u'TGM1', u'TGM2',\n",
        "       u'TGM3', u'TGM5', u'TGM7', u'THAP10', u'THBS1', u'THRAP5', u'TIAF1',\n",
        "       u'TIAM1', u'TIMELESS', u'TIMM17B', u'TIMM22', u'TIMM8B', u'TIMM9',\n",
        "       u'TINF2', u'TK1', u'TK2', u'TLCD1', u'TLR7', u'TM7SF3', u'TM9SF1',\n",
        "       u'TM9SF2', u'TMC3', u'TMCC3', u'TMCO5', u'TMED8', u'TMEM100',\n",
        "       u'TMEM102', u'TMEM107', u'TMEM16H', u'TMEM29', u'TMEM31', u'TMEM5',\n",
        "       u'TMEM83', u'TMEM88', u'TMEM9B', u'TMPRSS3', u'TMSL8', u'TNFRSF19',\n",
        "       u'TNFRSF1A', u'TNPO2', u'TNRC6C', u'TNRC9', u'TOB1', u'TOP3A',\n",
        "       u'TP53', u'TPM1', u'TPP2', u'TRA1', u'TRA@', u'TRAPPC2',\n",
        "       u'TRAPPC6A', u'TRAV20', u'TRHDE', u'TRIM3', u'TRIM37', u'TRIM68',\n",
        "       u'TRIP11', u'TRMT5', u'TRO', u'TRPC4', u'TRPC5', u'TRPM1', u'TRPM7',\n",
        "       u'TSC22D1', u'TSFM', u'TSG101', u'TSHR', u'TSPAN3', u'TSPYL2',\n",
        "       u'TSR1', u'TSSC4', u'TSSK4', u'TTC17', u'TTC23', u'TUB', u'TUBA8',\n",
        "       u'TUBGCP5', u'TUBGCP6', u'TUFM', u'TULP3', u'TXN2', u'TXNDC13',\n",
        "       u'TXNDC2', u'TXNL4B', u'TXNRD1', u'U2AF1L2', u'U2AF1L4', u'UACA',\n",
        "       u'UBB', u'UBC', u'UBE1', u'UBE2C', u'UBE2E4P', u'UBE2S', u'UBE2Z',\n",
        "       u'UBL3', u'UBL7', u'UCHL3', u'UFD1L', u'UGCGL2', u'UMOD', u'UMODL1',\n",
        "       u'UPF3B', u'UQCRC2', u'USH1C', u'USP12', u'USP14', u'USP18',\n",
        "       u'USP30', u'USP31', u'USP36', u'USP43', u'USP51', u'USP52',\n",
        "       u'USPL1', u'UTX', u'UXT', u'VAPA', u'VDR', u'VKORC1', u'VPS29',\n",
        "       u'VRK1', u'VRK3', u'VSIG1', u'VSIG2', u'VSIG4', u'VWF', u'WARS',\n",
        "       u'WBP11', u'WDR22', u'WDR44', u'WDR59', u'WDR68', u'WFDC6', u'WIBG',\n",
        "       u'WIPI1', u'WNK1', u'WNK3', u'WNT11', u'WNT5B', u'WSB2', u'WWOX',\n",
        "       u'WWP2', u'XBP1', u'XLKD1', u'XRCC1', u'XRN2', u'XRRA1', u'YES1',\n",
        "       u'YJDC', u'YLPM1', u'YWHAH', u'YY2', u'ZADH1', u'ZC3H12C',\n",
        "       u'ZCCHC14', u'ZCCHC2', u'ZDHHC13', u'ZFHX2', u'ZFP1', u'ZFP36L1',\n",
        "       u'ZIM2', u'ZMAT1', u'ZMAT5', u'ZMYM3', u'ZNF132', u'ZNF133',\n",
        "       u'ZNF135', u'ZNF140', u'ZNF154', u'ZNF175', u'ZNF18', u'ZNF180',\n",
        "       u'ZNF185', u'ZNF19', u'ZNF20', u'ZNF21', u'ZNF215', u'ZNF218',\n",
        "       u'ZNF224', u'ZNF226', u'ZNF227', u'ZNF23', u'ZNF236', u'ZNF24',\n",
        "       u'ZNF264', u'ZNF28', u'ZNF287', u'ZNF331', u'ZNF334', u'ZNF335',\n",
        "       u'ZNF341', u'ZNF345', u'ZNF384', u'ZNF397', u'ZNF41', u'ZNF417',\n",
        "       u'ZNF420', u'ZNF423', u'ZNF45', u'ZNF516', u'ZNF521', u'ZNF527',\n",
        "       u'ZNF528', u'ZNF530', u'ZNF537', u'ZNF548', u'ZNF551', u'ZNF553',\n",
        "       u'ZNF558', u'ZNF567', u'ZNF569', u'ZNF574', u'ZNF577', u'ZNF585A',\n",
        "       u'ZNF587', u'ZNF609', u'ZNF610', u'ZNF645', u'ZNF667', u'ZNF668',\n",
        "       u'ZNF673', u'ZNF690', u'ZNF75', u'ZNF81', u'ZNF83', u'ZNFN1A4',\n",
        "       u'ZNRF3', u'ZSCAN2', u'ZSCAN5', u'ZZEF1', u'1110019L22RIK',\n",
        "       u'1190017O12RIK', u'1200009O22RIK', u'1300013J15RIK',\n",
        "       u'1300014I06RIK', u'1600015H20RIK', u'1810031K17RIK',\n",
        "       u'2010001J22RIK', u'2010012C16RIK', u'2310015N21RIK',\n",
        "       u'2310045A20RIK', u'2410080H04RIK', u'2600010E01RIK',\n",
        "       u'2610017I09RIK', u'2610024A01RIK', u'2810022L02RIK',\n",
        "       u'2810451A06RIK', u'2900005J15RIK', u'2900093B09RIK',\n",
        "       u'3732412D22RIK', u'4631426J05RIK', u'4732486J07RIK',\n",
        "       u'4831416G18RIK', u'4921515A04RIK', u'4930447C04RIK',\n",
        "       u'4930511M11RIK', u'4930577M16RIK', u'4931406I20RIK',\n",
        "       u'4933407N01RIK', u'4933408F15', u'4933436C20RIK', u'5330439J01RIK',\n",
        "       u'5730446D14RIK', u'5730467H21RIK', u'5730596B20RIK', u'6030405A18',\n",
        "       u'6330514A18RIK', u'6332401O19RIK', u'6530402A20', u'9030611O19RIK',\n",
        "       u'9030612E09RIK', u'9030623N16RIK', u'9130213B05RIK', u'9430022A14',\n",
        "       u'9430023B20RIK', u'9430076C15RIK', u'9430077A04RIK',\n",
        "       u'9530027K23RIK', u'9930014A18RIK', u'A030009H04RIK',\n",
        "       u'A330008L17RIK', u'A630014H24RIK', u'A830053O21RIK',\n",
        "       u'A930008A22RIK', u'A930008G19RIK', u'A930038C07RIK',\n",
        "       u'A930040G15RIK', u'A930041G11RIK', u'AB041550', u'ABCA5', u'ABCF2',\n",
        "       u'ABHD3', u'ADAMTS8', u'ADCY8', u'ADCYAP1R1', u'ADH7', u'ADM',\n",
        "       u'ADRA1A', u'ADRA1D', u'ADRA2B', u'ADRB1', u'AI314180', u'AI849156',\n",
        "       u'AKT1S1', u'ALDH1A2', u'ALX3', u'AR', u'ARG2', u'ASB13', u'ASB3',\n",
        "       u'ASTN2', u'ATF3', u'ATOH1', u'ATOH8', u'ATP7B', u'AVPR1A',\n",
        "       u'AW125753', u'AW539457', u'B130055L09RIK', u'B230343H07RIK',\n",
        "       u'B230380D07RIK', u'B3GAT2', u'B830017H08RIK', u'BACH2', u'BARHL1',\n",
        "       u'BARX1', u'BARX2', u'BC014699', u'BC034054', u'BC038286',\n",
        "       u'BC055811', u'BC061194', u'BCAN', u'BCAS2', u'BCL2L11', u'BCL6B',\n",
        "       u'BHLHB3', u'BHLHB4', u'BHLHB8', u'BMP3', u'BMP6', u'BMP7',\n",
        "       u'BMP8A', u'BOC', u'BRUNOL4', u'C530028O21RIK', u'CACNA1E',\n",
        "       u'CACNG4', u'CALCA', u'CALN1', u'CAMK1D', u'CAMKK1', u'CAR10',\n",
        "       u'CAR11', u'CAR4', u'CARD10', u'CBLN2', u'CBLN4', u'CBX4', u'CCKBR',\n",
        "       u'CCND2', u'CCR9', u'CD14', u'CD200R3', u'CD24A', u'CDGAP',\n",
        "       u'CDH11', u'CDH22', u'CDH4', u'CDH8', u'CDK5R2', u'CDKN1C',\n",
        "       u'CDKN2A', u'CDKN2C', u'CDX2', u'CEBPA', u'CH25H', u'CHRD',\n",
        "       u'CHST11', u'CHST7', u'CHX10', u'CLDN23', u'CLSTN1', u'CNKSR2',\n",
        "       u'CNR1', u'CNTFR'], dtype=object)"
       ]
      }
     ],
     "prompt_number": 361
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len( chea.target.unique() )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 357,
       "text": [
        "47510"
       ]
      }
     ],
     "prompt_number": 357
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len( chea.select(lambda x: chea.ix[x, \"tf\"] == \"CREM\").target.unique() )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 356,
       "text": [
        "5776"
       ]
      }
     ],
     "prompt_number": 356
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len( uniqueSym( cp73_chronicHigh_up.symbol ))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 364,
       "text": [
        "1871"
       ]
      }
     ],
     "prompt_number": 364
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "test[0:50]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>tf</th>\n",
        "      <th>count</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>189</th>\n",
        "      <td>    (CREM,)</td>\n",
        "      <td> 868</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>24 </th>\n",
        "      <td>     (MYC,)</td>\n",
        "      <td> 860</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>180</th>\n",
        "      <td>    (SPI1,)</td>\n",
        "      <td> 780</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>54 </th>\n",
        "      <td>    (SOX2,)</td>\n",
        "      <td> 778</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>185</th>\n",
        "      <td>   (RUNX1,)</td>\n",
        "      <td> 761</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>51 </th>\n",
        "      <td>    (FLI1,)</td>\n",
        "      <td> 725</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>111</th>\n",
        "      <td>   (CREB1,)</td>\n",
        "      <td> 701</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>19 </th>\n",
        "      <td>    (EGR1,)</td>\n",
        "      <td> 653</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>4  </th>\n",
        "      <td>   (NANOG,)</td>\n",
        "      <td> 641</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>172</th>\n",
        "      <td>   (HNF4A,)</td>\n",
        "      <td> 625</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>31 </th>\n",
        "      <td>   (STAT3,)</td>\n",
        "      <td> 617</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>135</th>\n",
        "      <td>    (E2F1,)</td>\n",
        "      <td> 612</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>177</th>\n",
        "      <td>    (TP63,)</td>\n",
        "      <td> 605</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>95 </th>\n",
        "      <td>  (POU5F1,)</td>\n",
        "      <td> 595</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>163</th>\n",
        "      <td>    (MITF,)</td>\n",
        "      <td> 578</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>127</th>\n",
        "      <td>   (PPARG,)</td>\n",
        "      <td> 556</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>68 </th>\n",
        "      <td>      (AR,)</td>\n",
        "      <td> 545</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>181</th>\n",
        "      <td>    (TP53,)</td>\n",
        "      <td> 540</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>60 </th>\n",
        "      <td>    (KLF4,)</td>\n",
        "      <td> 513</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>192</th>\n",
        "      <td>   (SUZ12,)</td>\n",
        "      <td> 485</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>159</th>\n",
        "      <td>    (REST,)</td>\n",
        "      <td> 467</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>131</th>\n",
        "      <td>   (SMAD4,)</td>\n",
        "      <td> 464</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>169</th>\n",
        "      <td>    (TAL1,)</td>\n",
        "      <td> 462</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>129</th>\n",
        "      <td>   (SIN3B,)</td>\n",
        "      <td> 454</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>120</th>\n",
        "      <td>   (GATA1,)</td>\n",
        "      <td> 436</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>35 </th>\n",
        "      <td>   (GATA2,)</td>\n",
        "      <td> 431</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>63 </th>\n",
        "      <td>  (TRIM28,)</td>\n",
        "      <td> 429</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>48 </th>\n",
        "      <td>   (KDM5B,)</td>\n",
        "      <td> 426</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>171</th>\n",
        "      <td>   (FOXP1,)</td>\n",
        "      <td> 410</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>46 </th>\n",
        "      <td>    (TET1,)</td>\n",
        "      <td> 405</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>123</th>\n",
        "      <td>   (ASH2L,)</td>\n",
        "      <td> 394</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>150</th>\n",
        "      <td>   (PPARD,)</td>\n",
        "      <td> 387</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>66 </th>\n",
        "      <td>    (TCF3,)</td>\n",
        "      <td> 381</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>144</th>\n",
        "      <td>    (TCF4,)</td>\n",
        "      <td> 380</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>165</th>\n",
        "      <td> (SMARCA4,)</td>\n",
        "      <td> 375</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>101</th>\n",
        "      <td>  (SETDB1,)</td>\n",
        "      <td> 371</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>47 </th>\n",
        "      <td>    (E2F4,)</td>\n",
        "      <td> 371</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>118</th>\n",
        "      <td>     (ZFX,)</td>\n",
        "      <td> 360</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>151</th>\n",
        "      <td>     (ERG,)</td>\n",
        "      <td> 337</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>113</th>\n",
        "      <td>   (SMAD3,)</td>\n",
        "      <td> 335</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>121</th>\n",
        "      <td>   (RCOR3,)</td>\n",
        "      <td> 329</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>146</th>\n",
        "      <td>   (SFPI1,)</td>\n",
        "      <td> 328</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>90 </th>\n",
        "      <td>    (MTF2,)</td>\n",
        "      <td> 327</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>21 </th>\n",
        "      <td>    (GABP,)</td>\n",
        "      <td> 322</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>65 </th>\n",
        "      <td>   (FOXO3,)</td>\n",
        "      <td> 312</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>67 </th>\n",
        "      <td>   (SALL4,)</td>\n",
        "      <td> 306</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1  </th>\n",
        "      <td>     (VDR,)</td>\n",
        "      <td> 300</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>178</th>\n",
        "      <td>    (MYCN,)</td>\n",
        "      <td> 295</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>116</th>\n",
        "      <td>   (FOXA2,)</td>\n",
        "      <td> 293</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>28 </th>\n",
        "      <td>   (EOMES,)</td>\n",
        "      <td> 292</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 352,
       "text": [
        "             tf  count\n",
        "189     (CREM,)    868\n",
        "24       (MYC,)    860\n",
        "180     (SPI1,)    780\n",
        "54      (SOX2,)    778\n",
        "185    (RUNX1,)    761\n",
        "51      (FLI1,)    725\n",
        "111    (CREB1,)    701\n",
        "19      (EGR1,)    653\n",
        "4      (NANOG,)    641\n",
        "172    (HNF4A,)    625\n",
        "31     (STAT3,)    617\n",
        "135     (E2F1,)    612\n",
        "177     (TP63,)    605\n",
        "95    (POU5F1,)    595\n",
        "163     (MITF,)    578\n",
        "127    (PPARG,)    556\n",
        "68        (AR,)    545\n",
        "181     (TP53,)    540\n",
        "60      (KLF4,)    513\n",
        "192    (SUZ12,)    485\n",
        "159     (REST,)    467\n",
        "131    (SMAD4,)    464\n",
        "169     (TAL1,)    462\n",
        "129    (SIN3B,)    454\n",
        "120    (GATA1,)    436\n",
        "35     (GATA2,)    431\n",
        "63    (TRIM28,)    429\n",
        "48     (KDM5B,)    426\n",
        "171    (FOXP1,)    410\n",
        "46      (TET1,)    405\n",
        "123    (ASH2L,)    394\n",
        "150    (PPARD,)    387\n",
        "66      (TCF3,)    381\n",
        "144     (TCF4,)    380\n",
        "165  (SMARCA4,)    375\n",
        "101   (SETDB1,)    371\n",
        "47      (E2F4,)    371\n",
        "118      (ZFX,)    360\n",
        "151      (ERG,)    337\n",
        "113    (SMAD3,)    335\n",
        "121    (RCOR3,)    329\n",
        "146    (SFPI1,)    328\n",
        "90      (MTF2,)    327\n",
        "21      (GABP,)    322\n",
        "65     (FOXO3,)    312\n",
        "67     (SALL4,)    306\n",
        "1        (VDR,)    300\n",
        "178     (MYCN,)    295\n",
        "116    (FOXA2,)    293\n",
        "28     (EOMES,)    292"
       ]
      }
     ],
     "prompt_number": 352
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "down_2 = pd.DataFrame( countTFcombos(cp73_chronicHigh_down).items(), \n",
      "             columns=[\"pair\", \"c\"] ).sort_index(by=\"c\", ascending=False)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "61562\n"
       ]
      }
     ],
     "prompt_number": 308
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "up_2 = pd.DataFrame( countTFcombos(cp73_chronicHigh_up).items(), \n",
      "             columns=[\"pair\", \"c\"] ).sort_index(by=\"c\", ascending=False)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "47501\n"
       ]
      }
     ],
     "prompt_number": 309
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "down_2[\"frac\"] = down_2[\"c\"] / float(len(cp73_chronicHigh_down.index))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 312
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "up_2[\"frac\"] = up_2[\"c\"] / float(len(cp73_chronicHigh_up.index))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 313
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "both_3 = up_3.merge(down_3, left_on=\"pair\", right_on=\"pair\", how=\"outer\", suffixes=(\"_up\",\"_down\")).fillna(0)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 298
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "both_2 = up_2.merge(down_2, left_on=\"pair\", right_on=\"pair\", how=\"outer\", suffixes=(\"_up\",\"_down\")).fillna(0)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 314
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "both_3[\"diff\"] = both_3[\"frac_up\"] - both_3[\"frac_down\"] "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 299
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "both_2[\"diff\"] = both_2[\"frac_up\"] - both_2[\"frac_down\"] "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 315
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "both_2.sort_index(by=\"diff\", ascending=False)[0:20] "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>pair</th>\n",
        "      <th>c_up</th>\n",
        "      <th>frac_up</th>\n",
        "      <th>c_down</th>\n",
        "      <th>frac_down</th>\n",
        "      <th>diff</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>0  </th>\n",
        "      <td>   (CREB1, CREM)</td>\n",
        "      <td> 543</td>\n",
        "      <td> 0.185641</td>\n",
        "      <td> 395</td>\n",
        "      <td> 0.097821</td>\n",
        "      <td> 0.087820</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>5  </th>\n",
        "      <td>    (CREB1, MYC)</td>\n",
        "      <td> 429</td>\n",
        "      <td> 0.146667</td>\n",
        "      <td> 358</td>\n",
        "      <td> 0.088658</td>\n",
        "      <td> 0.058009</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>8  </th>\n",
        "      <td>    (CREM, FLI1)</td>\n",
        "      <td> 424</td>\n",
        "      <td> 0.144957</td>\n",
        "      <td> 374</td>\n",
        "      <td> 0.092620</td>\n",
        "      <td> 0.052337</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>25 </th>\n",
        "      <td>  (CREB1, RUNX1)</td>\n",
        "      <td> 373</td>\n",
        "      <td> 0.127521</td>\n",
        "      <td> 307</td>\n",
        "      <td> 0.076028</td>\n",
        "      <td> 0.051494</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>27 </th>\n",
        "      <td>   (CREB1, FLI1)</td>\n",
        "      <td> 365</td>\n",
        "      <td> 0.124786</td>\n",
        "      <td> 301</td>\n",
        "      <td> 0.074542</td>\n",
        "      <td> 0.050244</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>78 </th>\n",
        "      <td>    (CREM, TAL1)</td>\n",
        "      <td> 293</td>\n",
        "      <td> 0.100171</td>\n",
        "      <td> 216</td>\n",
        "      <td> 0.053492</td>\n",
        "      <td> 0.046679</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1  </th>\n",
        "      <td>     (CREM, MYC)</td>\n",
        "      <td> 525</td>\n",
        "      <td> 0.179487</td>\n",
        "      <td> 537</td>\n",
        "      <td> 0.132987</td>\n",
        "      <td> 0.046501</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>22 </th>\n",
        "      <td>   (CREB1, SPI1)</td>\n",
        "      <td> 376</td>\n",
        "      <td> 0.128547</td>\n",
        "      <td> 337</td>\n",
        "      <td> 0.083457</td>\n",
        "      <td> 0.045090</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>164</th>\n",
        "      <td>   (CREB1, TAL1)</td>\n",
        "      <td> 240</td>\n",
        "      <td> 0.082051</td>\n",
        "      <td> 163</td>\n",
        "      <td> 0.040367</td>\n",
        "      <td> 0.041685</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>42 </th>\n",
        "      <td>    (CREM, KLF4)</td>\n",
        "      <td> 328</td>\n",
        "      <td> 0.112137</td>\n",
        "      <td> 285</td>\n",
        "      <td> 0.070579</td>\n",
        "      <td> 0.041557</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>106</th>\n",
        "      <td>  (CREB1, PPARG)</td>\n",
        "      <td> 272</td>\n",
        "      <td> 0.092991</td>\n",
        "      <td> 209</td>\n",
        "      <td> 0.051758</td>\n",
        "      <td> 0.041233</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>4  </th>\n",
        "      <td>    (CREM, SPI1)</td>\n",
        "      <td> 439</td>\n",
        "      <td> 0.150085</td>\n",
        "      <td> 442</td>\n",
        "      <td> 0.109460</td>\n",
        "      <td> 0.040625</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>134</th>\n",
        "      <td>   (CREB1, KLF4)</td>\n",
        "      <td> 258</td>\n",
        "      <td> 0.088205</td>\n",
        "      <td> 193</td>\n",
        "      <td> 0.047796</td>\n",
        "      <td> 0.040409</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>6  </th>\n",
        "      <td>   (CREM, RUNX1)</td>\n",
        "      <td> 426</td>\n",
        "      <td> 0.145641</td>\n",
        "      <td> 427</td>\n",
        "      <td> 0.105745</td>\n",
        "      <td> 0.039896</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>41 </th>\n",
        "      <td>   (CREB1, SOX2)</td>\n",
        "      <td> 329</td>\n",
        "      <td> 0.112479</td>\n",
        "      <td> 298</td>\n",
        "      <td> 0.073799</td>\n",
        "      <td> 0.038680</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>88 </th>\n",
        "      <td>  (CREB1, NANOG)</td>\n",
        "      <td> 286</td>\n",
        "      <td> 0.097778</td>\n",
        "      <td> 243</td>\n",
        "      <td> 0.060178</td>\n",
        "      <td> 0.037599</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>53 </th>\n",
        "      <td>   (CREM, PPARG)</td>\n",
        "      <td> 317</td>\n",
        "      <td> 0.108376</td>\n",
        "      <td> 289</td>\n",
        "      <td> 0.071570</td>\n",
        "      <td> 0.036806</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>308</th>\n",
        "      <td> (CREB1, TRIM28)</td>\n",
        "      <td> 197</td>\n",
        "      <td> 0.067350</td>\n",
        "      <td> 124</td>\n",
        "      <td> 0.030708</td>\n",
        "      <td> 0.036642</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>98 </th>\n",
        "      <td> (CREB1, POU5F1)</td>\n",
        "      <td> 280</td>\n",
        "      <td> 0.095726</td>\n",
        "      <td> 239</td>\n",
        "      <td> 0.059188</td>\n",
        "      <td> 0.036539</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>71 </th>\n",
        "      <td>   (CREB1, E2F1)</td>\n",
        "      <td> 298</td>\n",
        "      <td> 0.101880</td>\n",
        "      <td> 264</td>\n",
        "      <td> 0.065379</td>\n",
        "      <td> 0.036501</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 318,
       "text": [
        "                pair  c_up   frac_up  c_down  frac_down      diff\n",
        "0      (CREB1, CREM)   543  0.185641     395   0.097821  0.087820\n",
        "5       (CREB1, MYC)   429  0.146667     358   0.088658  0.058009\n",
        "8       (CREM, FLI1)   424  0.144957     374   0.092620  0.052337\n",
        "25    (CREB1, RUNX1)   373  0.127521     307   0.076028  0.051494\n",
        "27     (CREB1, FLI1)   365  0.124786     301   0.074542  0.050244\n",
        "78      (CREM, TAL1)   293  0.100171     216   0.053492  0.046679\n",
        "1        (CREM, MYC)   525  0.179487     537   0.132987  0.046501\n",
        "22     (CREB1, SPI1)   376  0.128547     337   0.083457  0.045090\n",
        "164    (CREB1, TAL1)   240  0.082051     163   0.040367  0.041685\n",
        "42      (CREM, KLF4)   328  0.112137     285   0.070579  0.041557\n",
        "106   (CREB1, PPARG)   272  0.092991     209   0.051758  0.041233\n",
        "4       (CREM, SPI1)   439  0.150085     442   0.109460  0.040625\n",
        "134    (CREB1, KLF4)   258  0.088205     193   0.047796  0.040409\n",
        "6      (CREM, RUNX1)   426  0.145641     427   0.105745  0.039896\n",
        "41     (CREB1, SOX2)   329  0.112479     298   0.073799  0.038680\n",
        "88    (CREB1, NANOG)   286  0.097778     243   0.060178  0.037599\n",
        "53     (CREM, PPARG)   317  0.108376     289   0.071570  0.036806\n",
        "308  (CREB1, TRIM28)   197  0.067350     124   0.030708  0.036642\n",
        "98   (CREB1, POU5F1)   280  0.095726     239   0.059188  0.036539\n",
        "71     (CREB1, E2F1)   298  0.101880     264   0.065379  0.036501"
       ]
      }
     ],
     "prompt_number": 318
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "both_3.sort_index(by=\"diff\", ascending=True)[0:20] "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>pair</th>\n",
        "      <th>c_up</th>\n",
        "      <th>frac_up</th>\n",
        "      <th>c_down</th>\n",
        "      <th>frac_down</th>\n",
        "      <th>diff</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>454  </th>\n",
        "      <td>      (AR, HNF4A, SOX2)</td>\n",
        "      <td> 151</td>\n",
        "      <td> 0.051624</td>\n",
        "      <td> 272</td>\n",
        "      <td> 0.067360</td>\n",
        "      <td>-0.015736</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>3698 </th>\n",
        "      <td>  (FOXA2, POU5F1, SOX2)</td>\n",
        "      <td>  86</td>\n",
        "      <td> 0.029402</td>\n",
        "      <td> 181</td>\n",
        "      <td> 0.044824</td>\n",
        "      <td>-0.015422</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>3288 </th>\n",
        "      <td>     (FOXA2, MYC, SOX2)</td>\n",
        "      <td>  90</td>\n",
        "      <td> 0.030769</td>\n",
        "      <td> 184</td>\n",
        "      <td> 0.045567</td>\n",
        "      <td>-0.014798</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1582 </th>\n",
        "      <td>   (GATA1, HNF4A, SOX2)</td>\n",
        "      <td> 111</td>\n",
        "      <td> 0.037949</td>\n",
        "      <td> 212</td>\n",
        "      <td> 0.052501</td>\n",
        "      <td>-0.014553</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>7128 </th>\n",
        "      <td>   (FOXA2, MYC, POU5F1)</td>\n",
        "      <td>  68</td>\n",
        "      <td> 0.023248</td>\n",
        "      <td> 152</td>\n",
        "      <td> 0.037642</td>\n",
        "      <td>-0.014395</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>4333 </th>\n",
        "      <td>      (AR, EGR1, SUZ12)</td>\n",
        "      <td>  82</td>\n",
        "      <td> 0.028034</td>\n",
        "      <td> 171</td>\n",
        "      <td> 0.042348</td>\n",
        "      <td>-0.014314</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>717  </th>\n",
        "      <td>   (FOXA2, HNF4A, SOX2)</td>\n",
        "      <td> 135</td>\n",
        "      <td> 0.046154</td>\n",
        "      <td> 244</td>\n",
        "      <td> 0.060426</td>\n",
        "      <td>-0.014272</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>15597</th>\n",
        "      <td>   (AR, SMARCA4, SUZ12)</td>\n",
        "      <td>  49</td>\n",
        "      <td> 0.016752</td>\n",
        "      <td> 125</td>\n",
        "      <td> 0.030956</td>\n",
        "      <td>-0.014204</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>6135 </th>\n",
        "      <td>      (AR, FOXA2, MITF)</td>\n",
        "      <td>  72</td>\n",
        "      <td> 0.024615</td>\n",
        "      <td> 154</td>\n",
        "      <td> 0.038138</td>\n",
        "      <td>-0.013522</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>3403 </th>\n",
        "      <td> (FOXA2, HNF4A, POU5F1)</td>\n",
        "      <td>  89</td>\n",
        "      <td> 0.030427</td>\n",
        "      <td> 177</td>\n",
        "      <td> 0.043834</td>\n",
        "      <td>-0.013406</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>345  </th>\n",
        "      <td>   (HNF4A, MYC, POU5F1)</td>\n",
        "      <td> 160</td>\n",
        "      <td> 0.054701</td>\n",
        "      <td> 275</td>\n",
        "      <td> 0.068103</td>\n",
        "      <td>-0.013402</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>32019</th>\n",
        "      <td>     (AR, OLIG2, SUZ12)</td>\n",
        "      <td>  34</td>\n",
        "      <td> 0.011624</td>\n",
        "      <td> 101</td>\n",
        "      <td> 0.025012</td>\n",
        "      <td>-0.013388</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>10621</th>\n",
        "      <td>   (AR, OLIG2, SMARCA4)</td>\n",
        "      <td>  58</td>\n",
        "      <td> 0.019829</td>\n",
        "      <td> 134</td>\n",
        "      <td> 0.033185</td>\n",
        "      <td>-0.013356</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1328 </th>\n",
        "      <td>    (FOXA2, HNF4A, MYC)</td>\n",
        "      <td> 116</td>\n",
        "      <td> 0.039658</td>\n",
        "      <td> 214</td>\n",
        "      <td> 0.052997</td>\n",
        "      <td>-0.013338</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>3271 </th>\n",
        "      <td>     (AR, STAT3, SUZ12)</td>\n",
        "      <td>  90</td>\n",
        "      <td> 0.030769</td>\n",
        "      <td> 178</td>\n",
        "      <td> 0.044081</td>\n",
        "      <td>-0.013312</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>6929 </th>\n",
        "      <td>    (E2F4, HNF4A, SOX2)</td>\n",
        "      <td>  69</td>\n",
        "      <td> 0.023590</td>\n",
        "      <td> 149</td>\n",
        "      <td> 0.036899</td>\n",
        "      <td>-0.013310</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1524 </th>\n",
        "      <td>    (HNF4A, SOX2, TCF4)</td>\n",
        "      <td> 112</td>\n",
        "      <td> 0.038291</td>\n",
        "      <td> 208</td>\n",
        "      <td> 0.051511</td>\n",
        "      <td>-0.013220</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>2794 </th>\n",
        "      <td>   (FOXA2, NANOG, SOX2)</td>\n",
        "      <td>  94</td>\n",
        "      <td> 0.032137</td>\n",
        "      <td> 183</td>\n",
        "      <td> 0.045319</td>\n",
        "      <td>-0.013183</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>2969 </th>\n",
        "      <td>      (AR, FOXA2, SOX2)</td>\n",
        "      <td>  92</td>\n",
        "      <td> 0.031453</td>\n",
        "      <td> 180</td>\n",
        "      <td> 0.044577</td>\n",
        "      <td>-0.013124</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>4240 </th>\n",
        "      <td>   (FOXA2, SOX2, STAT3)</td>\n",
        "      <td>  82</td>\n",
        "      <td> 0.028034</td>\n",
        "      <td> 165</td>\n",
        "      <td> 0.040862</td>\n",
        "      <td>-0.012828</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 320,
       "text": [
        "                         pair  c_up   frac_up  c_down  frac_down      diff\n",
        "454         (AR, HNF4A, SOX2)   151  0.051624     272   0.067360 -0.015736\n",
        "3698    (FOXA2, POU5F1, SOX2)    86  0.029402     181   0.044824 -0.015422\n",
        "3288       (FOXA2, MYC, SOX2)    90  0.030769     184   0.045567 -0.014798\n",
        "1582     (GATA1, HNF4A, SOX2)   111  0.037949     212   0.052501 -0.014553\n",
        "7128     (FOXA2, MYC, POU5F1)    68  0.023248     152   0.037642 -0.014395\n",
        "4333        (AR, EGR1, SUZ12)    82  0.028034     171   0.042348 -0.014314\n",
        "717      (FOXA2, HNF4A, SOX2)   135  0.046154     244   0.060426 -0.014272\n",
        "15597    (AR, SMARCA4, SUZ12)    49  0.016752     125   0.030956 -0.014204\n",
        "6135        (AR, FOXA2, MITF)    72  0.024615     154   0.038138 -0.013522\n",
        "3403   (FOXA2, HNF4A, POU5F1)    89  0.030427     177   0.043834 -0.013406\n",
        "345      (HNF4A, MYC, POU5F1)   160  0.054701     275   0.068103 -0.013402\n",
        "32019      (AR, OLIG2, SUZ12)    34  0.011624     101   0.025012 -0.013388\n",
        "10621    (AR, OLIG2, SMARCA4)    58  0.019829     134   0.033185 -0.013356\n",
        "1328      (FOXA2, HNF4A, MYC)   116  0.039658     214   0.052997 -0.013338\n",
        "3271       (AR, STAT3, SUZ12)    90  0.030769     178   0.044081 -0.013312\n",
        "6929      (E2F4, HNF4A, SOX2)    69  0.023590     149   0.036899 -0.013310\n",
        "1524      (HNF4A, SOX2, TCF4)   112  0.038291     208   0.051511 -0.013220\n",
        "2794     (FOXA2, NANOG, SOX2)    94  0.032137     183   0.045319 -0.013183\n",
        "2969        (AR, FOXA2, SOX2)    92  0.031453     180   0.044577 -0.013124\n",
        "4240     (FOXA2, SOX2, STAT3)    82  0.028034     165   0.040862 -0.012828"
       ]
      }
     ],
     "prompt_number": 320
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "pd.DataFrame( pair_counts.items(), columns=[\"pair\", \"c\"] ).sort_index(by=\"c\", ascending=False)[0:20]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>pair</th>\n",
        "      <th>c</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>2488849 </th>\n",
        "      <td>     (CREB1, CREM, MYC, SPI1)</td>\n",
        "      <td> 211</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>3803349 </th>\n",
        "      <td>     (CREB1, CREM, FLI1, MYC)</td>\n",
        "      <td> 204</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>16294561</th>\n",
        "      <td>   (MYC, NANOG, POU5F1, SOX2)</td>\n",
        "      <td> 200</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>5217325 </th>\n",
        "      <td>     (CREB1, CREM, E2F1, MYC)</td>\n",
        "      <td> 190</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>11050087</th>\n",
        "      <td>  (NANOG, POU5F1, SOX2, TCF3)</td>\n",
        "      <td> 187</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>12708301</th>\n",
        "      <td>    (CREB1, CREM, MYC, RUNX1)</td>\n",
        "      <td> 186</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>7282345 </th>\n",
        "      <td>    (CREB1, CREM, FLI1, SPI1)</td>\n",
        "      <td> 185</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>4381082 </th>\n",
        "      <td>      (CREM, FLI1, MYC, SPI1)</td>\n",
        "      <td> 179</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>18644028</th>\n",
        "      <td>   (CREB1, CREM, FLI1, RUNX1)</td>\n",
        "      <td> 179</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>10696485</th>\n",
        "      <td>   (CREB1, CREM, RUNX1, SPI1)</td>\n",
        "      <td> 176</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>5156839 </th>\n",
        "      <td>     (CREB1, CREM, MYC, SOX2)</td>\n",
        "      <td> 171</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>7336689 </th>\n",
        "      <td> (NANOG, POU5F1, RUNX1, SOX2)</td>\n",
        "      <td> 170</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>127909  </th>\n",
        "      <td>    (CREB1, CREM, MYC, NANOG)</td>\n",
        "      <td> 170</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>19162240</th>\n",
        "      <td>    (CREB1, CREM, HNF4A, MYC)</td>\n",
        "      <td> 168</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>11540489</th>\n",
        "      <td>      (CREM, E2F1, MYC, SPI1)</td>\n",
        "      <td> 164</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>10903125</th>\n",
        "      <td>     (CREB1, CREM, KLF4, MYC)</td>\n",
        "      <td> 163</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>12612862</th>\n",
        "      <td>     (CREM, FLI1, MYC, RUNX1)</td>\n",
        "      <td> 163</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>2798627 </th>\n",
        "      <td>  (CREM, NANOG, POU5F1, SOX2)</td>\n",
        "      <td> 161</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>10484106</th>\n",
        "      <td> (NANOG, POU5F1, SOX2, STAT3)</td>\n",
        "      <td> 161</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>6263013 </th>\n",
        "      <td>     (CREB1, FLI1, MYC, SPI1)</td>\n",
        "      <td> 160</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 231,
       "text": [
        "                                  pair    c\n",
        "2488849       (CREB1, CREM, MYC, SPI1)  211\n",
        "3803349       (CREB1, CREM, FLI1, MYC)  204\n",
        "16294561    (MYC, NANOG, POU5F1, SOX2)  200\n",
        "5217325       (CREB1, CREM, E2F1, MYC)  190\n",
        "11050087   (NANOG, POU5F1, SOX2, TCF3)  187\n",
        "12708301     (CREB1, CREM, MYC, RUNX1)  186\n",
        "7282345      (CREB1, CREM, FLI1, SPI1)  185\n",
        "4381082        (CREM, FLI1, MYC, SPI1)  179\n",
        "18644028    (CREB1, CREM, FLI1, RUNX1)  179\n",
        "10696485    (CREB1, CREM, RUNX1, SPI1)  176\n",
        "5156839       (CREB1, CREM, MYC, SOX2)  171\n",
        "7336689   (NANOG, POU5F1, RUNX1, SOX2)  170\n",
        "127909       (CREB1, CREM, MYC, NANOG)  170\n",
        "19162240     (CREB1, CREM, HNF4A, MYC)  168\n",
        "11540489       (CREM, E2F1, MYC, SPI1)  164\n",
        "10903125      (CREB1, CREM, KLF4, MYC)  163\n",
        "12612862      (CREM, FLI1, MYC, RUNX1)  163\n",
        "2798627    (CREM, NANOG, POU5F1, SOX2)  161\n",
        "10484106  (NANOG, POU5F1, SOX2, STAT3)  161\n",
        "6263013       (CREB1, FLI1, MYC, SPI1)  160"
       ]
      }
     ],
     "prompt_number": 231
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "pair_counts.items()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len( list( itertools.combinations( cz.tf.unique(), 2) ) )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 145,
       "text": [
        "15576"
       ]
      }
     ],
     "prompt_number": 145
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cz.groupby(\"tf\").count().sort_index(by=\"chea_id\", ascending=False)[0:20]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>chea_id</th>\n",
        "      <th>tf</th>\n",
        "      <th>tf_pmid</th>\n",
        "      <th>target</th>\n",
        "      <th>pmid</th>\n",
        "      <th>exp_type</th>\n",
        "      <th>cell_type</th>\n",
        "      <th>organism</th>\n",
        "      <th>date</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>tf</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>SUZ12</th>\n",
        "      <td> 72</td>\n",
        "      <td> 72</td>\n",
        "      <td> 72</td>\n",
        "      <td> 72</td>\n",
        "      <td> 72</td>\n",
        "      <td> 72</td>\n",
        "      <td> 72</td>\n",
        "      <td> 72</td>\n",
        "      <td> 72</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SOX2</th>\n",
        "      <td> 44</td>\n",
        "      <td> 44</td>\n",
        "      <td> 44</td>\n",
        "      <td> 44</td>\n",
        "      <td> 44</td>\n",
        "      <td> 44</td>\n",
        "      <td> 44</td>\n",
        "      <td> 44</td>\n",
        "      <td> 44</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MYC</th>\n",
        "      <td> 43</td>\n",
        "      <td> 43</td>\n",
        "      <td> 43</td>\n",
        "      <td> 43</td>\n",
        "      <td> 43</td>\n",
        "      <td> 43</td>\n",
        "      <td> 43</td>\n",
        "      <td> 43</td>\n",
        "      <td> 43</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>POU5F1</th>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>REST</th>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "      <td> 38</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TP63</th>\n",
        "      <td> 34</td>\n",
        "      <td> 34</td>\n",
        "      <td> 34</td>\n",
        "      <td> 34</td>\n",
        "      <td> 34</td>\n",
        "      <td> 34</td>\n",
        "      <td> 34</td>\n",
        "      <td> 34</td>\n",
        "      <td> 34</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TP53</th>\n",
        "      <td> 32</td>\n",
        "      <td> 32</td>\n",
        "      <td> 32</td>\n",
        "      <td> 32</td>\n",
        "      <td> 32</td>\n",
        "      <td> 32</td>\n",
        "      <td> 32</td>\n",
        "      <td> 32</td>\n",
        "      <td> 32</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SPI1</th>\n",
        "      <td> 31</td>\n",
        "      <td> 31</td>\n",
        "      <td> 31</td>\n",
        "      <td> 31</td>\n",
        "      <td> 31</td>\n",
        "      <td> 31</td>\n",
        "      <td> 31</td>\n",
        "      <td> 31</td>\n",
        "      <td> 31</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>AR</th>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>RUNX1</th>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "      <td> 30</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NANOG</th>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HNF4A</th>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "      <td> 28</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>E2F1</th>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>EGR1</th>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "      <td> 25</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MTF2</th>\n",
        "      <td> 23</td>\n",
        "      <td> 23</td>\n",
        "      <td> 23</td>\n",
        "      <td> 23</td>\n",
        "      <td> 23</td>\n",
        "      <td> 23</td>\n",
        "      <td> 23</td>\n",
        "      <td> 23</td>\n",
        "      <td> 23</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAT3</th>\n",
        "      <td> 22</td>\n",
        "      <td> 22</td>\n",
        "      <td> 22</td>\n",
        "      <td> 22</td>\n",
        "      <td> 22</td>\n",
        "      <td> 22</td>\n",
        "      <td> 22</td>\n",
        "      <td> 22</td>\n",
        "      <td> 22</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>GATA2</th>\n",
        "      <td> 21</td>\n",
        "      <td> 21</td>\n",
        "      <td> 21</td>\n",
        "      <td> 21</td>\n",
        "      <td> 21</td>\n",
        "      <td> 21</td>\n",
        "      <td> 21</td>\n",
        "      <td> 21</td>\n",
        "      <td> 21</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>CREB1</th>\n",
        "      <td> 20</td>\n",
        "      <td> 20</td>\n",
        "      <td> 20</td>\n",
        "      <td> 20</td>\n",
        "      <td> 20</td>\n",
        "      <td> 20</td>\n",
        "      <td> 20</td>\n",
        "      <td> 20</td>\n",
        "      <td> 20</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TCF3</th>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>CREM</th>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "      <td> 19</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 130,
       "text": [
        "        chea_id  tf  tf_pmid  target  pmid  exp_type  cell_type  organism  date\n",
        "tf                                                                             \n",
        "SUZ12        72  72       72      72    72        72         72        72    72\n",
        "SOX2         44  44       44      44    44        44         44        44    44\n",
        "MYC          43  43       43      43    43        43         43        43    43\n",
        "POU5F1       38  38       38      38    38        38         38        38    38\n",
        "REST         38  38       38      38    38        38         38        38    38\n",
        "TP63         34  34       34      34    34        34         34        34    34\n",
        "TP53         32  32       32      32    32        32         32        32    32\n",
        "SPI1         31  31       31      31    31        31         31        31    31\n",
        "AR           30  30       30      30    30        30         30        30    30\n",
        "RUNX1        30  30       30      30    30        30         30        30    30\n",
        "NANOG        28  28       28      28    28        28         28        28    28\n",
        "HNF4A        28  28       28      28    28        28         28        28    28\n",
        "E2F1         25  25       25      25    25        25         25        25    25\n",
        "EGR1         25  25       25      25    25        25         25        25    25\n",
        "MTF2         23  23       23      23    23        23         23        23    23\n",
        "STAT3        22  22       22      22    22        22         22        22    22\n",
        "GATA2        21  21       21      21    21        21         21        21    21\n",
        "CREB1        20  20       20      20    20        20         20        20    20\n",
        "TCF3         19  19       19      19    19        19         19        19    19\n",
        "CREM         19  19       19      19    19        19         19        19    19"
       ]
      }
     ],
     "prompt_number": 130
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "alib.plots.clusterHeatmap( czm_df, \"\", None, None, \n",
      "                            cluster_columns=True, \n",
      "                            cluster_rows=True, \n",
      "                            distmethod='jaccard')"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}