{
 "metadata": {
  "name": "pd_supp_tables"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import pandas\n",
      "import cPickle\n",
      "import pd_analysis as pda"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 2
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# set path for output of supplemental tables\n",
      "output_path = \"/data/adrian/Dropbox/Projects/Broad/PD_mouse/results/2013_june_20/\""
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 182
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# load amalgamated table of statistics by probeset from pickled pandas file.  This file was generated by the pd_calc_stats notebook or script\n",
      "cp_both_wide_info = cPickle.load( open(\"/data/adrian/Dropbox/Projects/Broad/PD_mouse/results/2013_june_13/cp_both_wide_info.pandas.pickle\") )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 3
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Note: to support consistent naming of columns and computation on the column metadata, column names for all of the statistics (which are concatenated in the cp_both_wide_info table) are stored as strings which are json encodings of a list of tags. \n",
      "These may be encoded and decoded using the functions in the TagMan objects (defined in pd_analysis.py).\n",
      "\n",
      "The tag types used in this dataset are:\n",
      "\n",
      "+ cmp : comparison\n",
      "+ ct : cell type\n",
      "+ mc : multiple testing correction type\n",
      "+ st : statistic type\n",
      "+ tt : statistical test type"
     ]
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 1: >= 2-fold Differentially Expressed genes in Dopamine Depleted CP73 (Striatonigral SPNs)"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# define the columns we're interested in here\n",
      "cn_cp73_bh_ttest_pval = pda.tm.e([(\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]), (\"ct\", \"cp73\"), (\"mc\", \"bh\"), (\"st\", \"pval\"), (\"tt\", \"welch ttest\")])\n",
      "cn_cp73_median_foldchange = pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]), (\"ct\", \"cp73\"), (\"st\", \"fc_medians\")])\n",
      "\n",
      "# select probesets with b-h adjusted p-values less than 0.10, and log2 fold change either < -1 or > +1\n",
      "cp73_dopamine_depletion = (cp_both_wide_info.select(\n",
      "    lambda x: cp_both_wide_info.ix[x, cn_cp73_bh_ttest_pval] < 0.10\n",
      "    and abs(cp_both_wide_info.ix[x, cn_cp73_median_foldchange]) >= 1\n",
      "))[[\"probe_id\", \"symbol\", \"gene_name\", \n",
      "    cn_cp73_bh_ttest_pval,\n",
      "    cn_cp73_median_foldchange\n",
      "    ]].sort_index(by=cn_cp73_median_foldchange, ascending=False)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 191
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp73_dopamine_depletion"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<pre>\n",
        "&lt;class 'pandas.core.frame.DataFrame'&gt;\n",
        "Index: 91 entries, 1438967_x_at to 1436094_at\n",
        "Data columns (total 5 columns):\n",
        "probe_id                                                                                                                                 87  non-null values\n",
        "symbol                                                                                                                                   87  non-null values\n",
        "gene_name                                                                                                                                87  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]    91  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"fc_medians\"]]                                   91  non-null values\n",
        "dtypes: float64(2), object(3)\n",
        "</pre>"
       ],
       "output_type": "pyout",
       "prompt_number": 194,
       "text": [
        "<class 'pandas.core.frame.DataFrame'>\n",
        "Index: 91 entries, 1438967_x_at to 1436094_at\n",
        "Data columns (total 5 columns):\n",
        "probe_id                                                                                                                                 87  non-null values\n",
        "symbol                                                                                                                                   87  non-null values\n",
        "gene_name                                                                                                                                87  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]    91  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"fc_medians\"]]                                   91  non-null values\n",
        "dtypes: float64(2), object(3)"
       ]
      }
     ],
     "prompt_number": 194
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# write to tab delimited file\n",
      "cp_73_dopamine_depletion.to_csv(output_path + \"pd_supptable_1.tab\", sep=\"\\t\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 183
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# count the number of distinct gene symbols\n",
      "# note that there are some probesets not mapped to entrez gene symbols; these are not counted as 'genes', but are still included in the tables\n",
      "cp_73_dopamine_depletion_genes = set(cp_73_dopamine_depletion.symbol.unique())\n",
      "cp_73_dopamine_depletion_genes.discard(nan)\n",
      "print \"CP73 Dopamine Depletion >= or <= 2-fold differentially expressed genes (unique Entrez gene symbols)): \", len(cp_73_dopamine_depletion_genes)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "CP73 Dopamine Depletion >= or <= 2-fold differentially expressed genes (unique Entrez gene symbols)):  67\n"
       ]
      }
     ],
     "prompt_number": 195
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 2: 2-fold Differentially Expressed genes in Dopamine Depleted CP101 (Striatopallidal SPNs)"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cn_cp101_bh_ttest_pval = pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]), (\"ct\", \"cp101\"), (\"mc\", \"bh\"), (\"st\", \"pval\"),(\"tt\", \"welch ttest\")])\n",
      "cn_cp101_median_foldchange = pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]),\n",
      "            (\"ct\", \"cp101\"),\n",
      "            (\"st\", \"fc_medians\"),\n",
      "        ])\n",
      "cp_101_dopamine_depletion = (cp_both_wide_info.select(\n",
      "    lambda x: cp_both_wide_info.ix[x, cn_cp101_bh_ttest_pval] < 0.10\n",
      "    and abs(cp_both_wide_info.ix[x, cn_cp101_median_foldchange]) >= 1\n",
      "))[[\"probe_id\", \"symbol\", \"gene_name\", \n",
      "   cn_cp101_bh_ttest_pval,\n",
      "   cn_cp101_median_foldchange\n",
      "]].sort_index(by=cn_cp101_median_foldchange, ascending=False)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 198
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp_101_dopamine_depletion"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<pre>\n",
        "&lt;class 'pandas.core.frame.DataFrame'&gt;\n",
        "Index: 91 entries, 1460330_at to 1452135_at\n",
        "Data columns (total 5 columns):\n",
        "probe_id                                                                                                                                  86  non-null values\n",
        "symbol                                                                                                                                    86  non-null values\n",
        "gene_name                                                                                                                                 86  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]    91  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"st\", \"fc_medians\"]]                                   91  non-null values\n",
        "dtypes: float64(2), object(3)\n",
        "</pre>"
       ],
       "output_type": "pyout",
       "prompt_number": 199,
       "text": [
        "<class 'pandas.core.frame.DataFrame'>\n",
        "Index: 91 entries, 1460330_at to 1452135_at\n",
        "Data columns (total 5 columns):\n",
        "probe_id                                                                                                                                  86  non-null values\n",
        "symbol                                                                                                                                    86  non-null values\n",
        "gene_name                                                                                                                                 86  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"mc\", \"bh\"], [\"st\", \"pval\"], [\"tt\", \"welch ttest\"]]    91  non-null values\n",
        "[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"st\", \"fc_medians\"]]                                   91  non-null values\n",
        "dtypes: float64(2), object(3)"
       ]
      }
     ],
     "prompt_number": 199
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp_101_dopamine_depletion_genes = set(cp_101_dopamine_depletion.symbol.unique())\n",
      "cp_101_dopamine_depletion_genes.discard(nan)\n",
      "print \"CP101 Dopamine Depletion >= or <= 2-fold differentially expressed genes (unique entrez gene symbols):\",  len(cp_101_dopamine_depletion_genes)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "CP101 Dopamine Depletion >= or <= 2-fold differentially expressed genes (unique entrez gene symbols): 77\n"
       ]
      }
     ],
     "prompt_number": 201
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Genes changing on Dopamine Depletion in both cell types:"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print \"instances for which *same* probeset has a significant change in both cell types: \"\n",
      "cp_both_wide_info.select(lambda x:  \n",
      "    abs(cp_both_wide_info.ix[x, cn_cp101_median_foldchange]) >= 1\n",
      "     and \n",
      "    abs(cp_both_wide_info.ix[x, cn_cp73_median_foldchange]) >= 1    \n",
      "     and \n",
      "    cp_both_wide_info.ix[x, cn_cp101_bh_ttest_pval] < 0.10     \n",
      "     and\n",
      "    cp_both_wide_info.ix[x, cn_cp73_bh_ttest_pval] < 0.10          \n",
      "    )[\n",
      "    [\"probe_id\", \"symbol\", \"gene_name\", \n",
      "    cn_cp73_median_foldchange,\n",
      "    cn_cp101_median_foldchange]\n",
      "].sort_index(by=\"symbol\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "instances for which *same* probeset has a significant change in both cell types: \n"
       ]
      },
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>probe_id</th>\n",
        "      <th>symbol</th>\n",
        "      <th>gene_name</th>\n",
        "      <th>[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"fc_medians\"]]</th>\n",
        "      <th>[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"st\", \"fc_medians\"]]</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>1454256_s_at</th>\n",
        "      <td>          NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                                               NaN</td>\n",
        "      <td>-1.371467</td>\n",
        "      <td> 1.517925</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1442754_at</th>\n",
        "      <td>   1442754_at</td>\n",
        "      <td> C030013G03Rik</td>\n",
        "      <td>                        RIKEN cDNA C030013G03 gene</td>\n",
        "      <td>-1.556349</td>\n",
        "      <td> 3.628062</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1425608_at</th>\n",
        "      <td>   1425608_at</td>\n",
        "      <td>         Dusp3</td>\n",
        "      <td> dual specificity phosphatase 3 (vaccinia virus...</td>\n",
        "      <td>-1.008864</td>\n",
        "      <td> 1.163462</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1438796_at</th>\n",
        "      <td>   1438796_at</td>\n",
        "      <td>         Nr4a3</td>\n",
        "      <td>   nuclear receptor subfamily 4, group A, member 3</td>\n",
        "      <td>-1.975423</td>\n",
        "      <td> 1.089134</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1423851_a_at</th>\n",
        "      <td> 1423851_a_at</td>\n",
        "      <td>        Shisa2</td>\n",
        "      <td>                  shisa homolog 2 (Xenopus laevis)</td>\n",
        "      <td>-1.026607</td>\n",
        "      <td> 1.851813</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1418726_a_at</th>\n",
        "      <td> 1418726_a_at</td>\n",
        "      <td>         Tnnt2</td>\n",
        "      <td>                              troponin T2, cardiac</td>\n",
        "      <td> 1.174274</td>\n",
        "      <td>-1.817546</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1436094_at</th>\n",
        "      <td>   1436094_at</td>\n",
        "      <td>           Vgf</td>\n",
        "      <td>                 VGF nerve growth factor inducible</td>\n",
        "      <td>-2.386596</td>\n",
        "      <td> 1.694763</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "output_type": "pyout",
       "prompt_number": 218,
       "text": [
        "                  probe_id         symbol  \\\n",
        "1454256_s_at           NaN            NaN   \n",
        "1442754_at      1442754_at  C030013G03Rik   \n",
        "1425608_at      1425608_at          Dusp3   \n",
        "1438796_at      1438796_at          Nr4a3   \n",
        "1423851_a_at  1423851_a_at         Shisa2   \n",
        "1418726_a_at  1418726_a_at          Tnnt2   \n",
        "1436094_at      1436094_at            Vgf   \n",
        "\n",
        "                                                      gene_name  \\\n",
        "1454256_s_at                                                NaN   \n",
        "1442754_at                           RIKEN cDNA C030013G03 gene   \n",
        "1425608_at    dual specificity phosphatase 3 (vaccinia virus...   \n",
        "1438796_at      nuclear receptor subfamily 4, group A, member 3   \n",
        "1423851_a_at                   shisa homolog 2 (Xenopus laevis)   \n",
        "1418726_a_at                               troponin T2, cardiac   \n",
        "1436094_at                    VGF nerve growth factor inducible   \n",
        "\n",
        "              [[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"fc_medians\"]]  \\\n",
        "1454256_s_at                                          -1.371467                                                        \n",
        "1442754_at                                            -1.556349                                                        \n",
        "1425608_at                                            -1.008864                                                        \n",
        "1438796_at                                            -1.975423                                                        \n",
        "1423851_a_at                                          -1.026607                                                        \n",
        "1418726_a_at                                           1.174274                                                        \n",
        "1436094_at                                            -2.386596                                                        \n",
        "\n",
        "              [[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"st\", \"fc_medians\"]]  \n",
        "1454256_s_at                                           1.517925                                                        \n",
        "1442754_at                                             3.628062                                                        \n",
        "1425608_at                                             1.163462                                                        \n",
        "1438796_at                                             1.089134                                                        \n",
        "1423851_a_at                                           1.851813                                                        \n",
        "1418726_a_at                                          -1.817546                                                        \n",
        "1436094_at                                             1.694763                                                        "
       ]
      }
     ],
     "prompt_number": 218
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# genes for which *any* of the probesets have a significant, 2-fold change in both cell types "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 219
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp_both_wide_info.select(lambda x:  \n",
      "    cp_both_wide_info.ix[x, \"symbol\"] in cp_101_dopamine_depletion_genes.intersection(cp_73_dopamine_depletion_genes)\n",
      "    )[\n",
      "    [\"probe_id\", \"symbol\", \"gene_name\", \n",
      "    cn_cp73_median_foldchange,\n",
      "    cn_cp101_median_foldchange]\n",
      "].sort_index(by=\"symbol\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>probe_id</th>\n",
        "      <th>symbol</th>\n",
        "      <th>gene_name</th>\n",
        "      <th>[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"fc_medians\"]]</th>\n",
        "      <th>[[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"st\", \"fc_medians\"]]</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>1442754_at</th>\n",
        "      <td>   1442754_at</td>\n",
        "      <td> C030013G03Rik</td>\n",
        "      <td>                        RIKEN cDNA C030013G03 gene</td>\n",
        "      <td>-1.556349</td>\n",
        "      <td> 3.628062</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1425608_at</th>\n",
        "      <td>   1425608_at</td>\n",
        "      <td>         Dusp3</td>\n",
        "      <td> dual specificity phosphatase 3 (vaccinia virus...</td>\n",
        "      <td>-1.008864</td>\n",
        "      <td> 1.163462</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1434472_at</th>\n",
        "      <td>   1434472_at</td>\n",
        "      <td>         Dusp3</td>\n",
        "      <td> dual specificity phosphatase 3 (vaccinia virus...</td>\n",
        "      <td>-0.616291</td>\n",
        "      <td> 0.656043</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1456769_at</th>\n",
        "      <td>   1456769_at</td>\n",
        "      <td>         Dusp3</td>\n",
        "      <td> dual specificity phosphatase 3 (vaccinia virus...</td>\n",
        "      <td>-0.591154</td>\n",
        "      <td> 0.982760</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1421079_at</th>\n",
        "      <td>   1421079_at</td>\n",
        "      <td>         Nr4a3</td>\n",
        "      <td>   nuclear receptor subfamily 4, group A, member 3</td>\n",
        "      <td>-1.591397</td>\n",
        "      <td> 0.792590</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1421080_at</th>\n",
        "      <td>   1421080_at</td>\n",
        "      <td>         Nr4a3</td>\n",
        "      <td>   nuclear receptor subfamily 4, group A, member 3</td>\n",
        "      <td>-0.731990</td>\n",
        "      <td> 0.656550</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1438796_at</th>\n",
        "      <td>   1438796_at</td>\n",
        "      <td>         Nr4a3</td>\n",
        "      <td>   nuclear receptor subfamily 4, group A, member 3</td>\n",
        "      <td>-1.975423</td>\n",
        "      <td> 1.089134</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1416700_at</th>\n",
        "      <td>   1416700_at</td>\n",
        "      <td>          Rnd3</td>\n",
        "      <td>                               Rho family GTPase 3</td>\n",
        "      <td>-1.328060</td>\n",
        "      <td> 0.938429</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1416701_at</th>\n",
        "      <td>   1416701_at</td>\n",
        "      <td>          Rnd3</td>\n",
        "      <td>                               Rho family GTPase 3</td>\n",
        "      <td>-0.841677</td>\n",
        "      <td> 1.620293</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1423851_a_at</th>\n",
        "      <td> 1423851_a_at</td>\n",
        "      <td>        Shisa2</td>\n",
        "      <td>                  shisa homolog 2 (Xenopus laevis)</td>\n",
        "      <td>-1.026607</td>\n",
        "      <td> 1.851813</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1423852_at</th>\n",
        "      <td>   1423852_at</td>\n",
        "      <td>        Shisa2</td>\n",
        "      <td>                  shisa homolog 2 (Xenopus laevis)</td>\n",
        "      <td>-0.814173</td>\n",
        "      <td> 1.293696</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1426721_s_at</th>\n",
        "      <td> 1426721_s_at</td>\n",
        "      <td>        Tiparp</td>\n",
        "      <td>        TCDD-inducible poly(ADP-ribose) polymerase</td>\n",
        "      <td>-1.204744</td>\n",
        "      <td> 0.431951</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1452160_at</th>\n",
        "      <td>   1452160_at</td>\n",
        "      <td>        Tiparp</td>\n",
        "      <td>        TCDD-inducible poly(ADP-ribose) polymerase</td>\n",
        "      <td>-0.654461</td>\n",
        "      <td> 1.072748</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1452161_at</th>\n",
        "      <td>   1452161_at</td>\n",
        "      <td>        Tiparp</td>\n",
        "      <td>        TCDD-inducible poly(ADP-ribose) polymerase</td>\n",
        "      <td>-0.901025</td>\n",
        "      <td> 0.483829</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1418726_a_at</th>\n",
        "      <td> 1418726_a_at</td>\n",
        "      <td>         Tnnt2</td>\n",
        "      <td>                              troponin T2, cardiac</td>\n",
        "      <td> 1.174274</td>\n",
        "      <td>-1.817546</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1424967_x_at</th>\n",
        "      <td> 1424967_x_at</td>\n",
        "      <td>         Tnnt2</td>\n",
        "      <td>                              troponin T2, cardiac</td>\n",
        "      <td> 1.185493</td>\n",
        "      <td>-1.679462</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1440424_at</th>\n",
        "      <td>   1440424_at</td>\n",
        "      <td>         Tnnt2</td>\n",
        "      <td>                              troponin T2, cardiac</td>\n",
        "      <td> 0.260850</td>\n",
        "      <td> 0.159263</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1436094_at</th>\n",
        "      <td>   1436094_at</td>\n",
        "      <td>           Vgf</td>\n",
        "      <td>                 VGF nerve growth factor inducible</td>\n",
        "      <td>-2.386596</td>\n",
        "      <td> 1.694763</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "output_type": "pyout",
       "prompt_number": 217,
       "text": [
        "                  probe_id         symbol  \\\n",
        "1442754_at      1442754_at  C030013G03Rik   \n",
        "1425608_at      1425608_at          Dusp3   \n",
        "1434472_at      1434472_at          Dusp3   \n",
        "1456769_at      1456769_at          Dusp3   \n",
        "1421079_at      1421079_at          Nr4a3   \n",
        "1421080_at      1421080_at          Nr4a3   \n",
        "1438796_at      1438796_at          Nr4a3   \n",
        "1416700_at      1416700_at           Rnd3   \n",
        "1416701_at      1416701_at           Rnd3   \n",
        "1423851_a_at  1423851_a_at         Shisa2   \n",
        "1423852_at      1423852_at         Shisa2   \n",
        "1426721_s_at  1426721_s_at         Tiparp   \n",
        "1452160_at      1452160_at         Tiparp   \n",
        "1452161_at      1452161_at         Tiparp   \n",
        "1418726_a_at  1418726_a_at          Tnnt2   \n",
        "1424967_x_at  1424967_x_at          Tnnt2   \n",
        "1440424_at      1440424_at          Tnnt2   \n",
        "1436094_at      1436094_at            Vgf   \n",
        "\n",
        "                                                      gene_name  \\\n",
        "1442754_at                           RIKEN cDNA C030013G03 gene   \n",
        "1425608_at    dual specificity phosphatase 3 (vaccinia virus...   \n",
        "1434472_at    dual specificity phosphatase 3 (vaccinia virus...   \n",
        "1456769_at    dual specificity phosphatase 3 (vaccinia virus...   \n",
        "1421079_at      nuclear receptor subfamily 4, group A, member 3   \n",
        "1421080_at      nuclear receptor subfamily 4, group A, member 3   \n",
        "1438796_at      nuclear receptor subfamily 4, group A, member 3   \n",
        "1416700_at                                  Rho family GTPase 3   \n",
        "1416701_at                                  Rho family GTPase 3   \n",
        "1423851_a_at                   shisa homolog 2 (Xenopus laevis)   \n",
        "1423852_at                     shisa homolog 2 (Xenopus laevis)   \n",
        "1426721_s_at         TCDD-inducible poly(ADP-ribose) polymerase   \n",
        "1452160_at           TCDD-inducible poly(ADP-ribose) polymerase   \n",
        "1452161_at           TCDD-inducible poly(ADP-ribose) polymerase   \n",
        "1418726_a_at                               troponin T2, cardiac   \n",
        "1424967_x_at                               troponin T2, cardiac   \n",
        "1440424_at                                 troponin T2, cardiac   \n",
        "1436094_at                    VGF nerve growth factor inducible   \n",
        "\n",
        "              [[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"fc_medians\"]]  \\\n",
        "1442754_at                                            -1.556349                                                        \n",
        "1425608_at                                            -1.008864                                                        \n",
        "1434472_at                                            -0.616291                                                        \n",
        "1456769_at                                            -0.591154                                                        \n",
        "1421079_at                                            -1.591397                                                        \n",
        "1421080_at                                            -0.731990                                                        \n",
        "1438796_at                                            -1.975423                                                        \n",
        "1416700_at                                            -1.328060                                                        \n",
        "1416701_at                                            -0.841677                                                        \n",
        "1423851_a_at                                          -1.026607                                                        \n",
        "1423852_at                                            -0.814173                                                        \n",
        "1426721_s_at                                          -1.204744                                                        \n",
        "1452160_at                                            -0.654461                                                        \n",
        "1452161_at                                            -0.901025                                                        \n",
        "1418726_a_at                                           1.174274                                                        \n",
        "1424967_x_at                                           1.185493                                                        \n",
        "1440424_at                                             0.260850                                                        \n",
        "1436094_at                                            -2.386596                                                        \n",
        "\n",
        "              [[\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp101\"], [\"st\", \"fc_medians\"]]  \n",
        "1442754_at                                             3.628062                                                        \n",
        "1425608_at                                             1.163462                                                        \n",
        "1434472_at                                             0.656043                                                        \n",
        "1456769_at                                             0.982760                                                        \n",
        "1421079_at                                             0.792590                                                        \n",
        "1421080_at                                             0.656550                                                        \n",
        "1438796_at                                             1.089134                                                        \n",
        "1416700_at                                             0.938429                                                        \n",
        "1416701_at                                             1.620293                                                        \n",
        "1423851_a_at                                           1.851813                                                        \n",
        "1423852_at                                             1.293696                                                        \n",
        "1426721_s_at                                           0.431951                                                        \n",
        "1452160_at                                             1.072748                                                        \n",
        "1452161_at                                             0.483829                                                        \n",
        "1418726_a_at                                          -1.817546                                                        \n",
        "1424967_x_at                                          -1.679462                                                        \n",
        "1440424_at                                             0.159263                                                        \n",
        "1436094_at                                             1.694763                                                        "
       ]
      }
     ],
     "prompt_number": 217
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 3: Pathway Enrichment for differentially expressed genes in Dopamine Depleted CP73"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 4: Pathway enrichment for differentially expressed genes in Dopamine Depleted CP101"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<pre>\n",
        "&lt;class 'pandas.core.frame.DataFrame'&gt;\n",
        "Index: 1433 entries, 1415800_at to 1460700_at\n",
        "Columns: 199 entries, [[\"cg\", \"ctrl\"], [\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"stddev\"]] to gene_name\n",
        "dtypes: float64(186), object(13)\n",
        "</pre>"
       ],
       "output_type": "pyout",
       "prompt_number": 95,
       "text": [
        "<class 'pandas.core.frame.DataFrame'>\n",
        "Index: 1433 entries, 1415800_at to 1460700_at\n",
        "Columns: 199 entries, [[\"cg\", \"ctrl\"], [\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"stddev\"]] to gene_name\n",
        "dtypes: float64(186), object(13)"
       ]
      }
     ],
     "prompt_number": 95
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 5: Differential expression in CP73 on chronic low-dose L-DOPA"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp73_chronicLow = (cp_both_wide_info.select(\n",
      "    lambda x: cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"mc\", \"bh\"),\n",
      "            (\"st\", \"pval\"),\n",
      "            (\"tt\", \"welch ttest\")\n",
      "        ])] < 0.10\n",
      "    and\n",
      "        abs(cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"st\", \"fc_medians\"),\n",
      "        ])]) >= 1\n",
      "))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 96
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp73_chronicLow_genes = set(cp73_chronicLow.symbol.unique())\n",
      "cp73_chronicLow_genes.discard(nan)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 98
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len(cp73_chronicLow_genes)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "pyout",
       "prompt_number": 100,
       "text": [
        "1046"
       ]
      }
     ],
     "prompt_number": 100
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp73_chronicLow_anymagnitude = (cp_both_wide_info.select(\n",
      "    lambda x: cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"mc\", \"bh\"),\n",
      "            (\"st\", \"pval\"),\n",
      "            (\"tt\", \"welch ttest\")\n",
      "        ])] < 0.10\n",
      "))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 112
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 6: Enriched Pathways in CP73 low-dose altered genes"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 7: Differential expression in CP73 on chronic high-dose"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp73_chronicHigh = (cp_both_wide_info.select(\n",
      "    lambda x: cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"mc\", \"bh\"),\n",
      "            (\"st\", \"pval\"),\n",
      "            (\"tt\", \"welch ttest\")\n",
      "        ])] < 0.10\n",
      "    and\n",
      "        abs(cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"st\", \"fc_medians\"),\n",
      "        ])]) >= 1\n",
      "))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 107
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp73_chronicHigh_genes = set(cp73_chronicHigh.symbol.unique())\n",
      "cp73_chronicHigh_genes.discard(nan)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 108
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len(cp73_chronicHigh_genes)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "pyout",
       "prompt_number": 109,
       "text": [
        "2000"
       ]
      }
     ],
     "prompt_number": 109
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len( cp73_chronicHigh_genes.intersection( cp73_chronicLow_genes ) )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "pyout",
       "prompt_number": 111,
       "text": [
        "926"
       ]
      }
     ],
     "prompt_number": 111
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 8: Differential expression (>= 2-fold) in CP73 and any significant change in CP101"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp73_chronicHigh_alsoLow = (cp_both_wide_info.select(\n",
      "    lambda x: cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"mc\", \"bh\"),\n",
      "            (\"st\", \"pval\"),\n",
      "            (\"tt\", \"welch ttest\")\n",
      "        ])] < 0.10\n",
      "    and\n",
      "        abs(cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"st\", \"fc_medians\"),\n",
      "        ])]) >= 1\n",
      "    and\n",
      "        cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"mc\", \"bh\"),\n",
      "            (\"st\", \"pval\"),\n",
      "            (\"tt\", \"welch ttest\")\n",
      "        ])] < 0.10        \n",
      "))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 113
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp73_chronicHigh_alsoLow_genes = set(cp73_chronicHigh_alsoLow.symbol.unique() )\n",
      "cp73_chronicHigh_alsoLow_genes.discard(nan)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 120
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len(cp73_chronicHigh_alsoLow_genes)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "pyout",
       "prompt_number": 121,
       "text": [
        "1748"
       ]
      }
     ],
     "prompt_number": 121
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp73_chronicHigh_NotLow = (cp_both_wide_info.select(\n",
      "    lambda x: cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"mc\", \"bh\"),\n",
      "            (\"st\", \"pval\"),\n",
      "            (\"tt\", \"welch ttest\")\n",
      "        ])] < 0.10\n",
      "    and\n",
      "        abs(cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"st\", \"fc_medians\"),\n",
      "        ])]) >= 1\n",
      "    and\n",
      "        cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"mc\", \"bh\"),\n",
      "            (\"st\", \"pval\"),\n",
      "            (\"tt\", \"welch ttest\")\n",
      "        ])] > 0.10        \n",
      "))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 117
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 9: Differential expression in CP101, low dose"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp_101_chronic_low = (cp_both_wide_info.select(\n",
      "    lambda x: cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp101\"),\n",
      "            (\"mc\", \"bh\"),\n",
      "            (\"st\", \"pval\"),\n",
      "            (\"tt\", \"welch ttest\")\n",
      "        ])] < 0.10\n",
      "    and\n",
      "        abs(cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicLow\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp101\"),\n",
      "            (\"st\", \"fc_medians\"),\n",
      "        ])]) >= 1\n",
      "))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 123
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len ( cp_101_chronic_low.symbol.unique() )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "pyout",
       "prompt_number": 135,
       "text": [
        "41"
       ]
      }
     ],
     "prompt_number": 135
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "h(cp_101_chronic_low)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'h' is not defined",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[1;32m<ipython-input-128-683383cc0319>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mh\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcp_101_chronic_low\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[1;31mNameError\u001b[0m: name 'h' is not defined"
       ]
      }
     ],
     "prompt_number": 128
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 10: Differential expression in CP101, high dose"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp_101_chronic_high = (cp_both_wide_info.select(\n",
      "    lambda x: cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp101\"),\n",
      "            (\"mc\", \"bh\"),\n",
      "            (\"st\", \"pval\"),\n",
      "            (\"tt\", \"welch ttest\")\n",
      "        ])] < 0.10\n",
      "    and\n",
      "        abs(cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicHich\", \"6-OHDA, chronicSaline\"]),\n",
      "            (\"ct\", \"cp101\"),\n",
      "            (\"st\", \"fc_medians\"),\n",
      "        ])]) >= 1\n",
      "))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 131
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len( cp_101_chronic_high.symbol.unique() )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "pyout",
       "prompt_number": 134,
       "text": [
        "181"
       ]
      }
     ],
     "prompt_number": 134
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 11: Comparison between genes altered by high dose in CP101 vs CP73"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 12: Pathways altered by high doses of L-DOPA in CP101 "
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 13: Significant differences between High and low-dose, CP73"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp_73_chronic_high_vs_low_up = (cp_both_wide_info.select(\n",
      "    lambda x: cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"mc\", \"bh\"),\n",
      "            (\"st\", \"pval\"),\n",
      "            (\"tt\", \"welch ttest\")\n",
      "        ])] < 0.10\n",
      "    and\n",
      "        cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"st\", \"fc_medians\"),\n",
      "        ])] >= 1\n",
      "))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 141
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len( [s for s in cp_73_chronic_high_vs_low_up.symbol.unique() if s is not nan ] )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "pyout",
       "prompt_number": 160,
       "text": [
        "99"
       ]
      }
     ],
     "prompt_number": 160
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp_73_chronic_high_vs_low_down = (cp_both_wide_info.select(\n",
      "    lambda x: cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"mc\", \"bh\"),\n",
      "            (\"st\", \"pval\"),\n",
      "            (\"tt\", \"welch ttest\")\n",
      "        ])] < 0.10\n",
      "    and\n",
      "        cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]),\n",
      "            (\"ct\", \"cp73\"),\n",
      "            (\"st\", \"fc_medians\"),\n",
      "        ])] <= -1\n",
      "))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 146
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print \"# of gene symbols with significant negative responses: \",  len( [s for s in cp_73_chronic_high_vs_low_down.symbol.unique() if s is not nan ] )"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "# of gene symbols with significant negative responses:  30\n"
       ]
      }
     ],
     "prompt_number": 166
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 14: Pathways enriched in CP73 dose-dependent positively-correlated genes"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 15: Pathways enriched in CP73 dose-dependent negatively correlated genes"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "No differentially expressed genes in comparison between chronic high and low groups for CP101"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "(cp_both_wide_info.select(\n",
      "    lambda x: cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]),\n",
      "            (\"ct\", \"cp101\"),\n",
      "            (\"mc\", \"bh\"),\n",
      "            (\"st\", \"pval\"),\n",
      "            (\"tt\", \"welch ttest\")\n",
      "        ])] < 0.10\n",
      "    and\n",
      "        abs(cp_both_wide_info.ix[x, \n",
      "        pda.tm.e([  (\"cmp\", [\"6-OHDA, chronicHigh\", \"6-OHDA, chronicLow\"]),\n",
      "            (\"ct\", \"cp101\"),\n",
      "            (\"st\", \"fc_medians\"),\n",
      "        ])]) >= 1\n",
      "))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<pre>\n",
        "&lt;class 'pandas.core.frame.DataFrame'&gt;\n",
        "Index: 0 entries\n",
        "Columns: 199 entries, [[\"cg\", \"ctrl\"], [\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"stddev\"]] to gene_name\n",
        "dtypes: float64(186), object(13)\n",
        "</pre>"
       ],
       "output_type": "pyout",
       "prompt_number": 159,
       "text": [
        "<class 'pandas.core.frame.DataFrame'>\n",
        "Index: 0 entries\n",
        "Columns: 199 entries, [[\"cg\", \"ctrl\"], [\"cmp\", [\"6-OHDA, chronicSaline\", \"Ascorbate, chronicSaline\"]], [\"ct\", \"cp73\"], [\"st\", \"stddev\"]] to gene_name\n",
        "dtypes: float64(186), object(13)"
       ]
      }
     ],
     "prompt_number": 159
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Supplementary Table 16: Significant differences in comparison between nested linear models"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}