<%inherit file="base.html"/>

<div class="container-fluid" style="margin-top: 60px">

<div class="row-fluid">

    <div class="span2">
    
    <ul class="nav nav-list">

    <li class="nav-header"><a href="#expt_design">Experimental Design</a></li>
    
    <li class="nav-header"><a href="#aims">AIM scores</a></li>
    
    <li class="nav-header"><a href="#normalization">Microarray Preprocessing & Normalization</a></li>
    
    <li class="nav-header"><a href="#pairwise">Pairwise Comparisons of Gene Expression Changes vs. Treatment Groups</a></li>

    <li class="nav-header"><a href="#regression">Regression Analysis of AIMS vs. Treatment and Gene Expression</a></li>
    
    <li class="nav-header"><a href="#enrichment">Pathway and Gene Set Enrichment Analyses</a></li>
    
    <li class="nav-header"><a href="#outliers">Behavioural and Gene Expression Outliers</a></li>

    <li class="nav-header"><a href="#data">Raw Data and Source Code</a></li>

    <li class="nav-header"><a href="#refs">Supplementary References</a></li>    
    
    </ul>
    </div>


    <div class="span9">
    
        <div class="section" id="expt_design">
        
        </div>
    
        <div class="section" id="aims">
        <h4>AIM Scores</h4>
        Abnormal Involuntary Movements (AIMs) were scored as described in (Cenci, 1997).  Four AIMS subtypes were recorded: locomotive dyskinesia (Lo), axial dystonia (Ax),  orolingual dyskinesia (Ol), and forelimb dyskinesia (Li).  A summary AIM score for each mouse was computed as the sum of all of the the Ax, Ol, and Li scores over time.  AIM subscores are highly correlated with each other, as shown in Fig. [XXX] (<a href="http://pd.sciencespace.org/aims">http://pd.sciencespace.org/aims</a>) and the rankings of mice by AIMs severity are similar whether subscores or the integrated score is used. 
        </div>

        <div class="section" id="normalization">
        <h4>Microarray Data Preprocessing and Normalization</h4>
        Affymetrix .CEL files were normalized using the RMA algorithm from the R Bioconductor 'affy' package (Gautier, 2004) to produce log2 expression values for each probeset.
        
        </div>
        
        <div class="section" id="pairwise">
        <h4>Pairwise Comparisons Between Treatments</h4>
        
        <p>For each (Dose , Cell Type) group, log2 fold change for each gene was computed as the difference in median or mean expression vs. the matched saline-treated group of mice.  Statistical significance of differences between groups was calculated with the Tukey HSD test using the 'aov' and 'TukeyHSD' procedures in the R stats module (R Core Team, 2012).  The resulting p-values were adjusted using the Benjamini-Hochberg procedure with 'multicomp.multipletests' in python statsmodels (Seabold, 2010) to control FDR over all probesets.  Nominal and Benjamini-Hochberg adjusted p-values are also reported for pairwise t-tests between each group and its saline control.
        </p>
        
        <p>For each treatment group, a table of top differentially expressed probesets was generated by setting a fold-change threshold of 2 at an FDR of 10%, based on the adjusted Tukey HSD statistics. </p>
        
        
        </div>
        

        <div class="section" id="regression">
        <h4>AIM regression analysis</h4>
        To test the hypothesis that differences in gene expression might be correlated to variation in AIM severity, we considered a series of nested linear models relating expression of each probeset, dose group, and AIM score: 
        
        <div style="padding: 10px">
        <table class="datatable">
        <tr> <th>  </th> <th> model </th> </tr>
        <tr> <td style="text-align: center"> a </td> <td> AIM ~ expression + C(Dose) + expression:C(dose) </td> </tr>
        <tr> <td style="text-align: center"> b </td> <td> AIM ~ expression + C(Dose) </td> </tr>
        <tr> <td style="text-align: center"> c </td> <td> AIM ~ C(Dose) </td> </tr>
        <tr> <td style="text-align: center"> d </td> <td> AIM ~ expression </td> </tr>
        </tr>
        </table>
        </div>
        
        Models were fit using the 'ols' procedure in the python statsmodels module (Seabold, 2010).
        Since there is obviously dose-dependence of both AIM score and gene expression changes (for those genes that do change), we assessed whether there was a significant relationship between expression and AIM score, given knowledge of the dose group, by comparing model (b) with model (c) using an F-test.
        
        A single mouse in the CP101, chronic low dose group, CP101 #1343, had no AIMs observed.  This mouse was excluded from the regression analysis, but the genes that were differentially expressed in CP101 #1343 compared to the rest of its group are listed in a table of its expression <a href="/outliers?cptype=CP101&mouseid=1343">outliers</a>.
        
        </table>
        </div>
        
        <div class="section" id="enrichment">
        <h4> Gene Set and Pathway Enrichment Analyses </h4>
        
        </div>

        <div class="section" id="outliers">
        <h4> Behavioural and Expression Outliers </h4>
        
        </div>

        <div class="section" id="data">
        <h4> Raw Data and Code for Reproducing Analysis </h4>
        
        </div>

        
        <div class="section" id="refs">
        <h4>References</h4>
        
        
        
        
        </div>
        
        
        
    </div>

</div>