## -*- coding: utf-8 -*-
<%inherit file="base.html"/>

<div class="container-fluid" style="margin-top: 60px">

<div class="row-fluid">

    <div class="span2">
    
    <ul class="nav nav-list">

    <li class="nav-header"><a href="#expt_design">Experimental Design</a></li>
    
    <li class="nav-header"><a href="#aims">AIM scores</a></li>
    
    <li class="nav-header"><a href="#normalization">Microarray Preprocessing & Normalization</a></li>
    
    <li class="nav-header"><a href="#pairwise">Pairwise Comparisons of Gene Expression Changes vs. Treatment Groups</a></li>

    <li class="nav-header"><a href="#regression">Regression Analysis of AIMS vs. Treatment and Gene Expression</a></li>
    
    <li class="nav-header"><a href="#enrichment">Pathway and Gene Set Enrichment Analyses</a></li>
    
    <li class="nav-header"><a href="#outliers">Behavioural and Gene Expression Outliers</a></li>

    <li class="nav-header"><a href="#data">Raw Data and Source Code</a></li>

    <li class="nav-header"><a href="#refs">Supplementary References</a></li>    
    
    </ul>
    </div>


    <div class="span9">
    
        <div class="section" id="expt_design">
        
        </div>
    
        <div class="section" id="aims">
        <h4>AIM Scores</h4>
        Abnormal Involuntary Movements (AIMs) were scored as described in (<a href="#cenci">Cenci, 1998</a>).  Four AIMS subtypes were recorded: locomotive dyskinesia (Lo), axial dystonia (Ax),  orolingual dyskinesia (Ol), and forelimb dyskinesia (Li).  A summary AIM score for each mouse was computed as the sum of all of the the Ax, Ol, and Li scores over time.  AIM subscores are highly correlated with each other, as shown in Fig. [XXX] (<a href="http://pd.sciencespace.org/aims">http://pd.sciencespace.org/aims</a>) and the rankings of mice by AIMs severity are similar whether subscores or the integrated score is used. 
        </div>

        <div class="section" id="normalization">
        <h4>Microarray Data Preprocessing and Normalization</h4>
        Affymetrix .CEL files were normalized using the RMA algorithm from the R Bioconductor 'affy' package (Gautier, 2004) to produce log2 expression values for each probeset.
        
        </div>
        
        <div class="section" id="pairwise">
        <h4>Pairwise Comparisons Between Treatments</h4>
        
        <p>For each (Dose , Cell Type) group, log2 fold change for each gene was computed as the difference in median or mean expression vs. the matched saline-treated group of mice.  Statistical significance of differences between groups was calculated with the Tukey HSD test using the 'aov' and 'TukeyHSD' procedures in the R stats module (R Core Team, 2012).  The resulting p-values were adjusted using the Benjamini-Hochberg procedure with 'multicomp.multipletests' in python statsmodels (Seabold, 2010) to control FDR over all probesets.  Nominal and Benjamini-Hochberg adjusted p-values are also reported for pairwise t-tests between each group and its saline control.
        </p>
        
        <p>For each treatment group, <a href="/">tables of the top differentially expressed probesets</a> were generated by setting a fold-change threshold of 2 and an FDR of of 0.10, based on the Benjamini-Hochberg adjusted Tukey HSD statistics. </p>
        
        
        </div>
        

        <div class="section" id="regression">
        <h4>AIM regression analysis</h4>
        To test the hypothesis that differences in gene expression might be correlated to variation in AIM severity, we considered a series of nested linear models relating expression of each probeset, dose group, and AIM score: 
        
        <div style="padding: 10px">
        <table class="datatable">
        <tr> <th>  </th> <th> model </th> </tr>
        <tr> <td style="text-align: center"> a </td> <td> AIM ~ expression + C(Dose) + expression:C(dose) </td> </tr>
        <tr> <td style="text-align: center"> b </td> <td> AIM ~ expression + C(Dose) </td> </tr>
        <tr> <td style="text-align: center"> c </td> <td> AIM ~ C(Dose) </td> </tr>
        <tr> <td style="text-align: center"> d </td> <td> AIM ~ expression </td> </tr>
        </tr>
        </table>
        </div>
        
        <p>Models were fit using the 'ols' procedure in the python statsmodels module (Seabold, 2010).
        Since there is obviously dose-dependence of both AIM score and gene expression changes (for those genes that do change), we assessed whether there was a significant relationship between expression and AIM score, given knowledge of the dose group, by comparing model (b) with model (c) using an F-test.  After correction for multiple testing, at an FDR of 0.10 there was one probeset in the CP73 group (<a href="/probeset/1431935_at">1431935_at</a>), and two probesets in CP101 (<a href="/probeset/1453220_at">1453220_at</a>, <a href="/probeset/1432254_at">1432254_at</a>) for which a linear model including expression was significantly better than predicting from dose group alone.  In all three instances, the ranges of expression measurements overlapped between the dose groups.  

        <div>
        <img src="/static/modelcomp/CP73_modelComparisonHistogram.png"/>
        </div>
        
        <div>
        <img src="/static/modelcomp/CP101_modelComparisonHistogram.png"/>
        </div>
        
        Full tables of probesets ranked by the p-values of these model comparisons are provided for <a href="/table#dataset/CP73">CP73</a> and <a href="/table#dataset/CP101">CP101</a>.</p>
        
        
        <p>A single mouse in the CP101, chronic low dose group, CP101 #1343, had no AIMs observed.  This mouse was excluded from the regression analyses, but the genes that were differentially expressed in CP101 #1343 compared to the rest of its group are listed in a table of its expression <a href="/outliers?cptype=CP101&mouseid=1343">outliers</a>.</p>
                
        </div>
        
        <div class="section" id="enrichment">
        <h4> Gene Set and Pathway Enrichment Analyses </h4>
        The probeset-level expression data were collapsed to Entrez gene symbols by selecting the probeset with the highest median log2 expression value over all samples.
        
        <h5> Hypergeometric Overlap Analysis </h5>        
        
        
        <h5> Gene Set Enrichment Analysis </h5>        
        Gene set enrichment analysis (GSEA) (<a href="#Subramanian2005">Subramanian, 2005</a>) was run to assess enriched gene sets in each treatment group vs. its saline control, as well as between the high vs. low dose chronic treatment groups for each cell type. GSEA was performed using version 2.0.10 of the the java implementation of GSEA, obtained from <a href="http://www.broadinstitute.org/gsea">http://www.broadinstitute.org/gsea</a>.
        
        
        </div>

        <div class="section" id="outliers">
        <h4> Behavioural and Expression Outliers </h4>
        
        After correction for multiple testing, we did not observe probesets with statistically significant interaction with AIM score beyond what would be expected from knowledge of dose group.  However, expression measurements of some probesets do vary in some mice with AIM scores at the extremes of their treatment groups.  Tables <a href="/aims">[XXX-XXX]</a> show genes varying within each mouse ranked by their z-scores compared to the rest of their group.
        
        
        </div>

        <div class="section" id="data">
        <h4> Raw Data and Code for Reproducing Analysis </h4>
        
        </div>

        
        <div class="section" id="refs">
        <h4>References</h4>
        
        <ul class="refs">
        <li id="cenci">
        Cenci MA, Lee CS,  Björklund A. <a href="http://onlinelibrary.wiley.com/doi/10.1046/j.1460-9568.1998.00285.x/abstract">L-DOPA-induced dyskinesia in the rat is associated with striatal overexpression of prodynorphin- and glutamic acid decarboxylase mRNA.</a>  Eur J Neurosci, 10(8):2694-706. 1998
        </li>
        
        <li>
        Gautier, L., Cope, L., Bolstad, B. M., and Irizarry, R. A. 2004. <a href="http://bioinformatics.oxfordjournals.org/content/20/3/307.short">affy---analysis of Affymetrix GeneChip data at the probe level.</a> Bioinformatics 20, 3 (Feb. 2004), 307-315
        </li> 
        
        <li>
Kelder T, van Iersel MP, Hanspers K, Kutmon M, Conklin BR, Evelo C, Pico AR. (2011) WikiPathways: building research communities on biological pathways. NAR doi: 10.1093/nar/gkr1074 URL: <a href="http://www.wikipathways.org/">http://www.wikipathways.org</a>
        </li> 

        <li>
Liberzon A, Subramanian A, Pinchback R, Thorvaldsdóttir H, Tamayo P, Mesirov JP. <a href="http://bioinformatics.oxfordjournals.org/content/27/12/1739.abstract">Molecular signatures database (MSigDB) 3.0.</a>  Bioinformatics. 2011 Jun 15;27(12):1739-40. doi: 10.1093/bioinformatics/btr260 URL: <a href="http://www.broadinstitute.org/gsea/msigdb">http://www.broadinstitute.org/gsea/msigdb</a>
        </li> 


        <li id="Subramanian2005">
        Subramanian A, Tamayo P, Mootha VK, Mukherjee S, Ebert BL, Gillette MA, Paulovich A, Pomeroy SL, Golub TR, Lander ES, Mesirov JP.  <a href="http://www.pnas.org/content/102/43/15545.long">Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles.  Proc Natl Acad Sci U S A. 2005 Oct 25;102(43):15545-50</a>
        </li>
        
        <li>
Seabold, J.S., Perktold, J.  <a href="http://conference.scipy.org/proceedings/scipy2010/pdfs/seabold.pdf">Statsmodels: Econometric and Statistical Modeling with Python.</a>  Proceedings of the 9th Python in Science Conference, 2010.
        </li> 

        <li>
R Core Team (2012). R: A language and environment for statistical computing. R Foundation for Statistical Computing,  Vienna, Austria. ISBN 3-900051-07-0, URL: http://www.R-project.org/.
        </li>
        
        <li>
        McKinney, W. <a href="http://www.dlr.de/sc/Portaldata/15/Resources/dokumente/pyhpc2011/submissions/pyhpc2011_submission_9.pdf">pandas: a Foundational Python Library for Data Analysis and Statistics.</a>  PyHPC 2011: Python for High Performance and Scientific Computing.
        </li>
                
        
        </ul>
        
        </div>
        
        
        
    </div>

</div>